<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Computational Discovery of Directional Pioneer Factors | Coder Coacher - Coaching Coders</title><meta content="Computational Discovery of Directional Pioneer Factors - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Microsoft-Research/">Microsoft Research</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Computational Discovery of Directional Pioneer Factors</b></h2><h5 class="post__date">2016-07-07</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/hJFbJKMrXVE" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">each year Microsoft Research hosts
hundreds of influential speakers from
around the world including leading
scientists renowned experts in
technology book authors and leading
academics and makes videos of these
lectures freely available
you
it's a pleasure to be here today and
thanks to a Jennifer Toni and Ernest for
the opportunity to speak to you today
and I also have really enjoyed working
with Richard Tatsu Hashimoto and Charlie
O'Donnell on this project and charlie is
here and I think he'd be happy to answer
any questions as well as as being what
I've tried to do today is to give you a
view into how new computational methods
and new kinds of computational
horsepower couldn't solve certain kinds
of problems in biology and the problem
we're going to begin with is a really
fascinating mystery right the
fascinating mystery is that actually
even though there are hundreds of
thousands of cognate motifs for a
transcription factor or where could bind
in the genome in fact less than 5% of
them are typically occupied and we
really don't know why and we also know
however it's controlled because the same
transcription factor typically is bound
differentially in two different cell
types even though this motif is shared
so what we're going to explore today is
a way to approach this problem and see
if we can discover some of the rules
that dictate how transcription factor
binding is condition-specific
and I'm going to begin by introducing
the idea of a pioneer transcription
factor a pioneer trip search from vector
is a special kind a transcription factor
that has the property can bind to closed
chromatin and many of you are familiar
that DNA is basically wound up on these
spools called nucleosomes and they
actually can condense and protect DNA
from binding by proteins and pioneer
transcription factors have the unusual
property that they can bind to closed
chromatin so in the past these kinds of
factors have been found in small numbers
handfuls they're characterized typically
by induct molecular biology studies and
so a question we had was was there a way
computationally to discover
large numbers of these factors or at
least discover a larger set of these
factors from direct computational
analysis of high-throughput data and
that's the journey we're going to embark
upon now I'm going to start my talk with
a clip the conclusion slide so at the
end you can hold me to these conclusions
right so I'm going to I'm going to
approach for claims that my talk today
the first is that a new methodology
called protein interaction quantitation
can predict TF binding transcription
factor binding from DNA
seek data that's claim one claim to is
that piq can identify pioneer factors
that regulate chromatin opening de novo
claim 3 that certain of these pioneer
transcription factors are actually
directional the fact that wasn't known
that means that they actually open up
promted in one direction and not the
other direction and finally that there
are settler factors that follow along in
the wake of these pioneer factors they
sort of act like bulldozers opening up
the chromatin and these settler factors
binding to the genome and affect
regulatory control so these are the four
claims i'm going to make today but
before I begin I'd like to discuss the
basic methodology of DNA seek the idea
is this if you have the genome it's
going to be occupied by a set of
proteins and what we would like to do in
a single experiment is to discover the
entire protein occupancy of the genome
now many of you have looked at chip seek
data other kinds of data before I know
typically you do this one factor at a
time but it would be really great if we
could actually discover the entire
protein occupancy of the genome or at
least approximate it in a single
experiment and so the insight here is
that if we take a nuclear extract and
expose it to a cutting enzyme DNA's one
it's going to cut the DNA wherever it's
exposed and what's going to happen is
the proteins are in a cache shadows on
the genome and if we could somehow
deconvolve those shadows and look at
what's going on with the DNA signal we'd
be able to figure out where in the
genome particular factors bound is shown
here the cartoon in the bottom
we're both histones and individual
regulatory proteins are casting shadows
causing reads not to be present mind you
what's going on is that wherever there
is a cut we can read where there is no
cut we cannot read so we're to get
so-called hypersensitive data wherever
there are cuts here are some metal plots
of what different factors do in terms of
their binding profiles plus or minus 200
bases from where the cognate motif is
and you can see the each factor has its
own particular profile or that it leaves
on the genome and so we can learn these
automatically part of piq is alerting
these profiles
without being given the profiles and
these look really great right because
they're actually the amalgamation of
thousands and thousands of such events
aligned at the motifs and down here is
the typical meta profile for ctcf those
of you can see if the ctcf binding site
is here we actually have ctcf organizing
histone occupancy along the genome
unfortunately we have a small problem
right which is that our read density
isn't that high and this which is barely
visible to you is the actual read
density of a typical ctcf binding site
so we need new computational methods to
be able to recover where these proteins
are binding based upon very very sparse
data so the idea then is that we're
going to take our input genome we're
going to start with we start today with
733 motifs approximately now we're using
thousand over a thousand over to scan
the genome and wherever there are
matches those are possible binding sites
for motif sensitive transcription
factors and then what we want to do is
to take the profiles you just saw in the
last slide match them against the input
data which is very sparse and make
predictions about which one's of these
motifs are occupied so the input to the
algorithm is a genome sequence a set of
motifs and DNA seeker from one or more
experiments we automatically infer these
profiles and we output for every single
motif occurrence of all the different
factors of probability that the bound or
not okay so that's the essence of the
algorithm now piq was designed
specifically to handle this low coverage
and noisy data that we're coping with to
integrate multiple experiments to build
statistical power and a scale to the
whole mammalian genome it really isn't
any good if we can only do this over
small windows and we're getting our high
spatial accuracy through the use of
boaties and we want to have very good
worst-case behavior and I'll talk a
little bit about that later on
ok so part of the challenge of modeling
this kind of binding to the genome is
establishing a relevant baseline
that is along the genome we have reads
that are occurring in a certain certain
positions and what we would like to do
is to figure out what is the latent rate
of reproduction that normally would
occur at that position in the genome
which relates to whether it's open or
closed so we're gonna make an assumption
here to help fill in the missing data
we're going to assume is that we can
establish a correlation between bases
based solely upon how far apart they are
in the genome up to four hundred bases
and based upon that stationary
correlation structure in the genome we
can build a model of what the base level
emissions across the genome are in a way
a rough approximation of this is how do
you build a model of the unoccupied
genome so then you can put on top of it
the profiles that you're seeing see
whether or not you're getting extra
hypersensitivity from the binding of a
particular factor okay and you model
this with a multivariate normal log
rates and produce a resulting rate which
is going to be sampled to produce the
counts at a particular base position and
so determine if a particular factor is
bound at a particular motif what we'll
do is we'll do a log likelihood ratio
test with them without the binding
profile of that particular factor put on
top of the native underlying base
production of reads at that location so
the model is going to learn all of this
automatically and the essential idea is
that this integrative model is going to
take multiple sources of evidence it's
going to take evidence from the profiles
that we learn automatically that our
motif or sequence specific it's going to
take information from multiple
experiments and the correlation between
them to help build strength so we learn
a correlation matrix across all the
experiments that we could help build
strength across them the other thing
that's really interesting is that
instead of treating the opposite strand
reads in the same exact framework we
to take the positive strand the negative
strand reads and treat the MS the same
is to independent experiments we learned
the correlation structure between the
reads and thus we automatically learn
how long the fragments are by the
correlation structure we learn the other
thing that we do is that the score of a
particular motif against the genome is
often used as a prior for figuring out
whether or not something binds or not
instead of using that score in a raw
form we actually automatically learn a
per experiment and per motif monotone
function that map's the PWM score into a
prior for binding in a particular
location so these different components
are very helpful in improving the
performance of piq just to give you an
idea the computational scaling or
running this on mammalian sized genomes
typically 1,500 motifs or so were
scanning for and simultaneously
determine whether they're occupied not
this of course represents hundreds of
thousands of times millions of
individual binding sites across the
genome we're using a window size at four
hundred bases for computing our a
stationary correlation structure across
the genome and what we do now is we
actually go out to the Amazon Cloud and
we buy all of our compute time on a bid
basis because we're using a lot of
compute time and so we typically will
buy eighty CPUs or more and it takes a
one day of clock time to do a set of
experiments but that equates to about
eighty days of CPU time so the kinds of
experiments that we're doing now
actually are going beyond what we can do
on a cluster because we're running
multiple jobs like the simultaneously so
it's very very compute intensive and we
have our own little structure for
managing this inside of a cloud since
this is a computational venue I thought
I'd just show you a little bit about
that in the elastic block store on ec2
we put in the data for the job there's a
master that actually organizes all the
slaves that run on this and all of this
state on the right hand side is
transitory because we can lose our job
if we get outbid by somebody else on
to it suffice to say EBS services a
stable repository for the state of the
job and any intermediate results and it
worked very very effectively for us I
mean we have fallen in love with the
cloud only because the kinds of jobs
that we can run now in this context we
wouldn't be able to do at all and I
think that as molecular biology
progresses and you're dealing with
individual experiments have 200 million
reads you really need to be able to
commit you know months if not years of
CPU time to analyze one experiment our
current methodologies which I'm not
discussing today take almost a cpu year
per experiment to run ok so my first
claim was that we could detect the
binding and be competitive with with
other methodologies and as you can see
if we compare piq to other methods for
analyzing DNA seek to see a centipede
and DGF the area under the receiver
operating curve is much better for piq
in fact it's so good that in certain
cases we are tempted to use it as a
substitute for chip seek that isn't a
single DNA C cux parent we can begin to
learn the entire protein occupancy of
the genome all of the regulators and
begin to use that to predict regulatory
architecture but there's some caveats
the caveat sar important for certain
factors the correspondence with chip
seek bound sites is as an AUC of 0.9 and
above but the effectors DNA seek to tech
table only if it has a strong binding
motif it binds in DNA's accessible
regions and it has a strong binding
affinity to protect from DNA's so about
75 of the 302 factors I showed you on
the slide before are strongly detectable
via this methodology okay so let's claim
one I hope I've supported that let's go
on to claim to that we can identify
these pioneer factors
so recall that a typical transcription
factor has hundreds of thousands of
possible binding sites
we'd like to locate the ones that
actually does bind to and to look at
this we did a time series of endodermal
development into DNA seek experiments at
all these time points we asked a very
simple question which is in that time
series where does the chromatin open and
what factors are sitting there and can
we from that time series data recover
information about the structure of this
so we embarked upon the systematic
pioneer identification and the figure at
the bottom I think is indicative of the
way we approach this which is that if
you have a time zero and then another
time if you don't get a pioneering
activity the chromatin is not going to
open up and become accessible on the
other hand if you do have pioneering
activity you'll see the factor land and
clear out the chromatin to make it
accessible it's also possible that
certain pioneers are not going to open
chromatin in the same way and we won't
be able to detect them and furthermore
obviously we can only detect pioneers
that are active in two particular
lineage or cells that we look at so that
said we establish a number of metrics to
look at these different properties of
the cell and one such metric we call the
pioneer index which is an index that
measures how much chromatin opens from
one time to the next time by a factor we
also looked at the what we called the
chromatin opening index score which is
how much chromatin is open around a
given factor that's bound to the genome
we found these two were highly
concordant which allowed us to use
individual experiments to estimate
pioneer activity
we also found as you might expect that
around pioneers they're very social
creatures right there are a lot of other
factors that bind around them by what we
call a social index score but we're
going to focus on the chromatin opening
index score to actually determine which
factors are pioneers and which ones are
not
so these
pioneer transcription factors have
identifiable profiles we found 120 of
them and you can see the one that we
want that we identified have these
profiles that have elevated
hypersensitivity around them whereas
other factors do not have this elevated
hypersensitivity so we can classify the
factors into pioneers and non pioneers
now I know what you're thinking which is
okay that's all great computational
analysis is it really true so what we
did was we designed a toll based
insertion vector where we could place a
punitive pioneer or non pioneer motif in
it and see whether or not it enabled
retinoic acid receptor to bind to a
construct that would turn on GFP and we
did a number of controls to ensure for
example that simply adding the Pioneer
along with this motif did not turn on
GFP because we wanted to eliminate the
possibility that it in selvan was an
enhancer of gene activity
it was simply opening up the chromatin
and what you can see from this assay is
that eight of the nine pioneer factors
that we tested actually did turn on GFP
in a context where the genome was
presumably closed so this provides us
further support for the idea that these
factors are opening up the genome and
enabling other factors to bind to turn
on genes now I showed you this slide
before but I want to return to her for a
moment anybody sees it something unusual
about this about any of these profiles
here like for queso or nf1 well they're
asymmetric and so because these are non
palindromic motifs you can orient them
and it turns out that they actually have
more hypersensitivity on one side than
the other we thought aha perhaps these
things are directional maybe they're
serving like genomic parentheses in a
way and they're opening up the genome in
one direction but not the other
direction so we went back and tested
our system using the same toll insertion
vector putting the motif and to this
concert both in the forward and the
reverse direction and what you would
expect you get differential activation
of GFP depending on which way that motif
was oriented right because it would open
up the chromatin in one way versus the
other way and we selected these by
computing asymmetry asymmetry index
score and picked the ones that were high
in that score so when we did that we
found that in fact we could see the
directional activity of these
directional pioneers that is when the
motifs were put in the proper
orientation with respect to their
directional activity they did activate
GFP and when they put in the reverse
direction they did not so this further
helped us substantiate the idea that
these factors were directional in their
activity and we'll come back to another
support of that claim in a moment so
that's claim three and finally claim
four as we came up with some with an
analysis that looked at factors that
depended upon chromatin being open to
bind but that did not open it themselves
and so we thought that these might be
settler factors they dependent upon open
chromatin but they did not act like
bulldozers and open the chromatin up all
on their lonesome so in order to
validate this what we did was we built a
dominant negative assay so we took a
pioneer and we chopped off its head
right we chopped off its activating
domain the part that we thought was
mediating the opening activity and only
left the DNA binding domain so the
theory was when you expressed this
dominant negative pioneer it's going to
reduce hypersensitivity in the region
and also reduce the binding of proximal
settlers all right so when we did this
and we express this dominant negative
version both of nfy a and n are f1 which
are two of the pioneers that I talked to
you about earlier what happens is the
net hypersensitivity is substantially
reduced around those sites natively in
the genome
the other thing we wanted to see was
whether or not when we expressed these
down negative constructs for directional
pioneers it would eliminate or to reduce
the binding of downstream events so here
we go here's NF y a which is a
directional pioneer we turn on the
dominant negative and when the binding
site that's native in the genome is
downstream of the NF y a site binding is
reduced however when the binding site
for in this case emic is upstream it is
not reduced so this is for the support
of the idea that these are actually
operating directionally the final thing
I'd like to say is it appears that
pioneers appear to be conserved between
human and mouse so with that the
chromatin opening index of factors and
their homologs
both in mouse and human and they're
highly correlated so finally I just like
to say that I could go back to my
original claims that piq is a highly
accurate means of predicting T of
finding from DNA Sita we can identify
novel pioneer factors that regulate
chromatin opening certain these pioneer
factors are directional and finally that
the settlor factors follow the Pioneer
factor binding and loss of Pioneer
activity can reduce the binding of
settler factors so what I've presented
to you today is a multi-level model of
gene regulation where the Pioneer
factors are coming in opening up
chromatin in defined regions enabling
other factors to bind which helps us
understand why only a small fraction of
the cognate motifs of a given factor are
normally occupied all right I'll stop
right there and take any questions thank
you very much
yes Martha we did not look for peptide
signatures we did look at the profiles
and we cluster them in multi-dimensions
you can actually tell the family of the
transcription factor from its protection
profile so there's some mechanism there
the other thing I should say obviously
following on Martha's very nice talk
this morning about factor binding
specificities that when you are looking
at motifs alone you can't tell all the
time which specific factor is binding
because you're gonna look at families of
factors that bind the same cognate motif
so what we produce is an estimate of
which family members are binding
particular locations in the genome but
that's a good question Martha any other
questions yes
no we have not looked at IPS induction
but presumably we would see things like
AK 4 and some of the other ox ox na dog
so forth factors as pioneers but we're
not sure but they've been reported to be
pioneers in the past yes
I wish I knew the answer to that
question how your question is how
important this mechanism for opening
chromatin
it appears to us at does open chromatin
it appears to explain a lot of it what
we're doing now is we're actually
looking at the following question which
is I was telling you about which
pioneers appear to act in a static
situation at a particular time point
which factors are opening chromatin a
second door question you might ask is if
you have a set of time points of a
particular experimental treatment or of
a developmental time series between two
time points when you have differential
activity can you tell me which factors
are really important for that particular
differential activity in that particular
lineage and so now we're discovering a
secondary set of factors that appear to
modulate the activity the primary
factors that I've discussed here the
only questions okay thanks Dave again we</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>