<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Evolutionary Forces in Humans and Pathogens | Coder Coacher - Coaching Coders</title><meta content="Evolutionary Forces in Humans and Pathogens - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Microsoft-Research/">Microsoft Research</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Evolutionary Forces in Humans and Pathogens</b></h2><h5 class="post__date">2016-07-07</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/UabhTjKmrP8" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">each year Microsoft Research helps
hundreds of influential speakers from
around the world including leading
scientists renowned experts in
technology book authors and leading
academics and makes videos of these
lectures freely available
you
everyone for regrouping here we're going
to have another two sessions of talks
here and our next speaker will be part
of savetti she's of Harvard systems
biology center and also the Broad
Institute she'll be talking today about
evolutionary forces in humans and
microbes so thank you very much partes
thank you very much I'm really excited
about the opportunity to speak here
today at Microsoft Research we're in a
really in particular in such an exciting
time in computational biology where we
have an opportunity to make so many
discoveries and in particular my
interest is in the discovery of the
evolution of humans and the pathogens of
different microbes that infect humans so
I'm just gonna give you overview of my
talk which is essentially an overview of
my lab it's the areas that we're
interested in which is human evolution
pathogens and then method and tool
development around around these
different enterprises so it has long
been known that natural selection shaped
the human genome I think in basically in
1858 Darwin and Wallace published the
principles of natural selection that
really you know galvanized this effort
but even before then individuals many
individuals have been thinking about
evolution but we have very few examples
that have been elucidated some of the
classic ones that you might know about
our malaria resistance driven by
mutations in the hemoglobin gene one of
which causes sickle cell mutation
mutations and pigmentation pathway that
have evolved since humans spread out of
Africa into Europe and in Asia the
ability to drink milk lactose tolerance
as we domesticated cattle these are some
classic examples and we know about them
because of their sort of strong
biological hypothesis for evolution but
also because they leave distinctive
signals behind in our genomes and the
interesting thing is that as we now
begin to explore there are many such
signals in the genome yet we don't know
what many of them do and the wonderful
thing is now when we have sequencing
data from lots of genomes we can mine
these sequence data and what we see is
that actually when evolution does occur
leaves behind a footprint in the genome
as a mutation emerges and spreads
through the population there's a pattern
that can be detected and it's actually
quite a large pattern a large signal a
large footprint in the genome and the
question is what is also the mutation
that's driving it and that's sort of
what my lab is very much been interested
in and we can do this type of research
just by looking at a hundred individuals
even sometimes less than that but a
hundred individuals is usually
sufficient to find many of the most
distinctive signals of evolution in the
genome just a hundred people in a
population today and we you know collect
their DNA and begin to look for these
patterns and they look a little bit like
this so this is what I call the genomic
signal of natural selection
if you look here essentially I'll just
move over you look here is basically a
cartoon showing a set of individuals and
at some point they have a mutation that
emerges in one individual one person's
germline and that mutation if it's more
likely to enhance the survival or the
reproductive success of individuals who
carry it it'll you know be more likely
to that personal and more likely survive
reproduce pass it on to their children
their children will survive reproduce
pass on to their children's children and
that mutation will spread very quickly
right so that's the classic single of
natural selection that we are used to
but on a genomic level we can see those
patterns because if the mutation has
occurred recently in time you take those
individuals of population you have
before selection and after selection
there's two things that are generally
driving variation in the genome there
are different patterns but two classic
ones that we know about or you hear
about in your biology textbook one's
mutation and the other one is
recombination so mutation generates new
variants in the population this is just
a cartoon of five physicians along a
chromosome and mutation will create
these sort of new variants in the
population and recombination will
reshuffle the pattern on which those
variants exist on a chromosome and so
over time that recombination and
mutation Widow's away the background on
which a mutation arises and you get the
sort of mosaic pattern but when a
mutation emerges and spreads very
quickly through the population
it'll do so in such rapid time that
recombination will not have time to
whittle away that those relationships
and so we can see different distinctive
patterns in the genome one of them is
what we call long-range correlations
it's basically a segment of the genome
in which all of the variants are
correlated to each other another one is
population differences is when evolution
occurs in one population subject to some
pressure but not in other populations
it'll drive these large differences
between populations and there's a number
of other signals like those that we use
to detect the signal of selection in the
genome and now with data from sort of
the era of genomics we were getting more
and more resources in order to explore
that we have complete genomes of humans
chimpanzees Mouse gorillas all sorts of
organisms we've then moved to large
databases of genetic variation then we
started getting large datasets of
genotype variation and now full sequence
data it allows us for the first time to
really do these genome-wide scans of
natural selection but one of the issues
that have arise when we do this kind of
work is as I said that that mutation
emerges and it spreads through the
population it takes a whole chunk of the
genome with it right the whole like
chunk of a chromosome and so within that
it becomes very difficult to figure out
who is the actor right with that large
footprint you're trying to find a single
mutation and one of the things is here
are the different kinds of tests that
people have used to identify signals of
selection and you can see over a very
large region this is about a mega base
long there are mutations all along it
that are spiking over what you'd expect
that are telling you that there's a
signal of selection somewhere over here
but it's sort of very diffuse and so
what we decided to do is we developed a
composite framework or composite
likelihood framework to combine all of
these tests together to say what is the
probability that each mutation across a
region of the genome is the causal
mutation given its score by any one of
these tests and as we do that
essentially it cleans up the signal very
strongly so this is sort of the power of
using multiple lines of largely
independent evidence and here you can
see that essentially
a lot of this noise got cleared up
because any one of these mutations may
have a high score by one test but it's
very unlikely that they'd have a high
score by all the tests that they weren't
the driver and here you can see this
peak right across here and this is the
phenyl alanine - leucine change in a
gene that's critical for pigmentation so
even though we detected a signal
surrounding pigmentation you can see
it's a very large region region as we
use the combined framework it can
localize to the gene and to an amino
acid change within the gene and once we
sort of done that on a number of
examples that we were sort of felt
pretty comfortable with we you know are
positive controls things we expected to
be under evolutionary pressure we
applied it to lots of regions in the
genome and we could see a lot of those
signals clearing up and it gave us the
first opportunity to really explore what
are the mutations that are driving human
evolution and here's just I'm gonna give
you like a couple quick examples of the
type of work that we've done since that
time so here's another region it was
eight hundred kilobases 800,000 bases
long of a region that had selection you
can see this large sort of diffuse
signal but when we applied the test it
really localized to just a few mutations
that were high-scoring in this region
and right at the top here is a valine to
alanine change in a gene called a dart
and interestingly
Adar had been known before so there are
many genes across this but it really
localized there and this gene had
actually become sort of important in
evolution before we we found this and
stickleback fish where it actually
drives the development of scale and so
as stickleback fish we're moving from
freshwater sorry from seawater into
freshwater they were losing their scale
by mutations in this pathway and in
humans this pathway regulates many
things things on the ectoderm but in
particular hair and sweat but also a
number of other things and so in a study
that we did led by two members of my
group Yannick amber off and seizure Carl
seems you're wearing they we what we did
is we first explored the origins of that
mutation and showed that it emerged
sometime in central China about thirty
thousand years ago and
spread throughout central China and out
into the Asian populations and Native
American populations we placed the
mutation into a mouse the single point
mutation we placed into a mouse and what
we show it is actually that the mice who
got that in single mutation actually got
thicker hair on their body got smaller
mammary glands and and increase the
number of eccrine glands on their paws
and that is actually those the hair and
the sweat have been recapitulated in
human populations as well so that the
Association is there in humans and also
we've now been able to show in a model
system that that single mutation can
make those kinds of changes also so
we've essentially created a platform by
which we can explore this kind of work
and so starting with the genome scan and
then trying to find map those variants
using these types of composite
frameworks we then move to computational
annotations try to figure out all of the
different mutations in that region and
see what they're doing and what might be
important in them and once we do we move
to these functional experiments we first
try some high-throughput functional
experiments because you want to really
have good confidence in what you're
looking at before you might move to
model systems and then we can come to
adaptive hypothesis and in this other
paper done by four members of my group
here we am and a lot of the much larger
team we were able to show that a
mutation and what we call a toll-like
receptor which is one of the important
genes in innate immunity and this
particular one tlr5 regulates
response to flagellated bacteria so the
flagellum in these bacteria drive a
response that causes signaling cascade
that's very important and we showed that
a single amino acid change in the
toll-like receptor also changes
signaling to in response to this
flagellum and so these are the kinds of
explorations we can have interestingly
this particular gene we know that if you
knock it out mice don't respond to
basically an equivalent of typhoid fever
so that's one possible mechanism of
evolution but really that's we're still
the exploratory phase and that moves us
to pathogens
with it I've uh sort of come from a
medical degree as well as a research
background and very interested in
pathogens that's sort of a lot of what
my lab focuses on and so I'm just going
to tell you one story of of many but one
that we have explored which is the top
signal that we identified in one of our
first scans of an African population a
population from Nigeria the Yoruba we
found that the top signal by our first
scan localized to a gene called large on
chromosome 22 and so we began to explore
what the gene does and what might be the
driver because essentially when we start
these scans we're generating new
hypotheses we don't know what could
actually be the driver so we're trying
to explore and find it and what we
learned in our explorations is that this
gene actually modifies alpha district
like him so this is large this yellow
thing here and here's alpha district
like count alpha district iq is very
important in muscle development but it's
also important on lots of different
tissue types and so when this modifies
alpha district like an it puts these
glycosyl it causes it to be cleaved and
sent up to the cell surface where it
does a number of activities but one of
the things that is important for is it
can be co-opted by microbes so alpha
district lichen is basically an entry
route for these microbes without this
modification by large the Vie of these
microbes can't get in and one of the
most important ones that's been sort of
implicated is a virus and a pad and a
set of virus it's called arena viruses
and one of the most important ones is
loss of virus so essentially I was
beginning to explore this and saying
well this is very interesting because
last I was actually named after a city
in Nigeria where it was first discovered
and we have a signal selection in
Nigeria that's linked to this and so
we're trying to explore whether or not
that might be the cause and as a medical
student I never really heard about Lhasa
so it was sort of strange to me that
this could possibly be one of the
strongest drivers of human evolution yet
it's sort of little known but as it
turns out it's much more important than
I had realized at first so these this is
a little bit of there's not that much
known about it because there's small
groups working on it and it's a
biosafety level 4 virus meaning like
Ebola
Marburg smallpox you can only work on
the live virus in a in a BL for lab
right and it's an extreme it you know
has extreme fatality rates extreme
potential for being aerosolized or going
airborne there's a bioterrorist threat
so it's very difficult to work in this
area but as it turns out it's estimated
to infect over 20,000 people a year
sorry
caused fatality of over 20,000 people a
year and in fact many many more it
actually like I said not only was named
after Nigeria but if you look village in
Nigeria but if you look actually across
West Africa and look at seroprevalence
Studies antibody response there's a
great deal of exposure throughout West
Africa and in Nigeria
an antibody test suggests that 21
percent of the population has been
infected with this virus in Sierra Leone
Liberia and Guinea over 50 percent of
some populations some group villages
have been infected and it's not
surprising that it's that widespread
because this mouse here this its mass
Tamizh Nadu lensses is the sort of the
house Mouse that you see in many places
in West Africa it's developed it's
co-evolved to persistently breed Lhasa
in its in its blood in certain
populations that we've tested over 50
percent of the rodents have carried this
virus in its blood and in different
times it varies in population population
but some populations of extreme exposure
to this virus and so you know thinking
about the fact that we call it a
category a agent because sometimes
fatality rates can be we've seen the 80%
right but about 50 percent often but yet
you're saying that doesn't quite work
because most people actually get
infected and don't even get sick at all
we suggest that this is not only the
only category a agent biosafety level 4
agent that's a huge public health crisis
but also it might be a potential ancient
selective force driving evolution so
this is pointing us in that direction
and so once we found this signal in 2007
we went to the Arroyo specialist
teaching hospital set up a diagnostic
lab began collecting samples we also
partnered with Tulane University and Bob
Gary here to work with them on a site
that they've been developing in Sierra
Leone or the Kenema government hospital
focused on diagnostics and we have great
support from the NIH to sequence both
the virus and then humans infected with
the virus to analyze them to see what
are the hosts and viral components that
are important and susceptibility I'm
just going to show you a few images I
just want to warn you that these the
next image slide is gonna have some
disturbing images so look away if you're
not I'll let you know when it comes back
this is what we think about when we
think of la severe virus so these are
some of the patients that we see these
are ones we've seen in thoroughly own
and you can see this is the classic
sense of when you think about Ebola and
these different kinds of hemorrhagic
fevers extreme extreme bleeding from
from every mucosal bed the next slide is
now we're off that slide is um what many
of our cases look like so you have that
classic example you have in your head of
a hemorrhagic fever and what you're
expecting to find but what we found is
that a lot of our patients actually look
a lot like this where you know they just
look sleepy a little ill malaise but
nothing extreme but these ones actually
have similar fatality rates right so
it's not that many of these individuals
died but they do not look what you would
think is the classic example so it the
more we began to visit those regions and
stay in those regions the more we
thought are we talking are we just not
detecting things that are as lethal but
just don't look what we are like what
we're expecting to see and so you know
as I showed you before the
seroprevalence suggested this actually
is a widespread virus right it is uh
it's across all of this West Africa and
you're seeing High Sierra prevalence in
many populations we also see that animal
habitat is quite wide as well and this
nate'll Ensis has Co evolved and finally
as we sequence the virus ourselves just
most recently we sequenced over 100
strains from Nigeria and Sierra Leone
from humans and rodents we identified we
looked at the the data and it looks like
the virus is actually over millennia old
so it's at least about 1200 years old
circulating in Nigeria and seems to have
spread out of Nigeria into the rest of
West Africa within the last few hundred
years and actually entered Sierra Leone
likely around 150 years ago that's by
our estimates and so we're talking about
a virus that is not emerging right it's
actually circulating widely and has been
so for a long long time and we see the
same kinds of patterns when we analyze
data from Ebola and elsewhere and so it
led us to think are we dealing with
emerging disease or merging diagnosis
and that's a big force of the work in my
lab and it really affects the way we
think about care that essentially we
learned so much because you know we
started in here analyzing large
sequences in the lab but we had to set
up Diagnostics in order to do this work
but in doing so we actually engaged the
community because the hospitals were the
were empowered the community was engaged
we started to affect care and as we
collected samples the communities began
to respond and we started to essentially
get this large database of not just
individuals of Lassa but with lots of
other fevers that we don't know and so
we're really going out to explore what
are the causes of fever that we're not
detecting that are circulating widely
that we may not know about and then the
last thing in the last couple of minutes
I just want to mention to you that also
our lab is focused on methods so I told
you about one method the composite
method that we use to detect selection
I'll just show you a couple frames that
show you of the other kinds of methods
we've developed with one of my students
de bruschetta now his and his brother
you care we had he had entered my lab
also interested you know being an MD PhD
was interested in figuring out how we
could mind these large Health databases
and find patterns find new patterns and
so we looked at as we there was an issue
that there are lots of statistics that
people use but many of them only have
certain strengths and one issue that we
found was that actually different none
of the methods were what we call general
and equitable in the sense that they're
able to detect any kind of pattern and
data and for any noiseless function give
it a score of one so to give it a
perfect score of finding every noiseless
function in data and then as basically
we add noise to the function to be able
to drop off so that we can get some sort
of a score that's meaningful so that we
can mine these large databases looking
for the strongest relationships and so
this is just an example of
mutual information by crass cough that's
one of the state-of-the-art mutual
information techniques and one of the
issues you find with it as you look at
these 27 different functions we were
exploring with different noise levels
that the this kind of test actually
weights towards things that are more
linear so this sinusoidal relationship
that's noiseless we'll get the same
score as a linear relationship that has
lots of noise and you see those sort of
issues all over the way the cross so if
you're mining large databases looking
for relationships that are not you know
always linear and a lot of obvious
monotonic it's gonna be a struggle and
so we went out and we looked explore
that lots of different types of
state-of-the-art methods that have
different strengths like a Spearman is
very good for linear things but then
drops off for others or other
relationships that just kind of behave
differently for different functions we
went out and we set out to develop a new
statistic and this just shows you the
properties of the original statistic we
developed and we're continuing to work
on it but essentially it is approaching
more of what you're looking for which is
something that gives every noiseless
function a score of 1 and then drops off
in a linear fashion as you increase
noise to the data so now you can set out
to explore it explore your data in a
more sort of systematic way and we in a
paper that we published a couple of into
late 2011 we then showed how this method
could be powerful on data from wide
ranging from the world health to
baseball to yeast to microbiome to find
new patterns and data and the other the
rest of my lab also develops lots of
different tools and the Suites of tools
including visualization tools so Dave
and andreas in my lab have done a great
deal of work of developing a
visualization system to rapidly explore
your datasets put them in in these find
the patterns in the data and then and
look at them visually the Ben Frey and
Patrick forelli were a major force
behind developing sweep which is a way
of looking at natural selection in
genomic data and Eric found in my group
developed something called epi sampler
which allows us to do rapid collections
in the field with barcodes with GIS of
getting sort of high quality data
clinical data and sample data for the
projects that we do in global health and
with that I just want to say that a lot
of this work is done for my lab I just
show holiday cards so that you can see
the lab as it's evolved over the years
these are our annual holiday cards in
2009-2010 my incredibly creative group
every year sort of surprising folks with
their new methods and last year Gangnam
style and then this year this is
actually the first this is the first
time you'll see our latest card which is
SNL themed and so this sort of the end
of the night and you can see like they
go all the way because we've reenacted
almost every shot in the in the show my
human evolution collaborators my
collaborators from Nigeria from Sierra
Leone I'm extraordinarily grateful to
have such an amazing team and support
from so many different institutions
including the Gates Foundation thank you
yeah sure yep absolutely yes so the
question was can you say a little bit
more about the composite score and how
does it work so the composite it's a
parametric test in the sense that what
we do is we develop simulations of lots
of different types of selection
occurring so they're very good
simulations that one can do of human
evolution of genomic data and the
evolution of genomic data and so we have
these different tests many of which are
just sort of sort of nonparametric tests
that individuals have developed but you
essentially what you do is you for each
of those tests the composite tests
there's five that go into it you say you
create distributions of what the
mutation is that you care about and all
of the other mutations nearby those so
for every every time you have one
mutation that's the driver there's about
10,000 mutations next to it
you're trying to figure out which how
it's different from the rest and so
essentially you go variant by variant
and say what's the probability that it
is the causal variant
based on its score by this test right so
you're comparing it to those simulations
and then you basically sum the
likelihoods across all of the different
tests and give a final composite
likelihood the score so yeah
with that yeah how this thing sewed it
all depends on basically issues
suggesting that it depends on the
simulations that's a larger discussion I
can talk with you offline because it's
it's one of those things where I show
you the pretty cartoon and say this is
natural selection shifting but then like
99% of our work is spent with every
caveat as to why there are other things
that could influence what you're seeing
so but there's a lot of work that's been
done on those simulations - sure yep so
it was the ones I had mentioned at the
beginning but I'm briefly which is like
just one is population differences so if
you have an example of the mutation that
causes resistance to malaria there's
some population that has malaria another
population doesn't so that meeting is
only going to spread in the one
population and what it's going to do is
drive major differences between
populations so a classic example is a
mutation that protects from Plasmodium
vivax malaria where it has achieved a
hundred percent prevalence in Africa and
is absent elsewhere in the world and so
natural selection will drive these major
differences so one big thing is you just
go position by position across the
genome and look for places where there's
been a huge divide between and then the
other one I mentioned with those
long-range correlations when that
mutation emerges and is rises in
prevalence taking a whole chunk of the
genome that's another composite test so
it's essentially looking for those
different there statistics that
identified those different patterns the
only last question thank you</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>