<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Interpreting Cancer Copy-Number Alterations | Coder Coacher - Coaching Coders</title><meta content="Interpreting Cancer Copy-Number Alterations - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Microsoft-Research/">Microsoft Research</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Interpreting Cancer Copy-Number Alterations</b></h2><h5 class="post__date">2016-07-07</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/TEKzOYJPaHA" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">each year Microsoft Research helps
hundreds of influential speakers from
around the world including leading
scientists renowned experts in
technology book authors and leading
academics and makes videos of these
lectures freely available
you
Institute and arson associate member at
the Broad Institute
yeah okay so first of all I want to
thank the organizers for inviting me to
this exciting conference to speak and
please feel free to ask questions at the
end but also in the middle if you'd like
I prefer to be interrupted my laboratory
is interested like Ben Raphael's in
cancer genetics we have a particular
focus on copy number changes in cancer
and so I want to talk today about some
of the approaches we've been taking to
understand and interpret the copy number
changes we observe in cancer and and
we're in all our studies were motivated
by an understanding that copy number
profiles in cancer are complex so this
is a view washed out view unfortunately
so you can't really see some of the some
of it but the of 3,000 different cancers
each cancer is a column extending from
chromosome one the chromosome 22 regions
in red are amplified and regions in blue
are deleted and if there were a better
color in this slide we you could see
that every single region of the genome
is amplified in some fraction of cancers
and every single region of the genome is
deleted in some fraction of cancers and
more than that there are these complex
patterns where some samples have many
events across the genomes other samples
have very few events some regions have
many events and those events are
sometimes very big events and sometimes
they're very small events and then and
then there can be correlations within
there so there's a lot of information in
these data and been Raphael was
mentioning some approaches to to mining
these data and and and and we're taking
some of the same approaches in some
different ones so the types of questions
that these data can answer first of all
which are the driver genes you know not
every although every region of the
genome is amplified and also deleted in
different cancers we don't want to say
that every region of the genome is
important in driving cancer not every
gene amplification
is important in driving cancer so which
of these events are driving cancer and
what are the genes that they're
targeting another question is what other
important genes are affected along the
way to cancer as cancers evolve do they
genetically modify genes that may not
drive cancer but nevertheless shape the
behavior of those cancers and third what
are the patterns of change and what do
these apply about the mutational
mechanisms by which cancers arise and
the functional consequences of these
mutations so those are some of the
questions that we're trying to answer
and I'm going to give you some vignettes
to describe the sorts of approaches
we're taking the so you know as Ben
Raphael mentioned algorithms are
important and we're doing a lot of
trying to develop algorithms networks
are important but also data is important
so one of the main drivers of our work
is just increasing amounts of data this
is true for Ben as well and John and you
know that so you know just from TCGA
which have been described from April
2010 to July of 2012 just from doing
this sort of two-year time period there
was an explosion of cancer copy number
profiles where they were generated by
this one project you know from 2000
samples that had been profiled to over
14,000 and now it's much higher than
this so a huge amount of data is coming
la online now this is actually this
includes tumors and DNA from paired
normal tissue so about only and so these
numbers you have to divide them in half
to see that determine the number of
cancers that's still around 7,000
cancers that have high resolution copy
number profiles that have been generated
and so this these sorts of data give us
a great sort of playpen to work in to
try to figure out new things about
cancer I'm going to describe some
specific results from a paper we
published recently that that was done
through the TCGA pan cancer effort which
again was part of as well and where we
performed a joint analysis of 5000
cancer genomes across 12 cancer types
since then we've done additional
analyses of larger numbers of cancers
over 8,000 across more cancer types the
general themes of other reasons you know
the results I'm gonna be presenting
are mirrored in those larger analyses
and some of the reasons those results of
those larger analyses are online and
I'll point you to that website so we we
took each of these cancers and we
generated copy number profiles and then
we tried to interpret those couple of
profiles so how do we interpret them
well the first thing we want to do is
figure out which are the what's the
temporal sequence of events that led to
the copy number changes that we observe
and that's not what we get from the
original data the initial data is we
have two alleles and we can get the copy
number of each allele so for instance
here the one allele has zero copies and
here another the other allele has one
copy and then if you look down here this
one allele has one copy and the other
allele has two copies turns out that so
we can get these sort of allele at the
copy number profiles turns out that
they're not they're not phased so we
don't know whether you know say this is
from mom and this is from dad we don't
know whether this is from mom and this
is from that or vice versa from from the
initial data so the first thing we do is
we rephase these data to generate the
simplest profile we can on each allele
thinking that you know using Occam's
razor essentially - that the valent to
have the fewest breakpoints meaning the
fewest changes are needed used events
are needed to explain that profile so
when we do that here we take you know we
we combine these sorts of different
allylic information to get sort of a
high-level changes here and so where
this allele is relatively unaffected and
then we can determine the most
parsimonious set of events that could
explain our data and by parsimonious and
not only saying the minimal number of
events but also the taking the events
that had the maximum likelihood of
occurring and so for instance here we
can see what what we determined was that
we had a deletion involving this region
here a complex deletion so involved this
region here another may be in the same
event or a different event deletion
right here and then another deletion
here so not clear whether those happen
at the same time or you know adjacent
times but that ended up putting these
originally distant regions of the genome
at
together so that joined these distant
regions of the genome together so that
one amplification event could lead to
what looks like three separate
amplification events here because
they're deletions had preceded them and
then likewise we had an amplification of
this chromosome arm here followed by a
deletion of part of it so that so from
that sort of analysis we can determine
the most likely set of events that
occurred in each sample and across the
five thousand cancers we identified a
total of 200 mm events for a median of
39 of copy number of events per cancer
so so now we have a database describing
these 200 mm events their locations the
number of copies of change and their
lengths and so we can start to analyze
those and see what patterns emerge the
first thing we saw is something that
we'd seen actually previously in an
analysis of 3,000 cancers we've
published three years ago which is that
the most frequent copy number events are
the shortest ones and they decrease in
frequency with increasing length a with
exception of events that it could occupy
an entire chromosome arm exactly one
chromosome arm or exactly two chromosome
arms meaning a whole chromosome and
these are much more common than you'd
expect from the overall distribution
right so the most likely explanation for
this is that these events are occurring
due to a different mechanism these arm
level events are according to a
different mechanism from the focal
events and indeed there are mechanisms
that have been described that can
explain this right you can have
chromosomal missed segregation during
mitosis can lead to amplifications or
deletions of whole chromosome arms but
not a focal events and whereas you know
breakage fusions bridge sorts of events
might lead to more focal copy number
alterations but not changes in whole
chromosome so there mechanisms that have
previously been identified but what
we're finding here is that these
mechanisms shape the cancer genome it's
a different extents that the mechanisms
that lead to arm level events are much
more frequent than the mechanisms led me
to focal events for events of this size
so from these sorts of
yep huge that's a great point right so
everybody in the cancer everybody just
about everybody in the cancer field it's
focused on the focal events right
because you know their their focal you
can kind of interpret them you know they
some of them might affect 1g and so you
go ah that's the gene but it turns out
that you know 70 80 percent of the
alteration altered genome is due to
these arm level events the most frequent
genetic events in cancer are arm level
copy number events amplifications and
deletions of whole chromosomes or
chromosome arms those are incredibly
complex beasts that affect
simultaneously up to thousand or more
2,000 genes and so you know why they're
what they're doing precisely is not well
understood and I mean one approach to
understanding that may be to look at
correlations and analyses and anti
correlations analyses like been Rafael
is describing and I'll describe some
work we've done to but I mean you know
approaches to understanding those are
really important and not it and I think
we need a lot more work there so so this
describes you know something about the
mechanism by which copy number events
arising from this then we can look back
at the of the view of the data where we
saw these stripes going across the data
again this is washed out to apologize
for that but these so now we can
understand why we're seeing these
stripes
it's a mechanistic cause that leads to
an enrichment of whole you know
amplifications of all of 1q are all of
chromosome seven or all of 8q and then
deletions of like all of chromosome ten
and so on so it's a so mechanism is
leading this so that is so as we
determine is we need to these sorts of
analyses and understand more and more
about the mechanisms shaping the cancer
genome and that their role in shaping
the cancer genome then we can start to
go beyond them and try to understand the
Selective pressures shaping the genome
and so what are the driver genes and now
then describe this the the distinction
between drivers and passengers but I'm
going to reiterate it that not every
mutation and cancer drives cancer and
this is something that I think was not
really well understood
you know just ten years ago and now is a
fundamental issue in understanding
cancer genomes and as to say that how do
cancers arise well cancers arise because
as cells in your body divide they
acquire mutations at random throughout
the genome right and most of those
mutations are going to be neutral or
diety rias to the cell and not lead to
cancer but occasionally one cell may
undergo a series of events that
transforms it into cancer and that's
indicated by the lightning bolt then
that cell is going to divide and all of
its daughter cells you know that it's
going to come cancer in all its daughter
cells which will share those driver
mutations along with a passenger the
neutral or deleterious passenger
mutations that preceded it so how do you
distinguish between those two is a major
challenge the way the approach that we
take is that to do apply statistics from
a single sample you can't distinguish
between drivers and passengers but you
every cancer requires driver events so
the driver events will tend to recur
whereas the passenger events just are
occurring at random so you'll only
observe them at the background rate so
we a model that using gistic an
algorithm that we developed for come to
interpret copy number changes in cancer
where we profile each tumor for you know
the events that occurred and then we sum
across for each region of the genome we
sum across all tumors to generate a
score representing the combined
amplitude and frequency combined
frequency and amplitude of all the
amplifications or deletions in that
region then we ask well what would that
look like with this profile look like if
all the events were passenger events
were the passenger events you know in
theory could occur anywhere in the
genome with almost equal likelihood so
we simulate that by REO muting the
locations of each of these events around
the genome at random and regenerating
these gee score profiles and that sort
of gives us a distribution of gee scores
that you'd observed by chance it allows
us to assign a significance threshold
and apply that to the
data to identify regions that are
significantly aberrant likely to contain
genes that you know drive cancer so this
is what we get when you apply it to
5,000 cancer samples we see across the
genome on the lung the y-axis we could
see the significance along the x axis
for deletions on the left and
amplifications on the right and you know
what we see is that there are these
needle like production projections where
like very small regions of the genome
can get recurrently amplified or deleted
and as you go to the tips of these
projections they really focus in on just
a few genes so so and those few genes
are the ones that are most recurrently
amplified or deleted and most likely to
be the oncogenes and tumor suppressor
genes that account for these events so
we right we were able to deconstruct
these graphs in 270 independently
significant deletions 70 independently
significant amplifications and with a
small number of genes per peak region on
average because and the small number of
genes is up in part because we're able
to analyze such big datasets nowadays
what are those genes that were observing
well first of all of the 140 regions
only 38 contain known amplified
oncogenes are known tumor suppressor
genes and that includes most of the
known amplified oncogenes many of the
known tumor suppressors and but what it
leaves the majority of the regions
unexplained and so it to me what that
implies is that despite the profusion of
genetic data that we've obtained from
cam cancers over the last few years much
of the cancer genome remains to be
explained when we look at the most
significant regions of amplification we
see a few things these are the top 20
and 20 peak regions first of all most of
the these regions have only one or two
genes in them so we're quite specific in
identifying the genes that we think are
important in cancer the other thing we
noticed is that most of these regions
contain known uncle genes and in fact
contained most of the known amplified
oncogene cyclin do you want EGFR Mik
Irby - second t1 all these are so
cardinal oncogenes in cancer so so we're
also quite sensitive so we're both
specific and sensitive and that
indicates to me that for the regions
we don't know the target what the
oncogene is we have a good chance of
finding it right so for instance here
the regions red don't contain known
known oncogenes this region contains two
genes
one of those is Turk Turk is the
substrate for tert which is the the
Nonaka gene that's the single gene in
this this amplification peak here and so
to my mind we don't you know we know a
lot about Turk biology tert is
telomerase it allows for extension of
the telomeres to make cells immortal we
know a lot about its biology no one has
shown that Turk its substrate is an
oncogene but serve I mind this suggests
that we should look at that there are
other regions so many of these regions
contain epigenetic regulators WHS c1l
one cat6a KD m5a all regular regulate
you know chromatin structure and you
know epigenetics and and we're finding
that among the regions without known
oncogenes epigenetic regulators are
enriched and so again that suggests some
interesting targets to pursue as but as
potential oncogenes yep
so we have an so you know I described
logistic algorithm in one slide which is
a shame because it's like 10,000 lines
of code so the you know there are a lot
of additional you know functions or
routines within that algorithm so one of
those is to say if you have a peak of a
certain shape what are the boundaries of
that peak where where you'd expect that
that you know you would have a say 95
percent confidence that the gene lies
within those boundaries and so we set we
you can set whatever if confidence
special do you want 99 percent 95
percent 50 percent and so we set those
to some level and then say well what are
the you know how many genes are within
that interval and so this is using a 95%
confidence interval okay so you know so
I think there are a lot of interesting
results here that should be followed up
we don't have the power to do that the
ability to do that so we put all the
results up on the web at www.cpa.psu.edu
or to mention if I have time a couple
minutes left some other things were
noting about these the results one is
that I described that amplifications
analysis which i think is quite good the
deletions analysis has some weird things
going on for one thing weird thing is
that the large deletions tend to contain
the largest genes in the genome so among
the top 20 largest genes in the genome
measured by the amount of DNA they
consume twelve are among the most
significantly deleted regions of the
genome and that's much greater than what
you expect from random chance you'd
expect that one gene would be about you
know 5% of all genes are in these top 20
yeah and in our deletion Peaks so
that's a tremendous enrichment it's not
an enrichment that's observed for
amplifications and it you see decreasing
these continue to see enrichment for
slightly smaller genes all those
services decreasing levels so what's the
reason why large genes tend to be
deleted well the three possible
explanations first is that large genes
may be tumor suppressors preferentially
tumor suppressors that's possible and I
think in some cases it may be true but
for for twelve out of the top 20 largest
genes seems unlikely you know some of
those genes are the Duchenne's muscular
dystrophy gene for instance I don't know
of any any data that say that people
with muscular dystrophy have higher
rates of cancer so uh you know not clear
that this is correct
it's possible that large genes tend to
lie in fragile sites which are more
prone to breakage for mechanistic
reasons and therefore more often deleted
than the genome wide average I think
that is true and other folks have argued
for it and I think that's the case for
many large genes a third possibility is
that large genes are just not essential
that if you have a deletion that affects
a large gene it has it's less deal it
serious to the cell than deletions in
other regions and so therefore we see a
buildup of deletions in regions
containing large genes relative to other
regions in the genome and that's
something that I think we have some
evidence for so for instance large genes
tend to lie in gene deserts where the
local so it says if they've pushed away
the other genes in the genome and so the
local gene density around the 20 largest
genes in the genome is about a third of
the gene density genome wide the average
genome genomes of the genome wide for
slightly smaller genes it's the gene
density is not quite as low but it's
still much lower than the genome wide
average and so a deletion that affects a
large gene will affect fewer genes
overall then then a deletion in some
other part of the genome and in fact
when we look across that large database
we have of you know hundreds of
thousands of copy number events we see
that no matter where you go in the
genome large so the deletions tend to
avoid genes so for instance homozygous
deletions affect 30% fewer genes than
you'd expect
you know if they were randomly
distributed around the
and that's not true for amplifications
amplification stand don't tend to avoid
genes to the same extent so what you
know what's the reason for that well
it's pretty seems to me to be pretty
clear what a potential reason is is that
if you have a deletion that affects a
bunch of genes it's more likely to
affect an essential gene one that's
necessary for the cell to survive and
therefore it's a it's going to kill the
cell so we all tend not to see those
deletions in cancers and and we can see
that actually on an individual level so
we're now identifying where the
essential genes are in the genome in
cancer genome and this is this was from
a paper that we published last year with
bill Han where he had generated RNA
interference data from 86 cell lines
where they'd knocked down 11,000 genes
in each of these cell lines and so we
could then ask and then ask you know how
viable were the cell lines after
knockdown of each of those genes and so
we could ask well how does that
viability correlate with copy number and
specifically if the gene is hemizygous
Lee lost do we see cases in which that
hemizygous loss predisposes the cell to
be more vulnerable to further
suppression the idea is that for
essential genes if you have cells that
have hemizygous loss that maybe they're
closer to the edge closer to cliff so
that if you just knock down that
essential gene a little bit more you'll
kill that cell whereas that same level
of knockdown in a normal cell with two
copies would not affect it this is a way
of trying to identify a new universe of
therapeutic targets in cancer
non-drivers that might be nevertheless
therapeutic targets we found many genes
and followed one up but what we also
found was that most of those genes are
in fact essential genes as we'd expect
and we could see where they were and so
an example of that is SMU one here on
nine p9p
it contains p 16 the most frequently
deleted gene homozygous ly deleted gene
in cancer and it's and those homeles I
get so this is a view of multiple
different cancers each is a column and
again deletions how homozygous deletions
are shown in dark blue hemizygous
deletions are shown in light blue and
the homozygous deletions often lead
right up to SMU one but never encompass
it that would be
because s mu1 as we were finding in our
screen is an essential gene so any
deletion that covered s mu 1 would kill
the cell but what that implies is that
the universe of deletions available to
account on a cancer cell that would take
out P 16 a tumor suppressor gene but
leave s mu you know but leave the cell
viable is limited by the placement of s
mu 1 here so that if s mu 1 were placed
closer to the P 16 fewer deletions would
be available to cancer cells to to
enable them to become cancer so you'd
see so if s mu 1 were closer to P 16 you
observe smaller deletions around P 16
you'd reserve fewer deletions of P 16
and you'd observe fewer cancers right so
what that suggests is that the frequency
with which a tumor suppressor gene is
deleted it's shaped not only is
determined not only by the potency of
that gene as a tumor suppressor but also
by its environment by where essential
genes are in its neighborhood and so to
understand positive selection where the
driver genes are we need to understand
negative selection as well and that's an
area in which we're trying to develop
our algorithms to simultaneously
determine what the the the forces of
negative and positive selection shaping
the genome so I have what a couple more
minutes I'm gonna that I'm gonna skip
the last bits which would look at
correlations analyses and go straight to
the acknowledgment where I'd like to
thank first of all
Travis stat Zack Steve Schumacher
Matthew Madison Craig mourmelon GAD Getz
who did most of the work leading these
pan-cancer projects as well as many
people in my lab and other collaborating
labs thank you
a couple of questions so that's that's a
great question that's a great question
so what makes a gene large but I was
talking about the size of the gene in
the DNA right so that's mostly
determined by the size of the introns
not of the exons right the larger and
intron having very large introns there's
some data to suggest that that one way
in which a cell regulates the expression
level right so that the lower that
larger the introns the lower the
expression level so that would argue
that those genes I've you know are less
essential and more and often those genes
I mentioned the DMD gene Duchenne's
muscular dystrophy gene which is
obviously involved in you know muscle
and you know which is the nervous
essentially neuromuscular system many of
these genes are involved in neural
development so again not clear that
they're that important in a lot you know
colon cancer yes
don't know the direction of all these
correlations off the top of admin so so
so that that that explanation is that
replicate lately that replication time
predisposes you to having well being
make it makes it more fragile site
because you have less time to fix any
errors that are gained and so if you
have a breakage you don't fix it and
then it's you get a deletion there right
so it's a mechanistic explanation so
it's saying that there are fragile sites
and large genes I think that's you know
absolutely a reasonable you know
hypothesis and something that I think
works I think both of those explanations
essentially like lack of essentiality of
the large genes and the fragility are
two you know explanations that account
for in in tandem account for you know
the phenomenon we're observing you know
and we're working along those lines to
try to understand as you saw the
mechanistic explanations as well to you
know control for those in determining
what's going you know what are the
Selective pressures and you know taking
a replication time into account and so
forth we're still not there yet in terms
of really having a great that the ideal
biological model thank you very much
okay so before we convene I want to also
say not only all the speakers but
everybody who made a posters out of I so
a few of them I really liked them and we
should also talk so not only Tony but
Jessica who's standing there who really
helped us with all the logistics I don't
think I mean Tony is great but I think
without her this would have been a more
chaotic icing so let's all thank Jessica
okay and so also thanks all of you for
coming we hope to see you back in we
definitely will have another copy event
I think it probably will be in two years
at some point we'll need a different
building because this is growing fast
and for those of you who want to see
sort of the talks in more detail there
will be videos posted in a week or two
on our cabbie website okay thank you</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>