<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Message Passing: Modeling the Flow of Information in Cellular Networks | Coder Coacher - Coaching Coders</title><meta content="Message Passing: Modeling the Flow of Information in Cellular Networks - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Microsoft-Research/">Microsoft Research</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Message Passing: Modeling the Flow of Information in Cellular Networks</b></h2><h5 class="post__date">2016-07-07</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/RSIABsQAaYk" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">each year microsoft research helps
hundreds of influential speakers from
around the world including leading
scientists renowned experts in
technology book authors and leading
academics and makes videos of these
lectures freely available
you
start with the last session of the day
we have John Quackenbush from the
dana-farber cancer institute in harvard
school of public health who is going to
tell us about message passed a chick
passing well thank you for the
invitation thank you very much I'm going
to commit the intellectual equivalent of
the dine and dash and run out but I
started off adding slides after Jill's
talk and then as everybody else taught
me today we're finding explained a lot
of the underlying biology questions I
chill them back so what I'm going to do
is really focus on what I crunching sure
you're like ah please oh I'm always so
loud it never seems to matter there it's
green all right the light is green the
trap is clean name the movie all right
so I'm going to focus actually the
things I told you I was originally going
to focus on which is this idea of
message passing is a model but I have to
give you some background for them as
well I build models and I like quotes so
this is one of my fav favorite quotes
this is from Sam Carlin the purpose of
models is not to fit the question not to
fit the data but to sharpen the
questions and so I'm going to describe
models is away and I'm going to talk
about this later in which we look at
biological systems and try to understand
the underlying principles that we can
elucidate about those systems given the
data we have but I struggle with
answering your question which I can tell
you I cannot answer which is are the
models right and I'll come back to that
a little bit but this is a really
exciting time to do science and I like
quotes I'd like to quote the great
thinker so this is something you should
all put in your slides every revolution
in science from the Copernican
heliocentric model to the rise of
statistical quantum mechanics from
Darwin's theory of evolution and natural
selection to the theory of the gene has
been driven by one and only one thing
access to data
I totally like to think quote great
thinkers you should as well this is
probably the most profound thinker I've
come across but this I really think this
captures a lot of why I'm so excited
about the work that I do that we have
access to just unbelievable quantities
of data and this is really the future so
we tend to think about in my group the
whole history the life history of
disease and how we look at disease and
how we place the disease into context
given the data we have so we have these
technologies that have really been
spawned by the genome project that allow
us to measure genetic background in
estimate genetic risk to do better early
detection by looking for things like
circulating tumor cells or circulating
DNA from disease states but once we do
that we want to stratify patients stage
disease and really use this kind of
information to select the right
treatment options I think as Zack
pointed out we have all these things we
try to estimate like lifestyle and
environment that we really don't have
good data on but at the end of the day
what we end up doing is trying to use
all the data we have to really select
the best treatment options for our
patients and to improve their quality of
life and that sort of underlying
thinking drives a lot of what I and my
group have been doing over the years I
mean one of the things always like to
point out is our patients want exactly
the same thing we want we're all going
to come to the end of our lives we want
it to be as far away and as pleasant as
possible and so in order to be able to
help really inform this process we've
had access to tremendous quantities of
data a lot of this has been with arrays
but increasingly we're generating data
using these sequencing instruments and
what it's done is it increased the
volume velocity variety to use the Big
Data terms of the data we can generate
because it allows us to generate
integrated data sets with multiple
independent types of genomic data from
the same individuals and that really
opens up new interesting opportunities
to do analysis particularly as Zack
pointed out if we can place in the
context of the biological and clinical
data that we have so how do you turn
this day this vision into a reality well
there are lots of things that you have
to do and if I were giving a different
talk I talk about each one of these
because they're essential components you
have to assure access to samples and
consent you have to have a technology
platform for generating data
you need to think about integrating
information and this is really critical
that if you have disparate stores of
information you can't bring them
together you can't link them you can't
associate them there's almost nothing
you can do with the data as you do this
you have to think about communicating
information back to your constituents
and colleagues you have to think about
enabling research beyond what you can do
because people come to my door every day
and knock and ask for help analyzing
data so you need to come up with
strategies to support and enable that
you have to engage corporate partners
and this is something that in academia
sometimes is a bitter pill to swallow
because we think we can do everything
and we can it's just not as good as well
as other people can sometimes you have
to communicate your mission back to the
broader community and a lot of what we
try to do is actually do public outreach
and then you want to do research and I'm
going to really focus on this last piece
but being able to do research demands
that we do all of these other things in
some kind of sensible way so genomics
has produced tremendous quantities of
data has it transformed medicine while
assembling a reference genome has not in
and of itself transform medicine but
really these technologies have spawned
vast quantities of information and
really allowed us to generate data that
it itself the data itself is really
transforming biomedical research by
allowing us to ask new questions so this
is the era of big data and the real
challenge is to bring all this
information together to understand
things like disease so how do we do this
well the answer in my mind is to look at
networks and I'm going to tell you about
an approach have been using to look at
networks by starting with a very simple
question what are the subtypes in
disease and this is actually a question
you've heard alluded to a number of
different times today everyone tries to
do this everyone tries to build
predictive biomarkers most predictive
biomarkers fail many predictive
biomarkers do no better than random sets
of genes trained on the same data so
what's the reality well the problem with
disease is that we often don't really
understand what the subtypes are they're
difficult to recognize we have diverse
data that sometimes tells us different
things and again if I were giving a
different talk i could tell you about
how
we approach the question of robust
biomarkers and really looking at those
biomarkers but what I want to do here is
just focus on a simple observation so
we've been studying ovarian cancer for a
number of years it's the fourth most
common cancer in women in the US it's
actually has one of the greatest death
rates because women tend to be detected
or the disease tends to be detected late
when it's detected women are treated
with a standard regimen of chemotherapy
independent of what the subtype of the
diseases that they have because we
really don't understand the disease that
well and we really don't understand the
molecular profiles for these different
subtypes and so most women develop
chemotherapy resistant disease and
succumb to it we were actually involved
in a project in which we were able to
take 132 women profile them with micro
RNAs and mrna s and get gene expression
data on those patients Lincoln to
clinical data and our first goal was
actually to find genes that correlated
with resistance to chemotherapy and as
we tried to do that analysis we failed
miserably because what we discovered was
that the clinical definition of
resistance or sensitivity to
chemotherapy was a mean time of
progression of greater than or less than
six months which means that you can move
from resistance to sensitive simply by
having your appointment rescheduled
which means it's completely meaningless
definition and that's often what you
find when you look at clinical data the
clinical definitions have little to do
with the underlying molecular biology so
we started to look at the data in more
detail and we asked ourselves what are
their particular subtypes can we
identify subtypes in the data and so we
used a technique this has really worked
the stuff on benthic didn't working with
me I use the technique called Isis and I
forget what Isis is an acronym for but
he used Isis to really look at split to
the day to explain what Isis does and we
found a subtype associated with the
expression of angiogenesis genes we
curated all the publicly available data
which in it of itself is not easy to do
and we validated this separation so
here's the basic idea behind Isis what
we do is we take measurements in this
case on 132 patients for those patients
we have measurements on 12,500 genes
that were on this array so we have this
very
large problems we're looking for
correlations between profiles gene
expression profiles and subsets of the
data and anybody who's worked at big
data sets can tell you that if you take
a data set and randomly divide it you
can always find some feature in the data
set that distinguishes between the two
subtypes okay so what we do is we do
that but we recognize that something
approaching a real separation and the
data is probably not going to be
supported by just one or two
statistically significant differences
but by many and that on average they're
going to be more significant differences
than you'd see for a random partition
that has no basis and biology so you
take this and you divided many many many
times you look for a good partition in
fact you use Fisher's rule for combining
p-values to look at the top 100 genes
and then eventually you get to some
point where you're doing something like
k-means clustering and shuffling things
back and forth define really robust
subtypes so that's basically what we did
and we found this subtype now I
apologize the colors have changed from
blue and yellow to red and blue but if
you look at this what you can see is
there's one group clearly over here and
the rest of the group there about one
third of the patients about two-thirds
of the patients and these are patients
who exhibit an angiogenic subtype that
the separation is really driven by genes
associate with angiogenesis they're
statistically significant differences in
survival in our data set and in fact in
the two years this paper was in review
more and more publicly available data
was accessible and so what we're able to
do is really show that this difference
in survival persisted as the amount of
data group all right so it's a very
robust subtype it's clinically
significant because its associated with
angiogenesis and there are actually
clinical trials underway in which
patients are being treated by Angie with
angiogenic inhibitors in addition to
standard chemotherapy the numerology
behind it is that about one-third of the
patients on the treatment arm respond
okay and we would say that one third of
the patients which should be likely to
respond but we don't have any direct
data yet to tell us whether that
one-third are really the one-third who
are responding because we haven't had
data on
in these trials although that's changing
nevertheless about one third of the
patients have this angiogenic subtype so
we publish this I think in 2009 or 2010
and it describes these subtypes and
there's actually software for doing
classification to assign patients at the
different subtypes but how do you
actually use these subtypes as a way to
understand disease in the process
driving disease well we've been working
on building network models for quite
some time and what we really want to try
to do is use information about these
networks to understand the basis for
disease now why are we interested in
networks well you know biological
processes are not driven by individual
genes although we tend to think of them
as being associated with particular
genes and in fact everybody you talked
about disease that they talked about
genes because they're tangible elements
that we can discuss but we really want
to understand what's driving these will
understand the causal relationships we
want to understand the processes that
are happening and even in the earliest
papers describing gene expression
analysis had the implicit assumption
that if we look for correlated patterns
of gene expression those were probably
tied to underlying networks or processes
that were somewhere in the data so what
we want to do is we want to find
networks using available genomic data
which is largely expression data so when
we talk about networks what are we
actually describing well I can begin by
telling you what we're not talking about
we're not talking about metabolic
pathways like those in the keg database
we're not talking about signal
transduction pathways like those in bio
Carta we're not talking about
biochemical pathways like those in the
berenguer ingelheim wall charts we use
to terrify our graduate students where
we're really talking about something a
little bit more abstract so what we mean
is that genes are some kind of nodes in
the network those nodes are connected by
edges the edges are can be directed to
imply some kind of causal interaction
and what we want to do is try to infer
those edges recognizing that they may
not be direct interactions that we may
be missing things because we're not
measuring them that hidden behind this
edge what may be happening is there
proteins in G
and metabolites and all sorts of things
playing some role but what we're doing
is we're looking at the data we're using
the data to infer the state of the
purple gene given our measurements on
the state of the blue jean so it's an
inferential Network driven by data that
we have available that tries to capture
the association's that are really hidden
in the datasets we have and my thinking
about how to build these models is
really driven by my background I'm not
trained in anything reasonable for doing
what I do today I actually have a PhD in
theoretical physics but in physics we
recognize there are two strong pillars
supporting scientific inquiry theory and
experiment but we also recognize that
somewhere in the middle is a field that
we call phenomenology so ultimately what
we'd like to do is develop a theory that
describes the systems we're studying the
embodiment of that theory should be a
model but these systems were studying
are extraordinarily complex so we
recognize that there's this middle
ground between theory and experiment
that we call phenomenology and in
phenomenology what we try to do is build
empirical models that are consistent
with our theoretical understanding but
aren't directly derived from them okay
so we're going to try to capture the
essence of what we know without
necessarily fully modeling the system in
part because we know the data that we
have are incomplete so how can we
construct a model that captures the
essence of the system in a way that
needs us to a better understanding of
what's happening so as we do this we
recognize the question no longer is is
this model right but rather going back
to Sam Carlin's quote at the beginning
is this model useful can I actually
learn something that's going to inform
my understanding of the system and maybe
inform future experiments so how do we
do this well one of the fundamental
principles that's really guided my whole
group in doing research over the last
few years has been the idea that we
shouldn't have to derive these models de
novo from the data and I've gotten a lot
of grief from referees about this they
say oh you know you should be able to
learn this from fundamental principles
and that idea is just crap okay if you
know something use it it was actually a
group of
scientist at Microsoft who really
underscore the importance of this uh
this is a beautiful paper published in
nineteen ninety-six if you haven't read
it read it it's by Wohlford McCrady's
about something called no free lunch
theorem they were looking at this
problem of ensemble classifiers and
finding weights and ensemble classifiers
and what they demonstrated is there's no
general-purpose optimization algorithm
that works best in all possible
situations the solution to that is to
make a reasonable guess and what they
showed in this paper very eloquently and
very elegantly is that if you make a
reasonable guess you rapidly arrive at a
near optimal solution okay so if you
don't know how to solve a problem and
it's a computationally intractable
problem make a guess we actually first
use this in a paper that it was
publishing I think 2008 on bayesian
networks and describes seeded Bayesian
networks we since abandoned Bayesian
networks for a whole host of different
reasons including the fact that they
don't capture feedback in a very simple
way but we've used this principle of
making a reasonable guess is really one
of the driving principles in our
construction of models so what are we
trying to do we try to look at this
network or any network and we're trying
to understand how that network model
changes and what we're really seeing is
that as we look at normal to disease
transitions the network's fundamentally
rewire or as we look at different
subtypes of disease the network's
fundamentally change and part of the
reason we can't answer the question is
the model right is that when most people
are asking that they're asking those the
model you produce look like a picture in
my textbook which may be right for
normal tissue but isn't going to be
right for disease states we don't
necessarily know what those networks
look like and I'm going to come back to
that in just a few minutes but we want
to understand how these networks change
how they differ between states so what
we're going to do is model a very simple
very well understood process you've
heard alluded to a number of different
times today and that's transcription
factors in their activities in
activating genes ok so the model itself
is very simple we have a gene that
encodes are an a the central dogma of
molecular biology is DNA to RNA to
protein so this is really storing
information that's going to be turned
into an RNA that's eventually going to
make a protein and a cell is basically
machine made of proteins so you're
making the fundamental building blocks
for yourself all right in order to do
that you need to have an RNA polymerase
bind and make the RNA and typically it's
activated by the presence of
transcription factors that bind and turn
on the process and make it go all right
so how do you think about that or how do
we think about it that well the way we
think about it is using a very simple
idea the idea of message passing all
right so communication involves in this
case a transcription factor and its
downstream target in like in any
communication there are always two
active participants so at the end of my
presentation I'm going to give you a
quiz right and we're going to see how
well you do all right now if I'm having
a conversation with Francisco who
introduced me if none of you hear our
conversation and I give her the quiz if
she does really well then we know that
she did her job and listened and I did
my job and told her about it but if she
fails miserably right then they're
either one of two things had happened
she was busy sending emails and not
paying attention or I was talking about
you know my plans for the evening and
why I'm not going to be here okay so if
she doesn't do well you don't know where
communications is broken down but if I
give you all the quiz and you all get
about 90% except for one person then we
know where the problem is right so if
communication breaks down and we have a
big enough network we can actually
understand where the communication
channels break down so the language of
message passing the transcription factor
would be responsible for communicating
with this target the target has to be
available to respond and so if we have a
large enough network we can actually
capture those functions for each edge in
the network so what we did was he
developed a method we call panda because
all methods have to have cute names and
this is really the work of Kimberly
glass and we use prior information we
use the fact that we have the whole
genome sequence and we know based on the
work that Martha and other people have
done that transcription factors bind to
unique motifs in the DNA and we can map
those to the genome so what we can do is
we can create an initial Network
basically representing all possible
interactions then we can take that
network and use available data for each
edge in that network to estimate these
functions and we can it early estimate
the functions and trim the network until
we converge to a network structure that
makes sense that's stable given the data
that we have right so we downloaded data
on ovarian cancer which is why I was
telling you about this analysis we did
we actually took the biggest publicly
available data set from the the TCGA the
Cancer Genome Atlas and we normalize the
data we assigned samples to angiogenic
or non angiogenic subtypes and then we
ran this message passing analysis so the
approach was actually very simple and
very straightforward what we did is we
took the two independent data sets I
apologize is a little hard to see but
angiogenic is in red non angiogenic is
in blue we constructed networks given
the data and then we compared them and
this is the place where as we start to
build network models I've really become
interested in how we build these models
and how we use them because what we look
at in these networks are differences in
the edges and what we've come to
recognize is the atoms of the networks
are not the genes but the edges and I
can say that and I believe it and we
talk about all the time but when you
start to talk about the edges it becomes
very difficult because they're not
tangible things that you can hold on to
so when we start talking about edges
will go back to the genes that are
activated by those edges but remember
the key difference between one network
and the other are the edges the
communication channels how those genes
are being activated so we can look at
the edge weight so we can find those
edge weights they're different and we
can ask what genes they target and
what's really interesting is that we
look at these what we see is there are
unique genes that are targeted but
they're also genes that are targeted in
both subtypes but by different
mechanisms
okay the edges are different and that's
important so if we look at the
transcription factors that are actually
the most active what we find are there
ten that have the biggest switch from
one site subtype to the other and if you
look at these every single one of the
transcription factors is associated with
angiogenesis now it's really easy for me
to say this but in my group this is what
we call bio poetry because I could
probably take these ten transcription
factors and show their associated with
autism or Alzheimer's or almost anything
that you pick okay so are they really
useful are they interesting well the
answer is yes if we look at the
transcription factors and their targets
what we find is that the targets are
significantly differentially expressed
while the transcription factors aren't
necessarily differentially expressed if
we look at DNA methylation data what's
also very interesting is methylation
which you heard can regulate gene
expression these jeans are actually
differentially regulated in a way that
corresponds with the patterns of
expression so we can start to map out
what these are doing and whenever you
show do it give a talk like this and
talk about networks you have to show
these so these are the angiogenic edges
these are the non angiogenic edges
remember them they're going to be on the
quiz they're very difficult to try to
interpret but what Kimberly class did
was she built this representation and I
apologize the transcription factor names
can't be right around the center I call
it the spirograph representation but
what it's showing are the the
transcription factors which are more
active in one subtype or the other and
the genes that they're regulating or
controlling and what you'll start to see
is there's some that are controlled by
single transcription factors but there
are also complex patterns of regulation
it's those complex patterns of
regulation that are really interesting
so you can look at the genes that are
involved we characterize the different
genes depending on whether they're
targeted in angiogenic or non angiogenic
subtype whether they're up regulated
positive or downregulated negative or
whether they have complex patterns of
regulation they're targeting both and
they have complex patterns across the
subtypes but what's probably more
interesting is the
acting when we look at these we see an
association between these patterns of
regulation and now biological processes
associated with angiogenesis but more
interesting is the fact that when we
look at them and we look at the genes
that are being targeted by multiple
transcription factors there are sets of
transcription factors that occur far
more often than we'd expect by chance
all right so this is really interesting
because what it's saying is it's not
just different patterns of regulation
between the two subtypes but different
patterns of combinatorial regulation
between the subtypes so when you start
to look at this what you really see is
that there are opportunities to step in
and interfere with one subtype or
another not by acting directly on the
genes but instead by treating with
compounds that actually prevent the
dimerization of some of these
transcription factors and their
subsequent activation of gene expression
patterns okay so by looking at the edges
what we see is something very very
different and what we actually try to do
is look at different drug targets drugs
that target these interactions okay we
have some supporting evidence my boss is
standing up time you have to be quiet so
we have some supporting evidence that
that looks at this not necessarily in
ovarian cancer but in a breast cancer
model and the interesting thing that we
see is treatment with angiogenesis
inhibitors in this breast cancer model
produces in these different gene subsets
that we defined exactly the pattern of
expression that we would have predicted
okay so something's going on here we've
applied this in other situations we've
looked at asthma and treatment of
children with asthma in response to
corticosteroid drugs and see very
different responses in terms of the
edges that are activated or deactivated
we've applied in different situations
and in fact Kimberly just won an award
for applying this to Alzheimer's disease
and discovering that women we know that
women have twice the incidence of
Alzheimer's even correcting for the fact
that they live longer
and when we compared the brains of males
and females and looked at how their
patterns of expression change before and
after Alzheimer's what we discovered
using this approach in analyzing data
sets has been available for years is
something nobody ever found but as soon
as we found it was intuitively obvious
that the differences between male and
female brains relevant to Alzheimer's is
in the activation of genes that have
estrogen or androgen responsive elements
okay men and women swim in different
hormonal oceans they respond differently
it influences the development
progression of disease so there's lots
more that we can do with this and I
think we're starting to really see that
by looking at these channels of
information flow we're really starting
to discover something that's important
for understanding the mechanisms that
are driving disease so we're really
excited about panda because we can start
to integrate lots of other types of data
and if you ask me about it i'll tell you
all about it we have other approaches
which i'm not going to talk about and
I'll just close by telling you genomics
is here to stay 2010 I went to Australia
to do a little mini sabbatical I flew
down there with my wife and son my host
pick me up at the airport drove me to a
little cottage we'd rented for the three
months that in Brisbane they call winter
but we call summer but Brisbane is as
far south of the Equator is Orlando is
north of the Equator so it's really
beautiful there we landed we drove to
the house Christine Wells my host then
drove me to the car rental place I
picked up the car I drove back home if
your house works like mine I walked in
the door my wife looked at me she looked
at the keys and I said goodbye to the
keys for the next three months and rode
the bus so my second day there I got on
the bus that I looked up and I saw this
sign it says spinning his unacceptable
bus operators are now equipped with DNA
kits to assist with the apprehension of
offenders
okay so genomics is here to stay I'll
finally just close with a quote I'm a
physicist you now know you now know
physicist are the smartest people on
earth so I love this quote from Enrico
Fermi before I came here I was confused
about this subject after listening to
your lecture I'm still confused but at a
higher level so hopefully I've left you
all highly confused but your questions
will illuminate anything you don't know
so thank you for your patience and I'm
happy to answer questions and we're
across the river so if you want to come
over and visit and talk more just draw
me it was a joke might be out of time
actually I don't know what the
difference is it's a significant
difference but it's not that great and
part of the reason is these women all do
badly the hope is by treating those with
the angiogenic subtype who have a poorer
outcome that we can actually improve
outcome for them please I don't know
we've never really looked at that we're
trying to come up with ways to use
Bayesian networks because they're very
useful but Bayesian networks have this
NP complete problem that is really
difficult to overcome it's hard when you
have to discretize data but we've talked
about multi-tiered Bayesian networks
trying to pass information back and
forth and that's in part what led us to
start thinking about this overall
approach the thing I like about it here
is its modeling what's really happening
what we know is happening these
biological networks that the
transcription factors are actually
communicating with their targets
no so we haven't been able to identify
those but it may be other factors other
than just a causative mutation it may be
much more complex so we haven't really
built in although we're starting to
think about how to do it genetic changes
into the underlying model so the first
thing we've actually shown is you can
use epigenetic data to influence the
models and I think it was dug earlier
was talking about centipede we've
actually benchmarked panda against
centipede on the data they used to
generate centipede so it's dnase
hypersensitivity data and motif data
nothing else and when we do that we
actually get a better AUC then they do
with centipede alone and part of the
reason that we we believe that's
happening is that this message passing
is really optimizing the channels of
information flow through the network so
independent of any gene expression data
we're finding something that really
points to something we can verify with
other types of data so we're really
excited about it because we think it's
getting at some essential element of
this that's probably evolutionarily
driven Thank You Chun</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>