<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Clasp: Common Lisp using LLVM and C++ for Designing Molecules | Coder Coacher - Coaching Coders</title><meta content="Clasp: Common Lisp using LLVM and C++ for Designing Molecules - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/GoogleTechTalks/">GoogleTechTalks</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Clasp: Common Lisp using LLVM and C++ for Designing Molecules</b></h2><h5 class="post__date">2015-09-11</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/0rSMt1pAlbE" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">hello it's my pleasure to welcome
about 20 years we got started and or I
got to know him in a grad program at
UCSF biophysics and in the computer
graphics room and Christian was in there
as well and I learned a lot and I was
always completely blown away because he
was the first person I met who had
designed a pro team from scratch which I
thought was pretty amazing because I
thought that that was a really really
really hard problem and he made it look
really easy by writing a few computer
programs spending a few thousand hours
characterizing it as well Chris
Christian went on to do other great
things and now he's going to talk about
a system he's built which I think it's
really interesting because it's a
combination of software engineering and
science and using the software
engineering to push the science together
together and I'm really excited about
this because I've been using Chris
Christian software for 20 years and in
fact I just used his software a few
years ago as well and so it's got
incredible longevity because I was using
a 20 year old program and it's the state
of the art so I don't know whether
that's a good thing or a bad thing but
he's gonna tell us about what he's built
more recently now thank you
thanks okay um-hmm so I'm gonna tell you
about the work that I've been doing over
the last oh well
fifteen years pretty much I just wanted
to start with a side of my group so this
is my group here all of these
fresh-faced young people there are
synthetic chemists they go into the lab
every day and they make molecules and
none of them do any software engineering
or anything I'm the only person who does
that and I do it in my copious spare
time I'm gonna tell you about what we've
been doing with molecules and the
software and how I'm trying to bring
this all together okay so what I want to
do is I want to build molecules the way
that I can write code I want to build
large complex molecular devices that can
fix every problem that we
that can build new materials break down
the old materials that we don't need
anymore like plastics can you know
convert carbon dioxide sunlight and
water into fuel basically fix everything
and the it's all possible we just have
to figure out how to build the right
kind of molecules and this idea was
basically first sort of articulated by a
richard fineman who gave this brilliant
lecture in 1959 may be called the
there's plenty of room at the bottom and
in this talk he basically proposed for
the first time the idea of doing
manufacturing atomically precise
manufacturing on the molecular scale and
that if you could build devices down
there then we could be to build devices
that are exactly the same from one to
the next and this is something that's
really inspired me and what's driven me
forward alright so there's already
examples of this stuff we're all made
out of it the main ones are called
proteins proteins are long chains of
amino acids they're like charm bracelets
and when you put them in water they fold
up into well-defined three-dimensional
shapes that do things like speed up
chemical reactions trillions of fold
they build materials they build our
muscles everything that we're made out
of we've built with these machines they
can do things like organize little
pockets antibodies are big molecules
that have these two bindings right there
they can pretty much recognize and bind
almost any other protein and so they
protect us from viruses we didn't have
these things we wouldn't last a few days
we would die
they make channels in membranes that
pass ions and water and sugars and all
sorts of things in and out of cells and
they are perfect filters there are
amazing capabilities that are waiting
for us to develop if we could build
things like this now I originally
thought the way to do this was to build
things like that to build proteins and I
spent four years building this thing
this is the protein that just talked
about it's 40
one is I spent four years putting this
thing together I wrote some software I
designed a gene i express the protein I
expressed purified it crystallized it
showed that it was well folded and the
thing is a molecular boat anchor it
doesn't do anything and the big problem
is I didn't know how to make another one
and I didn't know how to make ones that
are more complex and could do all those
wonderful things I showed you before so
I got very fed up with proteins at the
end of my graduate career this is
another problem I mean they're amazingly
amazing if amazing capabilities you take
some proteins you hold them at 38
degrees for 35 days and you can make
things like eagles and people and build
wonderful functional things they're also
very fragile you hold them at 70 degrees
for a few minutes and you've got
scrambled egg which is delicious but not
very functional
so using making proteins and using them
for things that people want to use them
for like sensors and catalysts and stuff
they're just very very fragile we need
better materials to build molecules that
do things with that's what I thought um
here's another problem there
computationally they're almost
intractable alright so this is a chain
of ten amino acids it's a 10 mer peptide
a short little section of a protein a
real protein is usually a hundred to a
thousand amino acids but just ten amino
acids like this strung together in a
little charm bracelet if you go through
the math and you work out how many
shapes can this molecule have because
every one of these lines is a bond
around which you have essentially three
rotation and if you just do some simple
arithmetic you ask well how many shapes
can this molecule have got very
conservatively it's about 2 to 2 times
10 to the 14th different shapes so this
is not this is really not just one
molecule this is 2 times 10 to the 14
different molecules because they're
every shape can do different things
potential and it's constantly flickering
and moving between them that's a problem
now modern processors are not very fast
I mean you're great for video games
right to keep you you can do amazing
graphics and stuff with that but
chemistry they're pre slow yeah about
ten to the ninth floating-point
operations per second on a modern CPU if
it takes you ten thousand flash to do
one generate one of those confirmations
you can take you about six years per
molecule on one CPU to generate every
conformation of just this little
molecule all right six years for one
molecule and then there's a question of
how many of them are there if you take
all the 20 amino acids and you put them
in into every position in every
permutation they're 10 to the 13th
different molecules that you would want
to search through in order to find one
that does something like you know mimic
insulin or bind human growth hormone
receptor these are huge numbers this
huge problem for computational design of
molecules so what I thought was well
folding is the big problem this idea the
molecule undergoing all these different
conformational shapes so what if we
eliminate it what if instead of making
building blocks that connect together
through a single bond we make building
blocks that are rings and we connect
them through pairs of bonds so we make
ladder molecules and then we can hang
our functional groups off of it
what would that give us them well for
one thing it massively reduces the
number of confirmations shapes that it
can adult you're going to like a few
thousand all right so 10 to the 10th
fewer shapes that it can adopt the
amount of time that it takes to build
every possible shape is now 0.06 seconds
all right so it's it's 3 billion times
less work to build all the shapes that
this molecule will Canada pretty much
it's a rod and the side chains those are
groups that are hanging off just they
flop around a little bit how many of
these 10 Murs are there if we use the
same kind of side chains that are in
proteins yeah can't blacks a million
more molecules that you can build
because all those little stars in there
they represent what are called stereo
centers it's basically one bit of
information and that defines the shape
of the groups attached to the carbon
that each of those stars is next to so
you can use this thing called
stereochemistry to define the shape of
the molecule and program
shape in that way so you still get a lot
of diversity you solve a big design
problem so this is what I've actually
come up with in the last 15 years and
we've been developing it to the point
that we're at now so we make these
building blocks they are cyclic building
blocks so like a little Lego bricks and
they connect together this building
block connects to a next one through a
pair of bonds at this these two atoms
here's what this looks like in real
chemistry building block is connected to
another building block through a pair of
bonds to get a ladder molecule and we
can hang side chains roofs off of it and
you get a molecule that has a
programmable shape depending on what
stereo centers you put in there what
rings you put in there where you put the
different linking groups the amino acids
on the backbone you've got a lot of
choices here a lot of design choices
basically what we have are these little
Lego bricks you snap together through
pairs of bonds we can put them together
in any sequence we can make these little
sticks with stuff hanging off of them
okay the kind of things that we can do
in this well we can put you know a
couple two three four or five of these
little sticks together like your fingers
and make molecules that display a lot of
surface area and could wrap around a
protein all right all the medicines you
know if you take like a blood pressure
medication or or aspirin or something
these are very very small molecules that
would plug into a greasy pocket inside
of that an enzyme and jam it that's what
pharmaceutical industry is based on the
biotech industry is based on new
therapies based on antibodies which are
these big molecules that actually wrap
around the protein so what we can do
here is use chemistry to make molecules
that work like antibodies do but are
just maybe 5% of the sides so they could
get into places where antibodies can't
they also would be indestructible they
wouldn't need a cold chain a 4-degree to
be kept at 4 degrees all the time in
order to maintain their viability you'd
have a lot less problems with life batch
to batch variation
antibodies and from Baxter batch you can
get different results okay but they can
bury just as much surface area when
buying a protein as an antibody does
this is the big thing I want to do I
want to make catalysts I want to make
molecules speed up chemical reactions so
I need to make big molecules that have
pockets inside of them that it can
organize multiple groups in order to
push and pull on the electrons and
change a molecule that goes into this
pocket and then another molecule come
out if we could develop this capability
we could solve every problem we have
aging right get all these cross links
that cause the cause our scheme to get
cross-linked and start wrinkling as we
get older well you can think about
designing small cattles that are you
know a few percent of the size of a
regular enzyme that could reverse those
cross legs they could get in and just
loosen everything up so your natural
enzymes can get in and start fixing
things there's all sorts of ideas you
can think about too if you could quickly
develop catalysts the way we write
software okay and what it's all about is
about organizing groups in
three-dimensional space with the right
constellation and finding a scaffold
behind it that can hold them in place
we could develop new ways of killing
bacteria so we can make these triangular
pores I could bury themselves in
bacterial membranes hold open a hole
cause a bacteria to bleed its guts out
I'm just throwing up some ideas of what
we can do with this kind of shape
control but sticky greasy groups into
the center of these triangular molecules
I'm proposing here we can make receptors
that could bind almost any small
molecule and be used to detect small
molecules like you know nerve agents for
instance or other agents we can also
start to build channels we can make
molecules I could selectively pass water
through them and make perfect sort of
water filtration membranes for reverse
osmosis you know for desalinization or
separate aliens from seawater if I could
you know pull gold out of seawater I
wouldn't need to write grant proposals
anymore so so those are just some ideas
let me tell you about what some of the
things that we've actually done with
these
fifteen years so this is a molecule here
up on the upper right that we designed
to bind a protein called mdm2
it presents these and it organizes them
just the right constellations to which
is a protein that's evolved in a lot of
cancers and we designed this molecule
using the old version of my software and
when we threw that the protein it's
stuck
fairly weakly but it did stick to the
protein subsequently we improved the
molecule we've been able to make a find
reasonably well and when we throw it at
cells by passive diffusion either
confocal microscope images themselves
and you can see the compound is all
throughout the cell in the get T min it
stabilizes the protein that we designed
it to bind the levels of that shoot up
about 30 fold and there's another
molecule that Roche was actually
evolving to the cancer agent it binds
the exact same pocket on the protein and
kicks our molecule out I think because
they're big and structured they do
things that small molecules don't do to
publish this a couple of years ago we
now been able to like changing the
groups that hang off of the sample and
also when we change the group will not
find the protein so you have to present
the right groups and present them in the
right constellation and then you get
this binding activity another thing
we've done is we've made catalyst we
make molecules that speed up chemical
reactions so this is the same kind of
staffing right all these building blocks
connected through pairs of bonds here
they're displaying a base and alcohol
urea and because they can arrange all of
these groups in the right constellation
you can throw in methanol and volatile
trifluoride
and this molecule will transform the
into a new product the methyl
trifluoroacetic there and what we can
see here is product the amount of
product of time you're getting it build
up a product until you use up all the
starting materials and if we make any
change in this molecule you change the
backbone geometry or wear how the side
chains are presented we lose activity so
we've got the ability to sort of design
and create a somewhat optimal molecule
for this chemical reaction this is what
the thing looks like when you model it
constellation so that when this the
reagent comes in here you can get
attacked by this oxygen and activated by
this carbon needle and this this alcohol
of your gets its proton pulled off by
this base here so it's really a device
it is organizing the groups in order to
carry out this specific function about
what I really want to do is I want to
build big molecules these things that
we've made over the last couple of years
have been just basically one of these
segments and we've developed a couple of
demonstrations of applications for those
but there's still small molecules
they're about $700 $800 which is you
know like a drug sized molecule and I
don't think they're any better than
anybody else's small molecules they're
just versatile what I want to do is I
want to string three four or five them
together to create molecules with
pockets and large surfaces then we can
really start to do new things because
organic chemists can't make complex
structure of molecules of this size
we're the only people who can do that
with this kind of approach gotta be
careful there but let's make in terms of
making asymmetric complex programmable
molecules I really think we've got the
best and only way outside of proteins to
do that so that's what I've been that's
where the sort of software comes in I've
got a design problem I want to build
these big structured molecules and
ultimately I want to connect
them together to create even larger
things like robots these things that's
gonna take a lot of work we're gonna
have to get really good at designing
these molecules and connecting them
together okay so I've got this molecular
Lego this instruction set for four
functional molecules I'm gonna put this
way here and there I think they can do
all the sorts of things that proteins
can do but and that we can synthesize a
lot of them but I have a design problem
I have to pick out of this enormous
universe of molecules that I can make
the ones that could actually do
something useful okay so to do this I
basically need an Oracle I need
something where I can say I have a
certain requirements for a functional
molecule I want to organize these four
functional groups in just the right
constellation or around a pocket and I
want this Oracle to tell me what's the
sequence of the molecules that we can
then assemble in the lab that could
organize those groups that way that's
what the problem comes down to a
three-dimensional arrangement of groups
held in space what's the sequence of one
of our molecules that could hold those
groups in space so I've been writing
software that could let me do this okay
now the question is how do we design
molecules with computers there's a
couple of ways you could do it one idea
would be to just take some subset of lis
sort of depended undred and 60th
medium-sized molecules that Hemi
informatics people tell us you can build
they're possible take some subset of
those and then just build
three-dimensional structures for them
and ran them through a series of
computational filters and come out at
the bottom with a few molecules that can
satisfy your requirements and then go
and try and convince a chemist to make
that molecule
spend a year or two of their lives we do
this for small molecules that's how all
the things I showed you so far we're
sort of design one advantage that we
have though is that our library of
molecules that we start with can all be
made with exactly the same synthesis all
right so we make molecules and then so
we only design molecules based on the
steffel that we can synthesize in the
lab a more powerful way of searching
through sequence and structure or
conformational functional space is to
use Monte Carlo where basically you
picking a random starting structure and
then you mutate it and you score it
based on how well it satisfies our
criteria your mutation makes things
worse you roll a dice based on a
temperature factor and you decide
whether to keep that or not and you
basically just move through space trying
to find a locally optimal sequence
that's Monte Carlo now the software
requirements um to do these two things
are a little bit different the screening
you can basically use off-the-shelf
software and you can run it on separate
processors and you've got some some
serialization format that you can write
your data out you can send the data
between one processor in the next and
you can do pretty efficient screening
that way the Monte Carlo optimization
modules requires a different kind of
software it requires basically the
ability to build the three-dimensional
structures for the three-dimensional
structure and then mutate the molecule
it requires them all to be tightly
coupled together and one processor
because if you're sending stuff between
processors you're wasting a lot of time
so i'm the monte carlo optimization on
the right I need a new these a
programming language that's really
dedicated to this you need something
that can generate code that runs as fast
as possible to do this kind of Monte
Carlo optimization of molecular
structure so I went off on a little side
project I decided to try and implement
this I spent two years working with
and boost Python and all this hundred
thousand-plus lines of C++ code that I
had written to build molecules and I
basically almost went nuts because I
just can't control every aspect of this
thing out you know Python developers
made certain choices about the garbage
the way memory management is done C++ is
done a different way and getting
everything to talk to each other
everything that I do I basically have to
think about memory management and just
all sorts of stuff like that and it just
really drove me crazy
I wanted a computing environment where I
could control every aspect of it mostly
what I wanted is a computing environment
where my all my atoms and bonds and
everything would be garbage collected I
wouldn't have to worry about carbon
memory management
so I took my sequence plus code that I
wrote and C++ is an excellent language
for for you know building compact arrays
of complexity then you know writing
CAPTCHA where code is excellent for that
writing more complex you know software
you want to try something one day and
then try something else another day
there it's not quite as good because I
always felt like I kind of got locked in
on the code so I just threw together a
sort of archaic interpretive Lisp
interpreter within can do I could do if
and looping and things like that I mean
all the molecules that I just showed you
I designed using that software and I ran
this software with this archaic list
running on about 40,000 CPU
supercomputers and what was then a tera
grid and it designed those molecules and
it was wonderful it because by molecules
I suffer all garbage collected I didn't
have to worry about a lot of stuff for
government issues so I like that but I
didn't want to spin off into the ether
developing a whole new language for
chemistry I didn't think anybody would
use it and I'd have to documented all
these issues about that so I decided to
use a list that lots of people were
already familiar with and I had a choice
of scheme or Common Lisp and they
decided on Common Lisp because it
seemed a lot more full-featured for what
I wanted to do so I wrote a new compiler
for a common list
it's called clasp it stands for Common
Lisp and scientific programming and it
is the foundation that the can-do
software the molecular design
environment mean basically I had to
write a new common list because I want
convalesce to talk to C++ and there
isn't there are very few languages that
can interoperate interoperate with C++
in the way that I needed them to
interoperate so basically what this
compiler does is it takes Common Lisp
code at the top and C++ code it
generates LLVM I are at the time that I
was starting this LLVM was becoming a
thing and it is a wonderful thing it's
sort of a common language for sort of
portable machine language where I don't
have to worry about the details of
processors underneath I can just write L
off the MIR and it gives me the ability
to interface other LLVM languages and
there are really smart people working on
turning this nothing to very fast native
code for different processors so that's
what I did now why choose common list
especially for a scientific language
it's not very commonly used well it's a
compiled language that has a lot of
high-level language features and just
haven't made it into other languages I
mean it's mentioned some of them but
it's got things like you know macros
where you can write programs that write
programs and that's very powerful it's
that great condition system it's just
got a lot of stuff that just has a
maintenance other languages yet it's got
a well defined specification that the
Department of Defense funded back in the
early 90s that package managers asdf and
quicklist for bringing other people's
Common Lisp code into your environment
and they're all written in carbonless
it's a self-hosted language and thus the
cotton if I wrote a compiler in Common
Lisp I would have access to it and I
could tinker with it to make it generate
things like open CLR
you know it's specialized to generate
specialized code that has to run really
fast
there's a bunch of existing
implementations for it there's a great
user community
and an ecosystem existing software it's
also a tried-and-true
industrial-strength language it's used
to run Google flight products and
there's a bunch of other companies that
use common lisp so main thing about
using it for can do is it lets me
garbage call a garbage collect all my
chemistry objects atoms and bonds or
their they're implemented in C++ they're
exposed to common the list and they're
all garbage collect and I have compact
and garbage collection working with C++
objects I'll tell you a little bit about
that it's got a true macro so that grows
are functions that can generate code on
the fly all right it's what C++ template
programming is mimicking but I don't
know this kind of language but basically
the way I think about C++ macros and C+
sorry C++ template programming and
Common Lisp macros are C++ template
programming are two common list macros
what IRS tax forms are to poetry because
C+ was template for gonna be unless you
write a chunk of code where you can sort
of slavi in types okay and that's very
powerful I do a lot of template
programming okay I like it it's powerful
it's difficult though two weeks to
convey your intention to another
programmer because it's sort of really
spread out code on that prose Common
Lisp macros lets you generate code on
the fly in a much more natural way
I tried dynamic variables so dynamic
variables are like Global's but you you
can set them and then all the code that
you call underneath that uses those
variables but as soon as you return the
old values get restored so it's a
chemistry it's right because what it
would let you do is configure global
variables run some code and then
configure the global variables again a
different way and then run that code
again and it lets you it really
facilitates writing software that can
interoperate with other software in the
same ecosystem
Max is a great example of this Emacs is
thousands of people contributed to Emacs
and what I think really makes that work
is dynamic curve
generic functions I can write functions
at dispatching on just one class but
multiple classes of multiple arguments
so you know often I'll have a function
that right I want to call with a
particular energy function and a force
fueled
that's realized I'm spinning a little
bit out of you guys nope you know work
everyday
um there's a bunch of other features in
common list that are great for chemistry
and I I love to go into more detail
about that with anybody who asks um so
classicism is about 150,000 lines of C++
code there's about 30,000 lines of
Common Lisp code that I stole from
another implementation called ECL and
it's also a C++ template library that
lets me easily expose C++ libraries into
the Common Lisp environment so it's sort
of like boost Python I can just give it
the name of a function in Common Lisp a
pointer to the function in C++ and at
the C++ compile time it generates a
wrapper function for it and so it makes
it very easy to expose C++ libraries to
class using CL bind I expose the LLVM
API the C++ API so I am using LLVM C++
API and expose it to Common Lisp
I also exposed a bunch of the clang API
namely the ast clang ast library and the
ast match the library I do that because
I need to sort of grovel through my C++
code to build an interface to my garbage
collector and I've implemented tag
pointers for fixing Lamas characters and
single floats for that things lets you
generate really fast code and class
supports SDF slime and quick lists those
are Lisp things written in Common Lisp
that class uses so basically what it is
all you need is a C++ compiler clang to
compile this thing it starts up an
interpreter that's really slow that
loads some common lisp code which is a
compiler that compiles itself
so quite a bit faster and then the
bootstrapping compiler compiles another
that I got from my collaborator Robert
strand called cleaver and cleaver is a
is going to be a state of the art Common
Lisp compiler it will generate
performant code I can demonstrate in a
second class is very similar to ECL the
embedded Common Lisp which is a common
list based on C and I've done that on
purpose so that I can use a lot of their
code in class also does just-in-time
compilation and ahead of time
compilation
okay so cleaver is a compiler developed
by Robert strand here he's a professor
of computer science at University of
Bordeaux he's got decades of experience
writing compilers and he is trying to
write a common ELISA from Pilate that's
written in them using all the best
features of Common Lisp and it is a
wonderful compiler I've spent the last
couple of months hooking it into mine
and I can now generate code with it
that's runs about four times as fast to
see now will soon be I think comparable
so this is Robert here this is me we
both get our hairstyle by the same
person good old mother good old mother
nature and yeah he's developing this
called cycle which is there on github
and what Leever does is it takes
s-expressions generates and then some of
you guys help me get that down to native
code yeah there's a couple of other
things I generate dwarf debugging
information so I can use ll db2 Common
Lisp and C++ together my stack frames
are interleaved C++ Common Lisp C++
common list class supports to garbage
collectors it's got the main one that's
going to be the memory
system which is a compacting garbage
collector which means it moves C++
objects around in memory constantly
compacts them so could run sin Chi
forever on them on a fixed footprint
whenever right I used the Boehm GC just
a bootstrap this GC so it's a first
Common Lisp built on LLVM yeah I think
that's about all I want to say here
okay so exposing C++ libraries to class
because P plus function here that takes
two arguments returns another argument
all you have to do to expose it to the
common list is like def add numbers give
it the spring here that is the name in
common list pointer to the function and
then CL buying the wrapper c++ template
library which I stole from lua bind
which was built on boost Python will
build a wrapper function for it that
converts the types from common lists to
C++ all the C++ function and then takes
the the return value and converts it
back again and then that's a very fast
round-trip because everything is c++
underneath so this is something I've
just generated in the last couple of
days just sort of shows what the
performance is as of a couple of days
ago I just this is sort of the tip of
the spear tip of the spear in terms of
performance what I'm doing is I've set
up in lining in the cleaver compiler or
Robert strand has helped me and we've
gotten in working and just a simple
little algorithm that calculates the 70
Fibonacci number 100 million times this
the c-class compiler will run about four
times slower than C and Bank comments
which are comparable on my original
compiler that system so just in the last
couple of months I've been able to
this thing of about 220 times and I'm
now very close to performant code what's
missing is type inference and Robert is
building that in once we have type
inference I can see all these repeated
type checks will be stripped out of the
code and it should run very close to
source c++ speeds
okay so then there's candy so I'm gonna
say over here I have - oh like 10 -
right yeah
okay so just ten more minutes I'll give
you a quick tour
so can do is the molecular design
environment that runs on top of runs
within class and that runs on top of C
plus and now of the MIR can do can do
all sorts of chemistry I'm a chemistry
professor I teach undergraduates I crush
the dreams of undergraduates who want to
become a medical students because
they're not studying hard enough but I
teach a lot of chemistry organic
chemistry and so I built a lot of this
stuff into can do so it'll do
maybe let's see build and represent
molecular structures I can build force
fields I've got amber built into it it's
got energy terms for things like chiral
restraints it's important to maintain
the sort of stereochemical configuration
around carbons so I've built over
strength to him in there um it's not a
very fast energy minimizer that uses
three separate algorithms steepest
ascent conjugate gradients and then a
truncated newton-raphson method and it
switches between the different
algorithms as the gradient as you get
closer to this solution so it very
quickly converges on good geometry i've
got automatic type assignment built in
there Nino's climbing old prey a lot of
rules something I wish teach all my
undergraduates
it's got smarts chemical pattern
recognition which is like regular
expressions for molecular fragments
built into it you can and because it's
got a whole language Common Lisp built
into it you could rapidly build complex
chemistry tools within it and everything
is garbage collected internally so you
don't have to worry about memory
management
I can hook in all this external software
as well I've hooked in LLVM and clang
same way I can hook in things like open
mm2 doom electrodynamics amber or open
Babel is a software that will convert a
hundred different chemistry file formats
into hundred if other file formats I get
one of my collaborators on the Internet
is hooking in HTML so we can get
graphics in OpenGL graphics in their
electronic structure calculations we can
hook all this code in and expose it to
the Common Lisp environment and then you
could use it from within clasp rather
than running separate source standalone
programs so here's this an example of
how I'm using this thing ok so I want to
develop molecules that can cure HIV
working on this for two years trying to
make a molecule that can prevent HIV
viruses from using with human membranes
what they do in order to fuse with human
memorising they've got this protein on
them called gp120 gp41 which is on the
surface of the viral coat when it comes
near a human cell
it recognizes something on that cell and
sticks these greasy seams of the human
cell and then it undergoes this massive
conformational change membranes together
so they confuse and then the viral
contents can get injected into the human
self this is a proven mechanism for when
we make molecules that jam this process
HIV can't infect human self Andros was
developed developed a drug called fusing
on that they were there was available
for a while it might still be there's a
huge peptide this huge peptide here that
mimics part of the fusion machinery the
fusion machinery here's a picture of it
this is the gp120 from the crystal
structure and here we're looking at a
sort of hot down there are these three
inner Keela sees and then there are
three outer he'll disease this one's
missing that were part of this whole the
final sort of fusion
complex now the idea is if you can make
a molecule like mimic one of these helix
bombing down this groove then you could
prevent the machinery from folding over
and you can prevent the viral fusion
all right so Roche was synthesizing and
selling this molecule but it's a very
difficult to administer he's got to
basically inject one mil of a
concentrated peptide solution into the
muscle every day because this stuff gets
chewed up within minutes
it's a peptide our bodies know all about
peptides and how to chew them up um some
people would rather die than inject
themselves with you know something like
that every day it's a painful process so
really what you want are molecules that
you could take as a pill right that
would get across the gut and get in and
machinery so that's what we're trying to
develop trying to develop a molecule
that can replace one of these uses this
group that would survive their gut that
would survive all the protease is in our
bloodstream that are true of this true
of this thing very quickly and so we're
designing that and here's the top end
isoleucine
using this designing software on the
scaffold with the right stereochemistry
and we just managed to synthesize and
purify this like last week and we're
going to start throwing it at the
protein to see if it sticks two years
we've been working on chemistry is hard
it's much easier to write software okay
so then what we're going to do is if it
does sneak up here bringing in a Toho
then we can continue to extend the
molecule down build more and more stop
off of it to make more contacts so that
we can bind tightly to this groove and
prevent the machinery from folding up
because we've got to compete against
which is already there it's part of the
protein it's like we got to put a
molecule in there that's gonna Jam that
the closing of these scissors basically
and that's a tough challenge okay so
here's how we do it draw the molecule in
kendra ten draws a program that lets you
draw molecules every chemists use there
are other ways to annotate this molecule
they give it some information here it's
a cartoon of a molecule with a bunch of
annotations run sort of hijacking chem
draws sort of drawing abilities to
annotate the molecule then I load it
into and you you work with can doing
emails using this programming
environment called slime basically here
I'm writing functions that are common
lisp functions and filled molecules here
I'm giving instructions I'm telling can
do what to do with the functions and the
molecules that it loads this is a repla
read eval print loop system and
basically you could put them cursor on
one of these lines hit a key and it
sends this expression to clean ado it
does something and generates output so
what are the kind of things that I do
well I want to read in amber of type
rules and parameters so I'm reading e
antichamber type the rules up there
once this executes I've got a variable
called star F F star it's a dynamic
variable that contains the entire Ambler
force field and type rules in it then I
can load that ChemDraw file I showed you
and once this executes I've got this
dynamic variable AG which contains the
annotated molecule that I drew in
ChemDraw
then I suck out the stereocenters the
carbons that I can configure there five
of them making this molecule that could
be either S or R so each one is like one
bit of information
and control so there are 32 possible
configurations of this molecule and
since I don't have any idea what the
shape of this molecule is right now I
don't know if a crystal structure
anything I have to start from scratch
and scratch for me is just throw random
coordinates at each of the atoms he hit
this massive tangled net I send it to
this program chimera which is developed
up at UCSF to render molecules and then
I want a three-dimensional structure I
just say well build good geometry from
random coordinates so give it means the
molecule I built the force field what it
does is it automatically assigns titles
it automatically assigns force field
parameters and high-roller strings based
on these thermo chemistry I told it then
it minimizes the energy and as it gets
closer to the solution it switches
between these three different methods
then except if the geometry is good
because they had a lot of rings off them
rings will sort of interact and or
penetrate each other you'll get rings
that are chained walk together and
that's not good a good molecule so check
the job training if it's bad it goes
back up to here and it just keeps doing
this ok this is all implemented in
Common Lisp on top of C++ code and then
I want to generate all stereoisomers I
just write a loop I generates iterates
through all 32 of them it uses the
imager values 0 to 31 the binary pattern
to represent SNR any position made
assigning to stereochemistry notes a
geometry for the molecule writes it out
through a file in a multi-format
and then I can say array them all in one
structure and there they all are and
this takes about 15 minutes to to
generate it this way just to show you
that these have different shapes what I
did is I took that molecule they say
take three items from that top right
corner there and superimpose them for
every one of these 32 structures here's
two of them superimpose so here's that
you have the top there superimpose you
can see these are very different shapes
the blue versus the brown very different
shape there's all 32 of them on top of
each other this is the super post up
there
you've got thirty two very different
shake molecules alright so these are the
molecules that I can now use to screen
and computation look for the one that
might bind that groove the best okay so
I've got way down the rabbit hole here
and the chemists I'm supposed to be
writing grant proposals and guiding
students to make molecules and stuff if
I don't know what molecule is to have
them make we're kind of stuck we can
make these complex huge molecules but I
I need a way to design them see the
batter just make a whole bunch of them
and just screen them which is its own
problem its own challenges so this is
what I've done and I think I'm coming
out the other side with something in my
teeth that we can actually use to design
molecules and so that's why basically
what I've done I've developed this
hardware mr. Loomer sort of a new
instruction set for building functional
matter software to help us design it and
we're moving forward hopefully funding
allows to try and develop molecules that
can cure diseases buying small molecules
act as channels speed up chemical
reactions all sorts of things so class
is available on the internet it's on
github runs on OSX and Linux and candy
will be as soon as I get the licensing
stuff worked out with temple with that
thank you very much for inviting me and
I would love to answer any questions
that you have
yes so 50 is tough we've made 8 verse 8
MERS are 4 nanometers long there's not
much you want to do with that stuff up
but it's know you don't have to think
about it hard at all it's the same kind
of chemistry that that you can buy on
the internet DNA come by DNA
oligonucleotides you can bind by
peptides and it's basically robots can
mix together chemicals one that is time
you start with a plastic resin let's got
a group on it that you can attach
something to you throw in a lot of that
something it attaches you take off the
protecting group then you bring in the
next one and you attach it and then you
bring him the next one cycle time is
about two hours you can assemble these
one building block at a time with
precise control over the sequence
because you're adding building block X
then Y &amp;amp; Z I don't know what to do with
anything of that length everything we do
is it is difficult because we have to
synthesize all these building blocks
ourselves and we've turned that into
process chemistry where industrial you
you could make ton quantities of these
building blocks but right now we make
them in the lab by hand 150 grams at a
time and then we start using it so um
there is a fair amount of effort
expending a lot of effort it's better
than that putting them together is
fairly straightforward but you know
we've spent the last like only 5 years
ago where six years ago where we able to
start putting a functional group on them
so we've been developing all of this
stuff sort of as we go along but you can
assemble these things the same way you
some companies robots are assembled
peptides we do everything by hand right
now because it's just more efficient
and get better product bottom right so
each one of these represents one of
these building blocks and this thing
represents society we complete chemical
reactivity on there that can cross slate
with other side chains so we can create
create branched and network cross-linked
structures there isn't a lot there's a
fair amount of sort of cross-linking
chemistry that we can use to put things
together oh you just get that that's for
free oh you need to make a ring you you
if you take two rods and you connect
them through a little tether they will
flop around like logs that are chained
together 300 Kelvin room temperature
things are solving it cycles and we just
trying to get the first paper of that
published but we've made a bunch of
these macro cyclists they haven't
published any of them yeah and the trick
there is to make the cross occurs as
short as possible so you can still
control the shapes they don't roll
around and around they pucker they flop
they do everything the problem is not
these things are too rigid the problem
is that they're still too flexible
you've got to wrestle everything into
position and this is something I have a
lot of trouble convincing like mr.
Collins dynamics is really important
that these are like diamond they are not
they're very flexible they're just not
crazy flexible like peptides and
peptides in DNA are
in terms of the stereochemistry and the
clinic davina stuff it's perfect and
then we know exactly what we're putting
together the we've gotten a couple of
crystal structures they look exactly
like what you build it's almost boring
they look exactly like what you build
with a plastic model set now these
bigger things we barely made a few of
them just enough to get bass back
structural determination and I'm dying
to see what they look like in reality if
they have shape trying to make something
in a shape and then see what that shape
is and see if it looks like what we
predict it to look far fewer degrees of
freedom here to worry about you will I
mean I'm paying the cost of all that
computational searching they're more
computationally tractable yeah I mean
it's easier to pretty it's easy to
predict the shape of a ladder it's not
easy to predict the shape of a piece of
strength a ladder has one shape piece of
string has infinite number of binding at
protein would be I've got three four or
five functional groups there's there's a
bunch of Fitness functions we can use
one is just I've got these three four
five functional groups arranged in a
particular constellation in space and
can I make a scaffold that mimics that
constellation so it's basically a
root-mean-square difference between the
positions of a bunch of adders a more
sophisticated one would be to define an
energy function between the molecule
that I'm designing and a molecule I'm
trying to bind to and David Baker
University of Washington has written a
software called Rosetta and they can use
to computationally design protein
interface of the bind of the protein
they've been very successful at that I
want to use the same kind of idea here
so you need to find an energy function a
scoring function and you dock it against
the protein and then you move take this
and you talk it again it's a more
expensive computational process but
that's what I think Carlo I mean he uses
Monte Carlo he's the one who taught me
that yes
I want to build things like that and in
two days Eric Drexler and I will be at a
Department of Energy workshop talking
about that yeah I know Tom McClean
manufacturing so I think for these sort
of into building bigger bricks that we
could then connect together in is is my
goal here for that part of it so if I
can build I think we're gonna need to
use Monte Carlo to design things like
this but if I can design a couple of
bricks where I could put functional
groups on the outside that could link up
with other bricks make it then I then I
could manhandle the one on top of
another and snap together to make bigger
comically precise materials then I've
got something like a subroutine that I
can connect to another subroutine and
make something
yes no I'm using the memory I'm using
Boehm to bootstrap and I'm using the
memory pool system by Raven Brook which
is AC library that does copying garbage
collection it was developed for Apple
for Dylan this language that they were
developing for awhile and it is it's a
wonderful complex garbage collector that
does moving and garbage collection it's
written in C and I've just hooked it
into my thing the thing about it is you
have to when you move a C++ object from
one place in memory to another which
people you know in the C++ chatroom told
me was impossible you have to update
every pointer that points the point here
to there now when you've moved it right
and if one pointer gets out of place
that thing will tear itself apart a
printer will crash so I wrote a static
analyzer in common list that uses STD
names to match your library and you know
and it finds every pointer and writes
about 20,000 lines of C++ code updates
every pointer
continuously and it works perfectly so
that garbage collector is a generation
it has it's a generational garbage
culture and they've worked hard to
reduce the pauses I'm not so worried
about it because scientific programming
doesn't really mean I just wanted to run
as fast as possible and pulling up my
garbage and work in a fixed memory
footprint because if I run this on
supercomputers oh I've got two gigabytes
to work with I can't have a garbage
collector like Boehm which will fragment
and eventually run out of memory I makes
me very uneasy so also these it's a
generational garbage collector so it
continually compacts memory and it's um
it runs faster than ball by about two
fold
yes back then it wasn't a thing and no I
didn't want to work with Java and the
Java sort of oh they looked much more
and it was really well thought out all
right thank you very much thanks for
coming to Google</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>