<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Grid-based Integrated Bioinformatics Systems for High... | Coder Coacher - Coaching Coders</title><meta content="Grid-based Integrated Bioinformatics Systems for High... - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/GoogleTechTalks/">GoogleTechTalks</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Grid-based Integrated Bioinformatics Systems for High...</b></h2><h5 class="post__date">2007-10-08</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/yWr74Ihbyf4" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">first I want to apologize for being a
little bit late we've got lost which is
very typical situation for the
biologists you can get you can really
get lost in the amounts of biological
data and I will tell you more about it
during my talk and what I will be
talking about is integrated
bioinformatics systems and tools that we
are developing at bioinformatics group
at computational Institute of the
University of Chicago and mathematics
and computer science division at argonne
national lab there is a lot of people
working in this group and they're
absolutely wonderful and the one to talk
you will understand what exactly are we
doing by doing that so what are we going
to talk about first I will talk a little
bit about our approach to study
evolution of genomes with a baluch
networks protein families and many more
and this is about bioinformatics which
is absolutely fascinating subject which
promises a lot delivers quite a bit and
probably will do change this
civilization in the future ah so then i
will talk it'd tell you a little bit
what are we doing we are building the
extra-large information systems for
high-throughput analysis of genomes and
double constructions for public genomes
and user submitted genomes and if we
don't have the tools to analyze this
data we just build them I will also talk
about grid-based integrated
computational infrastructure because
their noses are so high throughput so
complex so tedious that it will be just
inhuman for people to do it manually so
we are automating most of the processes
that we are doing but the most
interesting part to me because I
actually I've heard myself giving this
presentation many many times so this
presentation is not interesting to me
but what the part that is really
interesting to me is the problems and we
do have a lot of problems and
bioinformatics and we do need help in
solving of these problems so let's start
on the last several years not more than
a decade there was a madness in
sequencing of the genomes there are so
many before I will go any further do you
have any background in biology do you
know what sequencing of the genomes is
so I in this case I will just talk your
language so there was a lot of
sequencing of the genomes and this is
this is sheer madness what drives this
madness is a lot of different groups are
interested in sequences and genomes of
particular organizers but what emerges
from this variety of the organisms and
individual interests of individual
groups is an ability to look at the data
and do the comparative analysis across
multiple evolutionary domains to do
efficient comparative analysis because
what is interesting is that the major
approach that bioinformatics takes is
the comparison it's in comparison
analysis it's a science of big numbers I
it thrives on the high-throughput
comparisons of what is known to what is
unknown so in this case you can transfer
annotations by similarity from what is
unknown from what is known to what is
unknown you can compare functional and
genomic properties of different
organisms from different taxonomic
groups like for example what is the
difference between the sequences and
what is the difference between the
metabolic pathways for the organisms who
live in thermophilic environment versus
the organisms who live in Antarctica
there should be the differences and this
differences can be analyzed and
formalized on the level of the sequence
genomic sequence analysis for example
what is the difference between the
pathogenic organisms and nonpathogenic
organisms just given you a small example
for example the BCG strain of
mycobacteria which is used for
vaccination against tuberculosis is
different from the pathogenic micro
plasma tuberculosis only in 13 genes
and by comparisons you can probably
identify the pathogenic factors that are
directly connected to the disease so in
this case I it's obvious that more data
you have more diverse the set of genes
that you will be looking at more
interesting patterns you can identify
more interesting comparisons you can do
and more interesting information you can
derive for bioengineering genetics
medicine so on the goal of this madness
is actually to make the efficient
information base for deriving the
conclusions about the organisms
functionality what is the same and what
is different in different biological
organisms and how it affects function
but also what I want to know you to note
on this slide is geometrical progression
of the data and then we return back to
this geometrical progression of the data
relating my talk in the problems on side
so we decided that the major thing for
us and the major emphasis is comparative
analysis but in order to support this
comparative analysis what do we need to
do first our biological systems they are
very complex it's not only the sequences
sequences Cal and like DNA is followed
by them and gene products gene products
perform functions functions are
connected into the complicated networks
of interactions so we need to understand
the functionality of biological system
so we can somehow control it and modify
it we need to understand functionality
of the systems on various deal levels of
organization and that's why we need to
integrate information from sequence
databases are features databases
metabolic databases regulatory databases
more information you have more different
angles you can look at this information
more comparisons you can do whether you
will understand the functionality of the
system so integration of the data is one
of the major requirements for efficient
bioinformatics analysis the second one
is we need to have tools and algorithms
for pattern recognition and comparisons
of bio systems on various levels of
organization is just the comparisons of
the sequences is way not enough for
contemporary bioinformatics we need to
push it much further and this is I'm
also going to talk about it and the
problems part of the talk will also need
but because the the amounts of data are
so humongous we need scalable
computational resources to do this
across the board comparisons we need to
have on demand scalable systems and in
this case we really need to have
heavy-duty computational background so
the good news is that we can do it now
sort of sometimes better sometimes worse
but we can do it now so let me tell you
a little bit about our approach to
understanding of the biological systems
we firmly believe that to understand
biological system is possible if you
will understand the way it was evolving
because what was happening during the
evolution it was almost like genetic
engineering in action the systems were
the the nature was tweaking the system
to adjust the different on the
environmental conditions it was
modifying it as much as needed for the
organism to survive so in this case if
you will look them from the evolutionary
perspective then you can understand the
logic of the changes you can understand
why this components will change that how
they were changed but on what is
happening that nothing is changing in
isolations like for example everything
in the biological system Cuervos that's
why it's a system it has a systems
response you have a signal you have the
system's response which means that
metabo like for example protein families
and the genomes they are changing
because metabolic networks are changing
and metabolic networks are changing
because the environment environmental
conditions are changing right for
example if you will pour sulfuric acid
onto the organism it's better to do
something but exactly what's needed for
survival under this harsh conditions so
pretty much on everything is going so if
we want to look just at the protein
families we will never ever understand
the logic of the changes so and what
forms this changes is environmental
conditions something changes in
environment you better mutate on
taxonomy taxonomy imposes constraints
because you can't evolve without
constraints but if I am a human i will
be moving like a human if I am an
elephant I will be evolving like an
elephant so that like if you are an arky
bacteria you have certain commitments
you have certain history and you have
the boundaries of changes so all our key
bacteria will have shared certain
properties so also physiological
profiles there are major design
commitments that organisms are making
like for example to breathe to use
oxygen for respiration or not to use
oxygen for respiration their major
building blocks major decisions web for
example TR de Chardin which was a
wonderful revolutionaries that the
beginning of the 20th century he thought
that evolutionary changes is everything
that is related to their you access to
the new energy sources or access to the
new kinds of information so this is
evolutionary changes there is a lot of
diversity but really big design
commitments what energy to use or what
the sources mechanisms for acquiring of
information to use and this is a
valuation so having all this in mind we
started to build down the integrated
information systems so
and with my group built the Puma to
narrow system to provide an environment
for co-evolutionary as we decided in
comparative analysis of genomes
metabolic networks and enzymes in the
framework of taxonomic phenotypic and
physiological information and the reason
why we did it because we really want to
understand the logic of the organism
without understanding of the logic of
the organism it will be very difficult
to interpret the events of molecular
evolution so what Puma has now puma is
an interactive integrated environment
for high-throughput genetic sequence
analysis and the tab electric
instructions are with grid-based
computational back-end every single word
in this definition means something so
and I will tell you a little bit more
what every single word in this means and
this other screenshots from the Puma
system so it integrates information from
over 25 genomic with a baluch structural
and taxonomic databases we are taking
all sequence data that is available in
the public domain from NCBI uniprot PIR
and reintegrating it in one environment
then we are also integrating information
from the available metabolic databases
and for example structural databases PDB
and so whatever we can go get hold on
the information we're integrating and
we're warehousing it I and I'm repeating
this where we're causing it instead of
doing something more reasonable and I
will complain in the last part of my
talk way where warehousing so but it's
not enough how this information is not
enough for us so we decide to amplify
this information so we are taking all of
the sequence data that is available and
we're actually
raising it on the grid technology allows
us to analyze it using blast blocks
interpret so we're amplifying the data
we're on updating it with that
additional features and the reason why
we are doing this because we want to do
the comparisons more efficiently we want
to recognize patterns more efficiently
and that's why we're amplifying the data
so we are doing automated metabolic
reconstructions and i will tell a little
bit more about what does it mean to do
the metabolic reconstruction but in in a
nutshell if you have the functions of
the genes predicted in the genome then
you can superimpose it on the functional
networks from the databases like care
good we have our database EMP with
metabolic pathways and without knowing
anything about the lifestyle from the
organisms or their physiology you can
actually reconstruct the physiological
profiles you can reconstruct the
features that organisms have and predict
certain features just from the genomic
repertoire it will be sure enough it
will be not complete sure enough it will
have some of the pathways some of the
genes they will not be expressed but at
least it's a valid model to start from
for the experimentalists so on we have a
suite of unique tools for evolutionary
analysis of enzymes and the double of
networks I told you we don't have tools
we don't wait we just go ahead and build
on and we have satellite databases on
the satellite databases because we want
to look at the certain parts of the
biological story in more details like
for example and also because some people
are paying as money to do that we're
very part of three by defense on the NIH
centers of excellence and that's why we
are creating the pathless database I and
we are part of the structural biology NH
structural biology center and we are
predicting by
biological interesting targets so we
have several different on control and
refined data sets so the this is the
interesting system about which I want to
complain in the last part of my talk I'm
promising so much complaints but I will
really do that this is an air system for
high-throughput analysis of user
submitted genomes if we have this
madness if we have this sequencing
madness which it comes as an avalanche
of data you better have automated
systems that will allow the user to
download your genome and it will be a
tomato automatically annotated and then
you need to provide the interactive
environment for the users to actually
refine the analysis manually but you
need to start from valid good automated
model for genome analysis and on so far
we have without any publication yet we
didn't even publish the paper but they
already have quite a bit of users and
like we have your senior pastors and
human fools and Felicity to the strains
of immortals influencer and we have
Savannah or Federation from geo is using
out so we have a lot of different things
we can process and fully analyze and I
will tell you what does it mean to
analyze the bacterial genome in four
hours and it performs the analysis and
displays the results so what are the
steps of analysis that Puma does and
Nair does first assignment of functions
to the genes and characterization of the
proteins so once users have submitted
the genome it is we are running it on
the grid against their array of tools
including blast as I said blast blogs
here measurement therapy from our own
tools chisel and we are collecting all
this information to input it into the
voting algorithm which will
on to predict the functions of the genes
so it's not enough just to round blasts
on the sequence data to assign functions
to the genes because it's too crude we
are collecting information about domains
and active sites and whatever we can get
in order to assign functions to the
genes reliably so our then what we are
doing we are super imposing it on the
collection of metabolic and functional
networks on the assigned functions were
superimposing them onto the known
pathways and this technology was
developed by in Guinea silcove in 1993
and so we are predicting the double up
pathways and providing it in interactive
environment for users to reform and then
my favorite part comes how to predict
the fan types how to predict the
physiological properties of the
organisms how to identify evolutionary
and co evolutionary patterns and
signatures and to do that we need very
efficient ways to do the pattern
recognition and it's better to be
multi-dimensional pattern recognition we
don't have a lot of algorithms that we
desperately need and it should be all in
comparative framework so this is how
sequence analysis is done and presented
to the user in puma in there so you have
information from public databases
integrated in the header you have
interact that you have pre computed data
on the right side of the slide and it
shows the similarity searches this is
block the center predominant all is lost
if you will go down there will be blast
against pdb structural databases now if
user if this information is not enough
for the user on the left side you can he
can do interactive analysis so that
there is a lot of possibilities to
refine the analysis and user can add
comments and validations
this is call the metabolic
reconstruction to look like and what is
happening is this is for example the
pathway from the P database we can also
do the reconstruction from the cab
database and where we need to have an
engine how to add the user submitted
pathways and what it gives us like for
example this pathway it represents the
glycolysis which has an alternative
behavior some of the organisms they
perform atp-dependent they have
atp-dependent phosphofructokinase the
other organisms they have pyrophosphate
dependent phosphofructokinase and work
for example from the genomic data we
understood that he muscled have waffle
is influenza was predicted to have
atp-dependent was referred to kinase and
we compared it and we have an ability to
compare across the board with all three
hundred different organisms for which we
have made about reconstruction yantra
panama pollen which causes syphilis and
it has pyrophosphate dependent
phospholipid kinase so you can start to
see the patterns you can and if you will
go systematically through the biological
system you can actually understand quite
a bit about its functionality so we will
also sometimes when you are looking at
the pathway you see that some enzymes
there are gaps work for example there's
I we didn't find phosphofructokinase in
salon strain of haemophilus so what we
are doing is we offering the user to
select the phylogenetic close
phosphofructokinase blast at the game's
genome and voila we found the kid good
candidate for this function so this is a
good hypothesis for experimentalists to
go ahead and actually very far either
this is really it we can do it from
here we can compare different strains
like what is the difference between
different strains and what is important
is to have different ways to look at the
pathways different ways to look at the
sequences and this is a fella Tibbett
point of view like show me how it's done
by thermophiles will show me how it's
done by pathogen so show me how it's
done by for example alpha proteobacteria
so that's what we are trying to do we
have created but what what we started to
think about is okay life is a continuum
so that's why everybody is so similar to
each other similar but not identical so
and like for example alcohol
dehydrogenase it's implemented
differently in different organisms and
in different domains in some organisms
it's one subunit and some organisms it's
two subunits in some organisms it's
atp-dependent in some other organisms it
uses so there could be different
variations of the same functions against
the genomes and the amounts of sequence
data now allow to look at these
variations and that's why we created the
rules based clustering algorithm which
allows to look at these variations why
is it important to identify this
taxonomic and phenotypic variations of
functions first for identification of
the protein and high-throughput analysis
but what is interesting is Diagnostics
if you want to for example there is a
bio defense problem you want to identify
organisms without sequencing really fast
so in this case if you have an array a
library of the taxonomy specific on
enzymes all of you all of the
nucleotides you can
use it for on identification of the
organisms in vitro it will be
interesting for biotechnology because if
you look at the sermon stable variation
of enzyme versus non terminal them it
will be also interesting for
biotechnology it's very interesting for
interpretation of metagenomes and
Nicholas's are doing the meta-genome our
noses ooh it's useful to do accurately
metabolic instructions and annotate the
organism specific pathways so the idea
is to identify what is the same and what
is different and how it affects function
we used this for adaptation of the
meta-genome from comfort site conferred
side is the de hi nuclear waste storage
facility and under this extremely high
levels of radiation tanks one of them
started to leak and this matter genome
was obtained from this population which
lives under the leaking high radiation
nuclear waste in boiling nitric acid was
extremely high concentrations of corneum
every single condition in this pushes
the boundaries of life as we know it way
beyond our imagination so we are
analyzing this organisms and actually it
helps on to identify the patterns so the
way we are doing it on so how different
that is taxa magnet with variations of
enzymes this is a small example which
will show you this is
phosphofructokinase no this is fructose
bisphosphate aldolase and other
glycolytic enzymes yellow line is the
consensus sequence for posterior lotsa
which is haemophilus the blue line
is on the eukaryotes the pink lion is
antibacterial and this is the alignment
as represented by pool is so you see
this part is completely different in
eukaryotes this much is are completely
different in pasture lotsa then enter
bacteria has its own unique part so it
can be played with it can be used for
different purposes the differences so
comforted meta-genome analysis we talked
about this one we have actually
identified over 28 species that live in
this cozy conditions so we have part as
database of packaging factors we have
brand new central database of signal
transduction proteins and this is how it
looks like and it's super imposed on
pathways and you can do comparative
analysis the signal transduction systems
it's like on the census for bacteria and
if they don't sense the sequence signal
then they can't do anything about it so
if you identify what they see yeah you
can infer the physiology to some extent
target database and this slide is
unintelligible but it's actually on I
put it here just to show you the
complexity of computational background
and the system continent consists that
drives all this analysis it consists of
five for conceptual parts so first you
need to have portal so you can support
analysis by the groups and you can have
the secure environment so and
partitioning of the web space then you
you definitely cannot do all this
analysis manually so we're expressing
our workflow pipelines
in virtual data language video and
camera systems were interacting closely
working closely with a global world
global group on ian's foster group on
and we expressing the workflows and
checkers in matter languages so and also
it's necessary to do that to drive
processes to the grid and we are using
grid as a computational background both
University of Chicago and Darwin they
have very extensive super computing
capabilities but it's not more because
you need the reservations and we are too
lazy to wait for the reservation so if
you want to do high throughput Gina
models as we're just grabbing the last
run that we had it was 1700 CPUs and
some of them they were just on second
day the day we in South Korea some of
them they're in Europe some of them were
we're using several different grids to
where to run the processes terribly the
open science grid do a science grid on
so we are using different grids and on
this is another way how to look at the
system so we have user integrated
database and we have knowledge base from
the public resources we have user
annotations and we have workflow
managers and have user interfaces so and
all of them they gone we're trying to
automate as possible automated update
cycle because it's very large amounts of
information and this is on the schema
from the last one the the resources that
we took on we are mostly using open
science grid which was created by the
physicists physicist song called can
find common language it never happens in
biology biology it's difficult
language now that will complain about
common language so that's what drives
these are different technologies that
drive all this automated workflows
behind the scenes so everything is being
controlled and we are rolling ahead but
we do have problems so the first problem
that we have are what is happening in
biology I I wanted to give you a slide
but unfortunately somehow it slipped out
on the number of the biological
anthologies that are currently being
developed or just in the state of
development and I'm personally a member
of like five committees on the
development pathologies and inversion
and I'm Russian as you can deduce from
my accent in Russia there is a the same
which has your friends against whom and
this is approximately what is happening
in biological anthologies because
different groups are developing
anthologies and sometimes they merge
them but sometimes it's very difficult
to merge them and I I personally think
that as somehow it happened that I told
you got into the domain of social
problem but I I think it's actually
technological problem and technological
problem I can illustrate why is a
technological problem and why
anthologies problem should be probably
sold slightly differently but I can be
wrong I was wrong so many times so this
is on why kahless's as represented by KO
right this is guaco losses as
represented by echo side please see how
many what did the difference in number
of components this is glycolysis
represented by NP database actually NP
database has 62 different versions of
glycolysis that are different in
different like insane composition and
everything
so how people can agree in the notion of
glycolysis I don't think it will ever
happen because all those representations
are absolutely valid this is just it's
normal that people have different
opinions about abstractions and what I
think it somehow people got pretty good
anthologies for representing facts like
sequences or sequence features something
that is can be interpreted unambiguously
although we biologists will all will
always find some ambiguity and
everything but on but once you go to
obstructions you will never ever ever
find any social consensus in
representation and that's why I think
that on what will be the solution to
this is some notion of flustering of
notion some such as some notion of
clustering of abstractions and distances
between different notions and distances
between different abstractions and
everybody will make the decision how
little or how far they want to go in the
relaxations of the intelligence of their
knowledge 'yes sir
yes there can you play us a little
digraphs here I'm lost
so what is happening is there is a
metabolic pathway which is called
glycolysis this is a pathway or which is
how to break glucose to pyruvate some
people think it's not the pyruvate some
people think it's further down some
people will include some additional
reactions to the some people will not
include the additional reaction like for
example in this particular diagram this
is the all-inclusive diagram from chem
and as you see it's called glycolysis
it's a very popular database they just
people using it the Lord ok this is
representation much more reduced
representation of the same path with
black hoses the people just name it
differently like they did it'sit's
glycolysis but they include only this
steps to pyruvate and fire it was here
this this one is a subset of this one
but if you'll tell biologist what is
this they will say this is quite
kahless's if you ask biologists what is
this they will say it's called courses
if you ask biologists what is this they
will say it's glycolysis there are 62
different versions of rifles one enzyme
will be different one step will be
different now they will still recognize
that this is glycolysis it's almost like
we like a chicken the white chicken and
the black chicken and the chicken is
it's still a chicken everybody knows
it's a chicken that's how we we we know
how to clusters all those chickens are
now add anthologies they don't know like
for example I will remember the vibe
packs group for the development okay
they will saying hmmm why kahless's is
this okay whatever um so it's a it's
it's a different way and I'm not
sure that it's possible to find the
social consensus dude I'm not sure that
it's it's I think it's an intrinsic
problem of the data and the the thing is
that in biology like physics they have
less problems with anthologies and the
reason why is because they have a
culture of abstraction which was
developed very pretty well biologists
they don't have well developed culture
of obstructions and that's why probably
it will be very like people will say
guapos is doing in this or this okay so
what whatever so to pyruvate
is it helped the DNA absolutely and we
actually after the science fool cam that
was here we were actually starting to
organize the workshop with a michael
berry who is a quantum physicist so we
wanted to bring the biology that
physicists to biology but this one I
think it's mostly well and the problem
of like how you will name the things
that don't have strict definitions so
and there is a you give them strict
definitions but how you will like for
example this part likelike oses it's a
well-known pathway how you will name
this part of the network which is not a
self-sufficient pottery it's just
something you need to formalize in this
case you need to formalize every part of
the every network you can get and give
it a name or somehow establish the
connections like for example this set of
let's even consider it like look at it
as a graph that the this node is most
likely to be connected to this this is
this is this and give me the depths of
the network reg'lar to do them I don't
know I'm just I know the problems I
don't know the answers but on its it's
actually very very strange that on there
are good anthologies but somehow work
for example high-throughput the
integrative people like us we are not
using them because they are restrictive
we are just being good citizens we go to
all the meetings and that's about it
so we need technologies for quarter in
oceans definitions and obstructions and
they should be hierarchical to something
somehow on databases so databases are
the thing is we're analyzing the genomes
for our users and users always have they
have one property in common everything
is different they always want more and
different so what they want is they want
to customize the databases they want
like for example analyzing malaria
parasite genomes and they want to import
the malaria parasite genomes from the
different databases but there are the
people who are in high modulus they
don't the continent spell apicomplexan
so in this case they they don't care
about this and they when the it becomes
like a Christmas tree you have this core
of integrated information but everybody
wants their own needles and it becomes
absolutely unmanageable and in the
situation when the anthologies are not
very well developed what to do I don't
know no we're just saying okay we can do
what we can do but not more than that
but it's not the solution it's not the
solution you were them so we have this
problem the other problem is actually
most of the data in biology and
bioinformatics it's in the form of the
trees or networks and we have very good
relationships with Oracle people and
we're interacting quite a bit with them
because we have an identity crisis for
our date and it's like is it relational
databases like object oriented know
maybe really let Sun flatten the nested
tables because they're too slow what's
what's the role of smash the tables so
we're constantly constantly looking for
some solution to the problem that
actually is a real problem for us how to
represent
we gave them complex networks and one of
the networks are very large what to do
because if you want to represent all of
the reactions where you want to
represent all of the connections between
the protein food products and the
protein space you just your lost its its
current problem in algorithms so let me
tell you about another madness that is
going on and bioinformatics so you've
seen this exponentially growing data so
what is happening one of the major tools
in bioinformatics is blast which is
based on pairwise comparisons between
the sequences you take the sequence and
you're comparing it with all sequences
in the databases the size of the
database is the increasing doubling or
tripling every year so what we are doing
is we're very optimistically and
joyfully go into the N squared problem
because on very soon blood is already
starting to lose its functionality
because on its I if you if you are
getting like on the blast results 400
strains of the same thing it's not
informative also you you can't identify
them on weak signal anymore so it's a
and we are personally trying to get rid
of lost but it will be wonderful if
somebody will get rid please or boss
because it's it's madness and basically
what blast does it does memoryless
alzheimer style clustering on the fly
because it does clustering it forgets
about the results and then it goes to
the next Island does it so it said I
don't think it's an efficient way to do
our noses
so and the other thing that we
absolutely need because as we discussed
before we want to look at
multi-dimensional patterns we want to
see how things are cool evolving now in
the biological systems we need
algorithms for identification of
multi-dimensional patterns where for
example how to correlate the particular
type of the enzyme characteristic for
the double group with a taxonomic
representation of network and to project
it on the genome how to analyze this
multi-dimensional events that we're
taking place I don't know it's on the
problem between and bioinformatics is
the difference in language between
mathematicians and the biologists
biologists think in pictures and the
complexities and mathematicians
especially I worked with a lot over
because they are logician at some point
that he was asking what is true and what
is not true and what is true in biology
sometimes this sometimes there's it's
conditional behavior so it's very
difficult to actually explain what to do
so are we need this algorithms for
identification of multi-dimensional
patterns so we can make sense on the
evolutionary of the evolutionary events
and this is my group and I was very very
lucky everybody is just on the way
probably talented and creative and
really love what they're doing and I
think that this is this is why we are
just moving the charging ahead and doing
things and thank you very much for the
to the globus guys now they really
helped us what we couldn't do anything
with all them and our ah managers and
everybody thank you</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>