<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Open Source Drug Discovery for Neglected Diseases | Coder Coacher - Coaching Coders</title><meta content="Open Source Drug Discovery for Neglected Diseases - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/GoogleTechTalks/">GoogleTechTalks</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Open Source Drug Discovery for Neglected Diseases</b></h2><h5 class="post__date">2007-10-08</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/2zJ7ZLAhvpE" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">I
hello welcome to the Google tactic we
are very pleased to have professor mark
material with us today let me give a
brief introduction he received his BSC
degree in genetics from Autonomous
University of Barcelona Catalonia in
1994 and he continued there with a UAB
scholarship and received his PhD in
biophysics in 1999 his ph.d was with
under the guidance of professors Martin
Karplus Frances levels and Baldomero
Olivia Oliver and he focused on
development of methods for folding
studies using molecular dynamics after
that he he he joined Sally lab which is
laboratory of molecular biophysics at
Rockefeller University as part of valcom
fund posture post actor postdoctoral
fellow and he was a research associate
at the same lab that he concentrated on
developing comparative protein structure
modeling methods since 2003 professor
mark metronome has been an adjunct
assistant professor at UCSF today our is
going to talk to us about topical
disease initiative White is interesting
to Google and in my opinion is very
important for the entire world please
welcome professor
thank you I question thank you guys and
also happy child who I know it's in New
York him is he's the guy who heard that
very same torque at Stanford where I
gave it about a month ago two month ago
and he invited me here so I thank you
both of you guys and as you heard I'm
biologist so don't expect me to talk too
much about computational science or
engineering I'm a biologist I did my
degree in genetics but then I moved into
the theoretical biology working on
protein structure prediction that's what
we do and that's what i call my day job
to do comparative modeling what what
competitive modeling is is we take all
known protein sequences and for you guys
that are familiar with the genome each
gene produces one protein and the
protein is actually what makes the
function you your cells function so we
take all those proteins and we try to
predict the three-dimensional structure
and here's the pic that each one of
these little DeRose here represents one
protein and we try to do all the
proteins that we get our hands on this
is in the framework of structural
genomics this is the next step after the
genomics effort and what we try to do
here is to predict the structure so that
you guys we can actually work and
developing new drugs but this is my day
job this is what UCSF pays me for and
that what pays my bills but a night I
work on TDI and unfortunately UCSF is
not as nice as Google so i don't have my
twenty percent but this is my twenty
percent my twenty percent starts at 8pm
and finish at one I am that's my twenty
percent so I work at TDI and I'm going
to be talking about TDI because I think
a TDI is more p p8 for you guys because
we are eager to welcome anybody not just
the google people and anybody to help us
on reaching our goal which is developing
new tools that is going to help people
researchers out there to develop new
drugs for neglected diseases there are
tons of neglected diseases mostly in
africa
and India and China and those neglected
diseases are called neglected because
the Big Pharma are not trying to do
anything about them we have also
neglected diseases in the US but they
are neglected because they are affecting
only a very small part of the population
of the u.s. in Africa what happens they
are neglected because people can
actually not pay for the drugs that they
would need we're talking about malaria
Chagas disease cholera tuberculosis etc
and is not affecting a small number of
people is affecting millions and
millions and millions of people so I
think it's important to start developing
some sort of open source biology so that
we actually can do that cheaper lower
the cost of developing the tracks and
then therefore these people can access
to those drugs so I'm dedicating my
twenty percent of the time to TDI some
point is for you guys if you want to
know a little bit more you should go to
tropical disease or does the website for
TDI I answer because it would hear some
flyers if you guys want to take them
with you we have two partnerships one
with the natural publishers they run our
wiki side and another one with which is
very good it's with the people at the
synaptic lip this this is a new
organization that they're a nonprofit
organization is lead by ginger taylor
she used to work for PeopleSoft then
oracle and now she actually quit and
decided to dedicate herself to
developing whatever she is doing at the
scene happy live her goal is to generate
hoons for an open-source biomedical
research so TDI is actually the first
client that she's getting she's going to
be developing tools for us and her goal
is to accelerate the drug discovery
process from basic research that's what
i do to clinical research that what
medical people do so if you want to if
you want to know a little bit more just
go into their website so let me let me
give you some probably gonna get this
with me let me give you some pointers
sometime back ago we started this about
two years ago in 2004 by publishing a
paper
in a journal called the public library
of science plus in medicine so we've got
the paper old and people were excited
about it we got tons of emails and here
I'm just picking some of them most of
them will say oh this is this is great
but what the hell is happening here what
what can I do for helping you another
one would say you guys a look good but
you're like an action of plan then you
have some of the people that say if
someone will tell me where I sign up
I'll give you some research topics to
begin to i'll be grateful so the message
was clear we like your idea but please
help us to do something for you this dis
i like very much is a programmer like
you guys most of you i guess our
programmers not even a buying from
addition who got interested in what
we're doing because probably so the open
source world someone and and he got
interested but he says no started with
rms he gave us a programming
administration tools to play with linux
started with lino so we get a new
operating system to play with please you
gotta get some body great in the field
and release something for everybody to
play with and that's why we actually
trying to do we're trying to get some
people that are experts in the field of
malaria research system is research
tuberculosis research etc to start
putting out there some things to do and
we try to actually provided holes to
those experts to play with the data and
this is a one of the founders of the
ideas to him our he's a lawyer at
berkeley and he's one of the first out
of that paper i mentioned before he says
that if we do science well everything is
going to come into place and we hope
that we actually stein start doing the
science well and i'm going to show you
few slides about science a widow so what
what the talk is going to be about is
tropical disease initiative we don't
have i'm going to show you what is TDI
and how it can affect the drug discovery
pipeline what computational biology can
do what computational biology has done
so far in the drug discovery framework
and finally and probably the most
interesting for you guys i hope you do
not abandon me here is the
that we are actually trying to develop
and where we actually need help from
engineers and programmers projects
within the TDI and TSL so what is the
pipeline for drug discovery for an
average track out there in the market it
takes a very undefined time period and
it's undefined because these initial
priya steps are undefined you can you
can spend your whole life trying to
improve that power here but when you
have some what we call the leads for a
drag which is molecules that look like
they're going to be good candidates for
drug development it takes about 12 years
before it reaches the market okay now
the success rate it goes from one
hundred percent suppose that we've set
the hundred percent here it drops
dramatically at the beginning and it
gets very small so you actually need a
lot of these lead compounds to reach one
drug in the market and the costs
skyrocket when you go into this step so
it's actually very important to two
things lower the time that you spend in
here and improve the accuracy of the
leads because you want to actually have
the success rate higher and the cost
lower because actually the cost is
penalized because you try a lot of
things with things that wouldn't work
anyway so how we do that that's the only
part that TDI is is interested on the
target lead identification we combine
computational tools and interactive
biomedical chemistry this is experiments
to predict what are the targets in your
cells that are going to be interesting
for a particular disease let's say
cancer we want to find all proteins in
your cell that are affected by that
particular cancer and we want to tackle
those and we want to find inhibitors or
leads or compounds that interact with
that particular target and that is done
computationally speaking and we have
developed tons of tools for that then
when we when we have set of initial list
of target inhibitor compounds there is
some chemistry that is going on to
validate those hits and so on and so
forth we go into circles and
why it takes so long because you
actually never reach a consensus until
the point well you have a list of
probable leads and then they sent they
are sent to lead optimization
preclinical etc so TDI we hope he's
going to short the time here and
increase the success rate because TDI
nowadays is possible for several reasons
he wasn't possible about 5-10 years ago
it's possible because we have
completeness of genome projects only
about half a year ago they complete the
malaria genome which is now we have a
whole list of parts of the malaria and
genome so we can start working on the
malaria at the genomic level not just in
particular genes we have new and
complete biological databases that we
can mine and we can actually subtract
some information there and if you guys
would ever go in to look in any of those
biological databases you probably would
jump right away from the window because
we are biologists we are not
computational people and we started
those databases something like 30 years
ago if you look at those is still
probably 20 years back in in
computational science so it's pretty
hard to actually work with databases of
biological data we have new software and
computers which get cheaper and faster
we have internet we didn't have some
time ago at least as spread as it is
today which means more people looking at
it which means less cost of developing a
new track the problems with facing in
TDI is that computational biology alone
is not enough we still need to do some
experiments and experiments are very
expensive from very time constant so
it's the steel limitations where we hope
to overcome those limitations this is
the flowchart that I mentioned to you
before but with more details we have
databases of all type of information in
biology genome sequence protein
structure llegan libraries data mining
from journals it's very important not a
lot of people is doing that and
hopefully the Semantic Web is going to
help towards that is we have millions
millions and millions of texts and
schoolers Google score is doing a great
job on that but we want to associate
one of those particular entries in the
library two genes and two proteins and
that's very difficult because I've given
protein might have 30 different names
okay people have named it so Maxim
differently it's very difficult to find
the proper links between the protein and
the data in the literature do we have
other databases and then we have
chemistry with what we hope is we going
to go around around around around until
we get some leads and companies called
virtual pharma are going to take those
to the production and clinical actually
first clinical preclinical clinicals at
certain search and then production and
then they're going to distribute that
into the third world what are beautiful
formality it's quite interesting
approach it and it's really need
something about 5-10 years it's a people
who has started a non-profit from a
surgical company so nonprofit and
pharmaceutical companies seem not to go
along but they actually do and the first
one is it's based here in the Bay Area
it's the One World Health Organization
and they actually are delivering a thing
in a year so the first drag into an
India village that they actually going
to take the first drag develop under the
help of vehicle farmer they are
nonprofit and what they do is they take
money mostly from Bill Gates Foundation
and an NIH and they do that so that they
lower the cost of production of
particular drugs so we hope that we're
going to be helpful to a bit to a pharma
by getting them better leads so that
they have higher success rate so let's
go into computational biology and there
are tons of tools that computational
biologists have used to develop attracts
you can find a list of them here if you
are interested if any of you guys is
really interested in buying for Maddox I
encourage you to go to that particular
list so what are those tools available
and I'm going to stay here for lunch if
you guys want to know a little bit a
little bit more about by informatics
itself and not just this open source
biology that we're trying to do i'm more
more than happy to talk to you guys
they run about by informatics itself but
this is this is one thing that we're
doing we're trying to map all
protein-ligand universe and ligands are
small compounds when you take a drug
you're actually taking this a small
compound into your body and that is more
compound it's gonna interact with
certain proteins in your body and it's
gonna make you feel good or bad
depending of actually how is your genome
and for those of you that were here
about in February thing as almond was
here giving a talk you probably heard
about these things so we trying to map
the universe of that how we do that by
using there are several groups involved
on that include myself which make all
known protein structures we take all
list of a small ligands and NIH at
Washington has a very good public
available list of drugs we take all
protein-protein interactions pipeline
and we apply our methods of prediction
towards this pipeline so at the end
including a graphical user interface at
the end a user a biologist who go there
and have a whole map of the protein
ligand universe so he wants to know what
a particular drug is interacting to a
particular protein here you go there and
obtain all this information through that
pipeline this is the first project we're
going to be implemented in TDI as the
science part is and we hope this is
going to have a huge effect we're going
to probably start with a particular
genome let's say malaria or tuberculosis
which are very much interest and and
we're going to pipeline all those
genomes through here and make all the
data available these are the databases
for you guys who may be interested there
are tons of different databases i
distributed them in this way we have
sequence-based functional based on a
structure-based this is the world i work
on more or less around here but there
are all these databases and I don't know
which one is scarier but they are the
databases and this is what we have to
deal with it would be nice and actually
Google or somebody indexes all those
databases and creates a unique database
with all
synonyms that connect between them and
actually make them available in an easy
to search way what we do is protein
structure modeling and we have a
software that does it automatically we
take all the sequences at the beginning
we build what we call the profile of
sequences so that we get all the
evolutionary information for the
particular sequence into our database we
scan for known structures and then we
build by comparative modeling the
three-dimensional models and we use that
as an interface to our software we every
software we develop we make an interface
so it's ready available for everybody
two years so what is comparative
modeling is what goes into here and what
we do is win further information it is
known in evolution that if two sequences
are somehow similar or what we call
homologous they're going to have
probably a similar two dimensional
structure so what we do is if we know
this the structure of protein B and we
want to know the structure of protein a
and they are similar we copy all the
coordinates of structure be into the
sequence of a and that's a way we
actually predict the three-dimensional
structure we're doing that in an
automation automatic fashion there at
least tons of algorithms behind that
which I'm happy to talk to you later on
if you are interested on we do that for
all sequences in pdb all sequences in
the world there are about three million
sequences known for proteins everybody
of us carries about 30,000 different
sequences in our body so but there are
tons of genomes out there a
microorganism is probably the most
extensive them there are about three
million known sequences we have models
for about one-third of those for you to
contrast experimentally speaking it is
this number what is known about the
three-dimensional structure of sequences
is only about 30,000 we know
experimentally speaking only 30,000
structures different structures for
those sequences so it is a huge gap
between 3 million and 30,000 and that's
what computational biology is trying to
fulfill and we actually find
doing a pretty good job we are at
one-third to completeness of the
sequences space when we have the
three-dimensional structure and here you
can see one protein particular protein
bound to a ligand this the vegan will be
in yellow and then in red are the
residues of that particular protein that
interact with the ligand when we have
that particular protein ligand
interaction we can actually predict some
things and here is what I show you that
we have tools that are able to obtain
information from the three-dimensional
structure and predict which one of these
ligands and that these are just the ID
code for different ligands ATP this is
what is give you energy in the morning
so we take a protein and we ask ourself
is it binding ATP or not if it binds
where it binds so if you do a random gas
you will get around twenty percent
accuracy which means that eighty percent
of the time you run but applying our
algorithm for ATP we reach at about
eighty percent accuracy so we with our
algorithm take ATP and any given protein
and try to figure out where in the
protein that binds and this is important
because the next step would be to use a
database of drugs and to find an
inhibitor the band's into the same place
that where ATP is binding and therefore
inhibiting the protein to do the
function that is trying to do imagine
that that function is against you don't
will you have in a cancer and you're
trying to inhibit a protein that is
giving you the cancer or the disease so
we want to inhibit it so we go into
databases like this one zinc database
developed by the group of airmen and
shake it at UCSF there is a freely
commerce of a free database of
commercially available drugs what it
means commercially available it means
that if you want to do an experiment you
can actually go into a company invited
by the compound they synthesize it for
you you get it in the lab and then you
try it so it's important it is also
available not all compounds are chemical
we can synthesize them chemically so
it's important it's available so we take
these database
and what we do is we take the structure
you remember that we have predicted
one-third of all the structures in the
PDB in the universe we take and now we
put it in the computer to carry with the
database of ligands and we say I tell
the computer that this is my binding
site where the M is okay by my compounds
are going to be somewhere binding here
and I want to inhibit that binding site
I take all compounds run it to the
computer and that usually takes a lot of
time and then get some heat that I gonna
put into experimental work and the
timeline so this is done computationally
speaking experimentally speaking is also
called on and it's called
high-throughput screening and
pharmaceutical companies do that but
it's extremely expensive okay so
computational time speaking that we save
time and money for example there was
this protein in tuberculosis that we
wanted to inhibit or a reductase it's
just a name of a protein it's a
particular function we took a particular
library and we find heats which are
significantly inhibit in the protein if
you do high-throughput the screen
experimental you get only about two
percent of the hits hitting that
particular protein inhibit inning in
docking we find about six percent so
with the docking is not only faster and
cheaper but it's also sometimes this is
a very good case sometimes it finds hits
that hydro put the screen and couldn't
find and I just keep that slide this is
telling you it same story it's not a
single case that we have been able to do
but we have lots of different proteins
that are affecting different diseases in
which we have used different methods of
docking and they were good enough to
find a particular ligand and then those
ligands have to be optimized and then
they have to go to preclinical clinical
etc those steps that I'm not going to
talk about because we do only basic
research not clinical research also what
one thing that is important a lot of
these drugs they are very good in vitro
but when you take it them orderly your
stomach this of them
and therefore it's not doing any good to
you or sometimes you take them and
basically is affecting with the
secondary effects are even worse than
the disease itself so not everything
looks as good as it is but we we are
advancing in that this is what we have
done so far using computed yes what it
means the complexity of funny girl
talking and making a bad choice how does
it affect the elevator you do question
was a difficult easy what are the most
inner and difficulties about during the
algorithmic docking this in silico
docking but most of the problem is that
we are talking about a very very
detailed representation of the system we
talking that if you move an atom
anatomic position for 0.1 am Strom's
away you may lose some certain
interaction and your energy might go
skyrocketing and therefore it is very
difficult to pick up this little subtle
changes in your energy function because
what you do is you try a lot of
different positions for the atoms pair
each one of those ligands you try a lot
of different positions and your energy
function it has to say this one here is
better than this one here but just a
slight movement might change your energy
function so the limitation that we're
facing is to energy function meaning
that the scoring function are not
perfect and the sampling because you
have to sample so much computational
speaking is expensive and each one of
these experiments may take months of
CPUs so we facing both sampling and
energy fits calling any questions I give
you two proteins and
anymore to figure out which one is
easier to master can compare them and if
we make that so what we do is we
actually take a library of ligands and
doc again a particular protein and tell
you which ligand is the most probable to
be inhibiting that particular protein
but we can also compare to other
proteins we can take now two proteins
and dr. same ligand and see which one of
those are the most probable to be
affecting to be docking the particular
drug so for you to get an idea what is
happening the first heat that used what
we call structure-based drug design
meaning that they took the structures
they look at the structures the use
computational tools and they predict in
certain inhibitors was at the end of the
80s with the HIV protease inhibitor at
the end of the 80s HIV was hot everybody
was actually researching on ITV
everybody was actually looking to find
some inhibitors for that particular
protein Rinus and the group of blonde
ale in England were one of the first to
use knowledge based protein modeling and
design and they actually came out with
good candidates and they are some of
those candidates they are now in the
market so the things are happening more
more things that recent 2003 we had the
science problem it was hot 2003 nowadays
and we're not talking too much about it
because they are not more new cases but
who knows what it's going to happen
these people Lynn dusky and colleagues
took the server they have a server which
is quite interesting it's a it takes
predictions from all our other servers
around the world that they do
predictions about protein structure and
then they collect that prediction is
what we call meta servers they collect
that predictions and they make a
consensus prediction we know that
methods a B and C might be wrong but not
everybody is wrong at the same time and
the same problem so the consensus
actually gave them access to a
particular prediction that nobody else
have
which was interesting enough to be
published in one of the top journals
papers a Cell paper ternal as a possible
target for drug development against
Dysart's so it's an interesting approach
it's done only computationally speaking
and these people have used methods that
they didn't developed to combine the
final information these are actually the
methods they used experimental approach
this is also very nice work done by
jodeci group at UCSF but they do here
they take malaria genome and they
actually go into a cycle malaria the
malaria cycle when it's inside your your
blood cell it goes to a particular cycle
which is very consistent all the time is
doing exactly the same things it doesn't
matter what happened to the surroundings
malaria is strong enough to actually do
all those steps in a given 24 hours
sorry 48 hours so what they did they
study all the genes each line of the of
here represents one gene of the genome
and they studied the expression of the
chin in that way they could know at our
six of the life of malaria in this blood
cell genome a was overexpressed and
genome be was under expressed what it
means I know what genome a is doing so i
can now guess what cell might be doing
okay they find something which is
extremely surprising and it never seen
before is that there is a regular
pattern you see green it means
overexpress red means under express it
says that the genome is regulated
completely regulated expression of this
of the genes follows a cascade when a is
on b b is off etcetera etcetera etcetera
so there is a cascade which means that
that particular organism is extremely
strong it doesn't matter what the
environment is he's going to do what it
has to do so it's going to be difficult
to find a drag to actually disturb that
particular cascade of expression so it's
a it's a very nice piece of work they
have a database with all
this information available which now
somebody has to go and data mining so
there are drugs out there for companies
that are based on drug development on a
structure-based drug development
informatica they have drugs that are
almost going to optimization of leads
vertex they have even drugs out there
let me see if yeah they have two drugs
out of the market predicts etc you name
it sgx lots of these companies around
here in the Bay Area biotech they
actually started using this type of
approach so we reach the final part
which is the TDI again I'd talk to you
about the signs that we're doing in TDA
but now I'm going to talk to you the
particular projects that we have
envisioned for TD a and I hope some of
you guys might be interested on giving
some of these twenty percent of your
time into helping TDA to move forward so
these are the projects there are five
projects in green means that we are in
advanced stage of the project actually
the structure that the prediction is
already done and in orange means that
we're thinking about what we actually
want know how to do them but we still
don't have anything to show and then the
gym basket is something that I hope you
guys are going to like it's something
that we're still thinking about how to
do and this is all in collaboration with
the synaptic paper so tuned cars what we
do here is that I told you there are
tons of databases about particular
information that people know about
proteins ligands and sequences and
function etc so what we have done is we
have collected all this information from
these three different databases biomart
is already a campaign of several
databases so it's not only three but
it's several of them ncbi this is NIH
the National Institute of Health run by
the government of the US has also a lot
a lot of different information from
different databases they have compiled
and put it on the one side so what we
try to do is we take
all the genes by malaria let's say in
that particular time malaria has been
defined that it has about 5,200
different genes and these numbers always
change that's that's the way it is
because we are not good enough to
predict exact number of genes in a
particular genome so we take that and we
collect all the information about
literature we try to make this matching
between literature and jeans annotation
functional annotation structural
information and function and we give
this is a collection of links so when
somebody goes into our website is going
to be able to find a gene card of all
the information the next step we want to
do is we want to move scientist which is
very difficult into the 21st century
tools it sounds strange but scientists
do not use wiki and block okay we do not
because that's not the way we were show
that we could share our information our
information is sure when we publish
something in a paper so we're trying to
actually encourage people to start using
wiki and blogs and we use Drupal so we
have the blogging module but we also
want to develop wiki that takes that
static information that we have now and
lets people to start not modifying
because this is actually experimentally
verified but adding more information /
chain that is something that we want to
want to develop and about two or three
months ago in nature there was a huge
discussion about there were there were
three or four articles about the new
technologies that scientists are not
using and there was an open public a
call for a wicked gene a weekly based on
gene information and and we hope
somebody's going to take that call and
actually developed wiki-based gene
information I told you that we have done
this protein structure modeling this is
the second project and is completed back
because we actually have this pipeline
software that we just press a button and
predict all the structures of the genome
with
the 5270 genes of malaria and we have
been able to build three-dimensional
structures for about two-thirds of the
database of malaria genes that is that
this is a total number of models which
is always higher than the sequences
because you have sequences have several
domains and so on but if you are
interested in this particular second
talk to you the next one is also
interesting it's what I call a voting
system oh there is plenty of people who
knows a lot about malaria okay but we
cannot reach them one to one we want to
do it is to put out there a voting
system what we call target selection
it's a fancy word to say voting system
what we want to put the out there it's
all the genes of malaria with the chin
cards people can look at that and then
and we're going to low users to post
questions such as which is the most
probable sequence or protein that is
going to bind to the particular track
please vote an expert so they are going
to go and vote for a particular gene and
rank those jeans given the voting okay
so that's what we're trying to do for a
structural genomics there is this
professor Ryman Hugh at the University
of Toronto who works on malaria and he
tries to predict all the structures of
malaria his group is dedicated to
malaria research but he wants to know
which ones of all the 5270 genes is the
most interesting one to predict the
structure so we're going to allowed
people to vote and tell ryman even if
they never actually met the ryman this
is a very important protein please the
tremendous structure because if you
actually determine the structure i'm
going to do x and y is pyramids on it
and we hope to find a drag for the
particular structure so this is this is
our our first pilot project for target
selection and we are still liking of the
software the interface software for
people to actually vote because it's not
just of voting to do you have to be sure
that the people actually put in the the
right information in there so we have to
put some filters
this is the wicked gene I already told
you we want to modify that particular
gene cards by experts in the field and
these are actual experts in the field to
to put the input and this is the gym
basket what we're trying to do here is
is similar to yahoo flickers or
delicious are doing and these we want to
do it four chains let's imagine that
this is my set of genes okay and the
information that we have nowadays is
like this you have set of genes that
they do have Sun names I mean some of
them have a name some others do not have
a name but what you can do actually is
you can start comparing them if you
compare them there are tools out there
to compare jeans you're going to know
that oranges and oranges at similar and
they have the same shape that Apple's
but they are not similar at all than
bananas but they actually are fruits all
of them some of them you're going to
find that they are identical so you just
put them together and some of them
you're going to find that this
particular brand of chips goes with this
particular brand of chips for some
others you cannot say anything is just a
gene you don't know anything about it so
the next thing we're going to do we're
going to allow people to actually put
jeans into Dale baskets and tag them and
give some information about them they
can go into internet browse the internet
get into NCBI website for literature and
click into a bookmarklet that they will
bring their basket up and it's going to
ask them okay you just want to put that
particular link or information into this
particular basket which team do you want
to associate it to and then you click
gene a or b to associate that
information this is this is very
powerful I think because while we have
done biologists have done now so far is
automatically we have generated
algorithm that go and their search for
literature and try to associate
automatically the literature or the
pieces of information to genes this is
powerful but it has a lot of false
positives because the prowl
of the synonyms not all proteins are
named the same way and several proteins
have several names so it's a lot of
false-positive in here we're trying to
take the fourth workforce of people and
ask them physically to voluntarily click
and associate things so now what we're
going to have is that we're going to
have things associated apples with
apples oranges with oranges but not only
that if they start putting tags we know
that these are fruits this is automatic
connection before we knew that these two
baskets had the same shape and now we
also know that bananas actually our
fruits as well as apples and we can
associate them this way and not only
that they might have the same pieces of
information that the people have found
out there an Internet associated to
different genes and that's a link that
didn't exist before anywhere not only
that now we go and we ask ourself who is
actually working on that particular
basket or the genes in that particular
basket we have two experts they know
each other they actually work in the
same problem they probably know each
other but the third one they didn't know
that he or she was working in that
particular gene so they can now use the
tools that we have in Drupal to
communicate to each other and start
talking about science which is important
we have to make the life of scientists
easier to work with this huge amount of
data that is out there in the databases
of biology so this is our team basket we
still thinking about how to do that what
best to doing that and if any of you
guys is interested on developing the
tools behind that we will be more than
happy to welcome you this is something
that just came out this week this is the
this week's volume of nature in nature
you can actually write a letter to the
editor and they publish it and you ask
for something a mark resting is at Yale
he's one of the leaders in this country
for bioinformatics he and I didn't past
the initial part but the initial party
started saying Google Earth has been
very good we have now a tool to go
around the wall from point A to point B
and we know what we have to
to go from point A to point B and not
only that we know what we're crossing by
and we know that in this corner that is
the pharmacy store and the other corner
that is x and y why don't we have that
for biological science maps so it would
be extremely nice to have a sort of
flexible Google X map that allows
scientists to actually produce
coordinate for you guys and then we can
actually go and froze it and let's say
that I'm talking about a particular
protein three dimensional structure of a
protein and I want to know how to go
from residue a to residue be towards a
channel that has a specific properties
and it would be nice to just click once
and find this channel because this is
something the biologists do not have and
it would be extremely extremely
extremely useful and I'm just giving you
an example of a protein but let's say
that you want to go from point A in the
cell to point B in the cell going toward
towards passing through the ribosome and
go into a particular compartment in the
cell that would be beautiful to have and
this is something that mark has asked in
the latest nature so when you take it
here finally what you can do for TDI is
that we have to put this concept of the
power of us which means the
collaboration and freely giving away all
your information so that the rest of the
people can actually use it so that
sometimes it's very individual is
biometry church community gets not that
individualist and so we need
collaborative tools and that's what the
TSL people they think synaptically
people are trying to do we we want we
want to reach worldwide we need exposure
and I don't know it's you work on the
search engine it would be nice if
somebody types malaria TDI he goes into
the top in Haines research we need to
explore the port existing bio computing
tools into the Intel
and this is something that biologists
have always struggled because we are
biologists at the end of the day and so
it's difficult for us to actually get it
right when we put things out there and
internet for everybody to use if you are
a researcher or somebody that is just
interested on it if you join you will be
part of an innovative new approach of
research nobody have done open surface
heard so far this is something that
people start talking about you're going
to be able to connect to peers that are
interested in the same problems and
you're going to be able to develop new
research tool that didn't exist before
and by research tool I mean biological
research tools somebody's going to do
biology real biology using using the
tools that you developed and of course
they give good karma to everybody so I
always like to talk to finish my talk
with that slide and it's what I call the
five ninety-five percent ninety-five
percent are the clients of TDI it's
ninety-five percent of biologists out
there who has no idea they don't have an
idea of using by doing bioinformatics
and they would love to have an
easy-to-use interface they give them
access to all the information and
collaboration tools and a and and and
these are the ones that we like and so
far because the five percent which is me
including and if you guys bone to you
probably you're going to end up in this
five percent are the developers who have
to do the work now to develop those
tools so that this ninety-five percent
of people actually can do science using
the TDI trolls so I think we're doing a
good progress here and as anything in
open source and collaborative
communities nowadays we have a plateau
that we hope we don't it's going to
explode at some point and we're going to
get all those volunteers to do the
science at some point and we feeling
that we are pretty close to that we have
about 50 people actively joined the TDI
TSL site but it at any point of time we
have in between 30 and 40 people
visiting the website which is not bad
for something that has been out there
for me
for months now so let me finish by
knowledge in the people the people in my
in our club who has actually done some
of the research I've shown you today the
people involved in the tropical disease
initiative which is myself steven bauer
are too dry and we shall eat thomas
cable Matt Todd the guys at the synaptic
lip lead by ginger taylor and rain
decrease weaken can on the people at the
natural publisher group Kim ohana and
Declan Butler who are running our
website and I encourage you to read more
in these two articles and I have a copy
of this one right here if you want to
take one as well as two flyers for the
TDI and TSL websites and you can go to
this natural vol 4 38 in which there are
those articles talking about scientists
using blogs and wiki's and moving
towards the more collaborative efforts
so it's interesting to read and I think
you guys will be interested to read that
particular and entry in nature if you
want more information about the
particular research I do you can go here
or for the tropical disease or the
synaptic lip you can go here and now
I'll take any question you may have
thank you
so we don't have anything of the human
genome in any of them oh yeah I have to
repeat the question so the question was
how much of the human genome is covered
by the gene cards or the team basket we
only have the malaria genome editing
cars so far but you can go to NCBI
website of the National Institute of
Health and there you have all the human
genome parts and compile we are focusing
only on neglected diseases and for now
we're doing all the genomes for
neglected these ford parasites or
organisms that produce that neglect this
is in particular Jim basket doesn't
exist it's only an idea so far
in city as well
I just like these these other websites
yeah the question was how gene cars
differentiate from other existing
databases that collect this information
also in a concise manner i believe so
the difference is that we are focusing
on a particular genomes one of one of
the difference but that wouldn't be
enough I think is just a matter of
putting a card a card it doesn't contain
all the information just the links and I
could go back but it's not this make a
difference it we collect links we are
not duplicating the data we just
collecting linds and we're trying to
find the synonyms between the databases
so a protein a mighty name aight prime
in another database and we have to make
a link that doesn't exist for all the
databases that we collect the links
because we collect links from different
databases the point here is that the
user of TDI TSL will be able to see at a
glance all the information available for
a particular gene before it makes a
comment or a vote so the additional
value of TD i will be those tools that
do not exist nowadays yet
thank you very much</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>