<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Pimp my Genome! The Mainstreaming of Digital Genetic... | Coder Coacher - Coaching Coders</title><meta content="Pimp my Genome! The Mainstreaming of Digital Genetic... - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/GoogleTechTalks/">GoogleTechTalks</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Pimp my Genome! The Mainstreaming of Digital Genetic...</b></h2><h5 class="post__date">2007-10-08</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/ty_o2Nr-DZY" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">my name is Eunice Carlson and I'm happy
to introduce Andrew hassle here who's
going to talk about genomics here at
Google and this talk is video recorded
for a wider audience which is going to
be available in Google Video later on
and rehearsal has been a genomic
scientist and facilitate the development
and adoption of DNA technology for over
more than 15 years and his first focus
was on sequencing later on he worked on
bioinformatics analyzes since 2003 and
practical writing of DNA code and he's
going to talk more about that which is
quite interesting subject making it
available to lay people he's worked with
leading a genomics group in industry and
academia including Amgen incorporation
fastens oak in California University
Health Network in Toronto Canada MIT and
Cambridge Massachusetts okay thank you
it's a pleasure to be here I don't
expect that many of you have a
background in in genetics or
bioinformatics so you won't need you
will need that to understand this talk
today I haven't really worked in
industry for a number of years or
academia what I've tended to do for the
last few years is go around and tell
stories about what's possible and bring
people together and hook them up to do
some interesting work the last couple of
years has been with MIT and most
recently i'm working with a group in
alberta called the alberta ingenuity
fund which is funds long term science
projects so it's an ideal opportunity to
hook people together so this talk pimp
my genome we were trying to appeal to a
younger audience we're trying to
actually show that you know we're
getting to the point where we can do
some modifications so this talk will
focus on the mainstreaming of what I
call digital genetic engineering biology
is the study of life for those of you
who have no background in biology
historically it has been classifying
animals and putting them into some sort
of evolutionary framework human beings
of course there are considered to be at
the top of the tree this is my friend
Elizabeth and her daughter Leah large
animals like elephants smaller animals
like cats
many many different species of plants
and as we start getting smaller in the
animal kingdom insects and beetles etc
we see a lot more diversity thousands or
millions million many millions of
different species finally moving into
bacteria this is e coli under an
electron microscope and finally viruses
viruses are typically considered living
organisms they're really just
information packages that tend to hijack
various cellular systems lots of
different morphologies as you can see
here we don't know how much life there
is on this planet in terms of diversity
the numbers vary between two million and
a hundred million and I'm I think that
may be off by a factor of 10 or more we
really know very little about
microorganisms in fact a lot of projects
that are going on in science today is
doing something called metagenomics
because most of the microorganisms that
do exist on this planet we can't grow
there's a lot of projects now that will
go out and actually collect materials
from the environment just lice the
materials take the DNA out of it and
actually try and understand what
organisms are there just based on their
DNA code much more computational
analysis this is being applied in a lot
of different areas craig Venter one of
the the sequencer formerly of sanara
celera genomics one of the groups that
was doing the human genome sequencing a
number of years ago has been focusing on
this taking his boat sorcerer 2 out and
taking ocean water samples at various
areas learning a lot about microbial
diversity and viral diversity in the
oceans people are doing this even in the
human gut in the guts of various animals
like cows etc so we're learning a lot
about microbial diversity on the left
moving basically to a subcellular
component we're starting
there's a tremendous amount of
information that's coming out these days
on the left is a bacterial cell it's
basically been squashed Nets it's linear
DNA that's been pressed out of it and an
on and on the right those are human
metaphase chromosomes basically a very
condensed form of DNA there are
thousands millions actually have
different enzymes that operate in the
cell all in real time this is alcohol
dehydrogenase I sure you're familiar
with it if you've ever had a drink it
plugs into a much larger pathways inside
the cell this is in the upper corner
here you'll see just a known biochemical
pathways alcohol dehydrogenase fits in
right here takes alcohol and shunts it
into fatty acid metabolism which is why
you get a beer gut very complex
information all of this working in real
time most of biology isn't digital
though this is typically how biologists
work they write things in notebooks very
hard to get the information out science
one of the leading publications is is
again a print journal and we share a lot
of information at conferences and justin
hallways this is not the most effective
way to do science over the last 15 years
though a lot digital biology has come to
the forefront largely because the human
genome sequencing effort created vast
amounts of genomic data and following
from that we started getting other omics
proteomics the study of proteins
metabolomics the study of metabolism in
cells etc and we realized we're just
getting swamped by all this data we had
to start creating some sort of digital
frame work to make it understandable i'm
going to focus mainly on genomics which
is my area of expertise going back to
the you know the structure of DNA is
fairly recently it was discovered in
1953 so it's the famous picture of
watson and crick and cambridge DNA
itself is is a biochemical information
storage molecule it's digital its base
for not binary but an
has the unusual feature that it can
replicate duplicate perfect fidelity and
copy itself by this unwinding and
enzymatic processes for duplicating each
strand using it as a template I've
always looked at DNA as as essentially
machine language for biochemical
processes going on in cellular systems
genomes are very special programs that
actually have enough information to
encode the processor as well as the
machinery to duplicate the program and
make sure it gets installed on the new
processors that are made not all genomic
circuits not all DNA circuits or
constructs have to have this much
information if there is an existing
operating system in a Cell we can just
add a small kemal bit of DNA that can
just add a new feature or metabolism but
a genome typically is a self-sustaining
program genomes vary widely small
bacterial genomes have about 1.8 million
base pairs of DNA humans have about 3
billion you'll notice that the number of
genes doesn't scale quite the same
escherichia coli has about four thousand
genes actually a human being has
somewhere between 25 thousand and thirty
thousand so bacteria are still very
complex organisms they're just highly
compressed from a computer scientist
point of view DNA can really be looked
at as information on a hard disk zeros
and ones varies very similar ideologies
going on between the two most of the
processes that you will have on reading
and writing information to a hard disk
we can find we can find similar
processes happening in in biochemical
systems in the cell
we didn't create cells that we had to
learn how to read the code and this has
been an ongoing problem you know since
DNA was discovered we started to make
progress in the early 1970s this fellow
is Fred Sanger who developed a method
for reading DNA depicted with this
high-tech tool here this was a really
laborious process for trying to read DNA
code it was toxic it required
radioisotopes it was slow in 1980 if you
could do five hundred base pairs of
sequence a day you were doing really
well a fellow by the name of Lloyd Smith
ended up developing a way to do this in
an automated fashion which was
ultimately commercialized in the late
80s by a company called applied
biosystems and throughput started to
increase this is about the time that the
human genome project was announced still
was knees machines weren't fast enough
to do the human genome but they felt
that the technology was on a roll and
would continue to grow quickly so by
1995 the throughput had gone up to about
a hundred and forty four thousand bases
and 1998 when solera genomics started to
to do the human genome sequence and the
public groups working in a very high
throughput capacity they're using a
machine called the ABI 3704 sub the
human genome today and you know almost
ten years later we have throughputs that
basically are approaching about a Giga
base add a very fast one sequencing run
extremely cheap so the ability to
sequence DNA has grown very rapidly fact
grasped out the rate of the rate of
being able to produce sequence and the
cost of producing sequence essentially
falls to the point in another 10 years
15 years we should be able to see your
entire genome sequenced as a standard by
a therapeutic or diagnostic procedure
that's assuming no dramatic leaps in the
technology however the XPrize foundation
and I notice you have that wonderful
spaceship one hanging in your lobby has
announced late last year that they would
have a ten-million-dollar incentive
prize for technology that would allow
basically a hundred genomes to be
sequenced in 10 days at a cost of about
a thousand dollars per genome so they
they said the first prize was to go and
venture into outer space now they want
to go into inner space so we're going to
see some dramatic developments in the
short term DNA is a language okay so
we've got reading pretty much on track
comprehension has changed a lot over the
last over the last 15 years this was my
first bio informatics machine
essentially and with small little
shareware programs like DNA Strider that
which was more than sufficient for
analyzing the amount of genetic sequence
I had to play with my first copy of
GenBank came on floppies now bottom is a
picture of blue jean l one of the most
powerful supercomputers on the planet
which is now often tasked to various
bioinformatics I processes beyond my
league so biology is moving digitally
this is just a representation of an
enteric genome basically a small gut
bacterium genome we can lay on lots of
different data now most of these types
of genome visualizations and analyses
are pretty much fully automated today we
can do comparative genomics looking at
how one genome compares to another most
of these processes are done very quickly
with the newer machines we can sequence
a bacterial genome in basically one run
and and produce a map of its metabolism
in an afternoon so we've got about five
or six hundred bacterial genomes
available publicly now and more genomes
are being added every day we can also
look at other
areas or windows into the genome this is
what's called comparative genomic
hybridization is basically looking at
changes in copy number from breast
cancer tissues this is the type of thing
we sometimes see where regions of the
genome have been selectively amplified
because of some gene is important and
the cancer process or that's involved in
cancer processes is being up regulated
at a genomic level genomes are not fixed
entities they're highly plastic and
dynamic and more groups are starting to
realize this now and publish data on
this we can also look at things like
gene expression over time this is this
type of heat map is taking a look at
essentially messenger RNAs which are
like working copies of the genome going
out into the cellular metabolism and
being turned into proteins etc this is a
really quick wave of looking at what the
cell is doing at a given point in time
and we can analyze basically every gene
in the genome the human genome in one
single experiment and or any other life
form basically at this point yes sorry
you can basically look at it as a
functional biochemical unit the old the
the old terminology was one gene
produced one protein in the cell but
there's a lot of regulatory genes as
well and other genes that may only act
at the level of controlling some other
process it doesn't necessarily have to
go always to a protein but so you can
just look at it as a functional unit
today the digital biology has been
growing so quickly and pulling in a lot
of different areas against metabolism
proteins reaching out and looking into
literature as well digital literature
and protein-protein interactions etc
it's a complicated web of data now that
has been growing very quickly in fact
we're starting to run into the problem
that you know we keep drilling down
to finer and finer levels into the into
cellular processes and we run up against
complexity we just don't really
understand what's going on and it takes
a tremendous amount of processing power
the more we look the more complexity we
see today just storing and accessing
some of the biological data sets is
becoming a real challenge because
there's no real commercial model to
support the growth of the server systems
and the data systems there most of it is
publicly supported and of course there's
our limited finite comprehension we
simply can't wrap our brains around
biology most of the most of the people
working today will drill down into very
fine specializations so we're getting a
lot of functional barriers between
between different groups and working
this has led to the growth of something
called systems biology which is really
applying algorithms to doing things like
determining gene function doing
diagnostics and being able to make
predictive models and visualizations etc
this is Lee hood he's probably one of
the he is one of the leaders in systems
biology work he is an institute in
Seattle called the Institute for systems
biology and he's he's struggling in some
ways because they still don't even they
can't even agree on a definition of
systems biology today in many ways it's
just biology in a digital age and it's
going to become increasingly complicated
why is it complicated well it's because
we can't always extract the information
of why things are this is a
representation of a structure in a Cell
called a nuclear pore it's basically a
doorway between the inside of the
nucleus where the where DNA is kept in a
eukaryotic cell and the cellular
metabolism as you can see in this
cartoon there's all sorts of little
little parts and pieces in this
mechanism I have no idea how this thing
works I can't it if i needed a poor or
some sort of channel in a synthetic cell
i was building I wouldn't necessarily
recreate this because I
don't know what it can or cannot do this
has been you know the work of hundreds
of researchers to determine this
structure this is an evolved electronic
circuit most electrical engineers
wouldn't be able to figure it out either
it was evolved in an environment that
basically selected for a certain output
this will take the square root of an
input voltage but most we can't
determine how or why it works very
easily it's not it's not designed for
human comprehension it's on it's been
selected for a function possibly the
same way that the nuclear pore has been
selected for a function and we may never
be able to figure it out completely so
evolution a lot of the reasons and and
basically the documentation behind why
something is doing working a certain way
is lost through the evolutionary process
so if we're going to do genetic
engineering that means we have to be
able to write code I put engineering in
quotes because it hasn't really been
engineering up to now it's largely been
one-off art forms and this is where we
have to start changing the way we think
as biologists if we can't build it we
really don't understand it genetic
engineering dates back to the early
1970s the first the first manipulations
of the DNA molecule were essentially
done in 19 72 and 73 they were pretty
simple but they sure set off a chain of
a lot of speculation as to be what would
be possible the main thing that was done
by doing this work was that the species
barrier was dropped we could take DNA
from a plant or a marine organism and
splice it in with humans or any other
animal this simply wasn't possible
before with breeding even then it took a
number of years past 1972 for this to
really start reaching mainstream
consciousness this is what 1977 before
really started making the front page of
time and then the biotech boom starting
in nineteen eighty one the first
I tech boom the wave of companies many
of which still exist today capitalizing
on some of these new techniques DNA and
electronics have in some ways a very
shared history DNA was discovered about
five years after the transistor has
really a lot of potential for the same
potential for large influences on
society in some ways almost very similar
industrial growth curves the companies
that were that that grew out of the
1970s and 1980s shared very similar
economic curves in their growth but have
recently flattened out even looking at
labs level four containment laboratory
where you don't want people getting
infected by the materials working with
looks pretty similar to a chip fab where
you don't want the materials that are
being manipulated infected by humans the
first generation DNA manipulations were
largely done using what's called
splicing gene splicing this is using
molecular scissors to cut DNA at a
certain point as depicted in this
wonderful little cartoon and put sticky
ends together they're basically like
basically like doing ransom note type
work I'll show a little graphic on that
this is a typical molecular biology lab
today it's liquid handling devices all
sorts of chemicals and reagents a lot of
them homemade a more sophisticated unit
with robotic liquid handling being able
to do some high-throughput work and you
know people literally working at lab
benches it's a lot like witchcraft
because they simply cannot see what's
going on they have to trust that the
reactions are proceeding in a certain
way and that things are happening the
way they think they should it's all very
indirect after about ten years working
in biotech I thought there has to be a
better way I took a year off and I went
to this beach in Thailand and just
started thinking processing the last few
years of my life and going where can I
take this to a new level where we can
actually start speeding things up the
biotech industry just seemed to be
getting slower and slower ten years to
make a drug over
billion dollars to get something out it
just seemed like it wasn't going to be
able to respond to the needs of humanity
over time so I started to think a lot
about genomic programming cutting and
splicing applied to text would give you
something like this i sat down with last
month's wired magazine and tried to
write this simple sentence it took me an
hour and 19 minutes it took me about 45
seconds to type it on my word processor
have it spell checked print it and
upload it to my blog for the world so
the technology that we're using today
for most of our genetic engineering is
obsolete and this technology is in use
today worldwide it's the it's still
considered the gold standard at very
best it's it's like typesetting the
words have to exist yes if you if you
put it all together you can you can
print a book but it's not very dynamic
and flexible so we've had this
tremendous advance in sequencing going
from physical DNA to digital DNA over
the last over the last decade the
reverse process seemed obvious we need
synthesis we need to be able to go from
some sort of digital sequence that we
may devise and go back to physical DNA
so we can upload it into the cell and
test it when I started talking with
people about this though it they just
couldn't seem to get it no one wanted to
work building synthesizers no one wanted
to put the effort in and I couldn't
really understand why so I just had to
let it drop for a while we've had the
technology for doing DNA synthesis for
over 20 years it's a very simple process
it's cyclical basically a DNA is built
on a solid glass matrix a new base is
added any reaction that doesn't go to
completion is capped the next base is
added and it goes on in a cyclical
process the problem is errors
accumulates so we can't make very long
chains of DNA we can only after about a
hundred basis you end up getting with a
one percent error you end up having an
error somewhere in that sequence
these are different types of DNA
synthesizers oh sorry are you going to
be able to get that up okay
there we go so these are different DNA
synthesizers that are available from
various manufacturers in the upper left
hand corner is a beckman 8 channel
device up in the upper right our
machines called mermaids there they go
up to 384 channels these were used to do
a lot of the sequencing work in the
public Human Genome Project newer
technologies use the same chemistry's
but just on smaller scale so the reagent
costs are a lot cheaper this is a
microreactor basically working on a lab
on a chip system these are this is the
digital light processor tip from texas
instruments which has about 780,000
elements meaning you can make seven
hundred and eighty thousand different
strands of DNA on a solid glass matrix
in one single experiment and newer
technologies have basically no moving
parts are just ultraviolet LEDs with
capillary tubes you can go I pulled this
up on ebay last night you can go and buy
this equipment basically for next to
nothing $89 pick up your own DNA
synthesizer take it home play around in
your basement this is just showing how
this slide just shows how they put these
smaller fragments together this is where
all the work is we can make DNA for
essentially nothing today but going
through the assembly process taking
these smaller pieces and putting them
together into larger constructs that we
can actually make something useful is
where the technical challenge remains
today there's been a there's been some
very large steps forward in this in the
last few years but here we're just here
they're assembling a gene that glows
green if it assembles properly it's only
714 base pairs but trying to get to
something that's large enough to encode
some
you can buy DNA synthesizers for nothing
now they're basically just big door
stops as you can see the prices vary
we're getting down to around 69 cents a
base pair that's for basically any size
DNA you want small viral genomes will
have about five thousand base pairs
large viral genomes about 200,000 and a
minimal genome somewhere on the order of
300,000 so we're getting to the point
where we can do some pretty interesting
genomic programming having mail-order
compiling send your DNA sequence they'll
give it to you you send it back graft
out Moore's Law is the red line DNA
sequencing is the blue line synthesis is
the green line our capacity for for
synthesizing is growing rapidly and will
probably continue to do so for some time
there are groups around the world that
are that do DNA synthesis now and
essentially it's it's if you have a
credit card and can type you can get DNA
so while I was sitting on that beach in
Thailand I put together another idea
which was well if you can synthesize DNA
and you can essentially program it like
you're programming a computer mmm
neither's you can decide how you're
going to program it you can do a
proprietary and keep all the code for
yourself and try and figure out all the
biology or you might want to try doing
open source biology and so I started
writing about this and and found and
then that introduced me to people that
were also thinking along these lines and
I came across these two guys on the left
is Tom night at MIT and electrical
engineer and on the right is drew endy
one of his colleagues as originally
trained I believe in civil engineering
they both focus now on engineering
biology and they really supported open
source biological programming in fact
they were pretty far along when I found
them so the engineering process is
basically you know have some success
refine it continue to refine it so you
have more success and so on and so on
and the complexity tends to increase
and when you build something you want it
to do what you've what you've designed
it to do it's it's stopping the
evolutionary process in some ways at
certain points it's applied to
electronics of course software
Aeronautics structures materials
automotives but not to biological
engineering so these guys started
engineering biology they took section
they took DNA code and essentially put
it in a wrapper gave it a part number
gave it a functional name this one being
a terminator it stops it stops RNA
polymerase at a certain point so that it
says basically okay and here then they
went out and measured as many spec
parameters of this part as they could
and started to build documentation this
allowed the part to be used with other
parts in assemblies to build systems for
example a sender device that actually
puts a molecule out across a bacterial
membrane put it all this together and we
can start having some sort of a
framework for engineering biology with
parts devices and ultimately systems
that fully assembled could even go on to
make full cells mediated between the DNA
code and parts is really synthesis
everything else is just assembly
processes of the of the parts and
standardized data that can be machine
readable shared between the different
levels you want to design systems you
don't need to know the DNA code you just
have to be able to pull the parts off
the shelf so they built the first
catalog they had some funds for DNA
synthesis they put together a catalog of
various parts show that it's working
made them available open source and
started putting them out to the world
documenting as many as many functions as
they could measure this is their best
part it has the most measurements most
of the parts that aren't don't have that
many measurements test and measurement
and biology remains a big challenge they
also created something called I gem
international genetically engineered
machine
it's an open source program that
essentially that essentially tracks peep
of people that want to play with these
parts and use them and see what they can
build with them it's like a hobby kit
for electronics so it shares everything
parts codes protocols experience
publications only one real shared back
it's grown quickly I've worked with item
since about 2005 I'm always impressed by
by the types of kids that we get out to
it and how much effort they'll put in in
the first years in 2003 was just in
house at MIT as an independent
activities project they opened it up in
2004 with some five friendly schools in
2005 there was there were 13 schools
last year we had 37 schools from 15
countries and it just keeps growing
around the world we just closed
registration for this year 57 teams 20
countries it's really international now
like I was really surprised by the
additions of Russia and you know a
number of teams from China so we're
growing a community of people that can
actually do this work engineering
biology and grow the number of parts
very quickly and it's getting a lot of
attention internationally so which
continues to reinforce the growth of
this program some of the projects that
have come out of this in the last few
years just a light-sensitive bacteria
essentially making a bacteria so you
shine a light on it ends up producing a
color pigment you mix a bacterial film
so here are just a few of the images
they made a hello world of course tip of
the hat Darwin and one of their their
supervisor Andy Andy Ellington from the
University of Texas yes
yeah it's they built a circuit that that
basically was light sensing and pigment
producing and put it into a bacterial
chassis of the background yeah not yet
not yet Oh certainly they were just the
question was whether this was a complete
genome or basically just a couple of
functions added to a bacterial chassis
and indeed it was it is not a complete
genome that's been assembled it's just a
new circuit added to the bacterium this
was a group from Scotland last year that
created a circuit that essentially
allowed built a biosensor for the
detection of arsenic in groundwater
which is a big problem in Bangladesh
they're actually trying to commercialize
this unit it's extremely sensitive
better than the chemical tests that are
available it costs about fifty cents a
test the if the bacteria if there's
arsenic in the water it ends up
producing an acid and a color indicator
turns red means you shouldn't drink the
water a group at MIT decided that their
lab was a little too smelly bacteria has
a auto der so they knocked out that odor
first of all and then they made it so
that the bacteria would smell like
wintergreen or bananas depending on its
growth whether it was in log phase and
actively growing or whether it was in
stationary phase and had used up all its
sugar and here's there they're
demonstrating the smells this actually
has some really practical applications
because a lot of reporter systems
require instrumentation to measure but
our nose is extremely sensitive at parts
per million this is what everyone
competes for an IM the I Jem cup really
that's about it recognition having the
chance to to share their stories at MIT
and a big block of aluminum schools are
around the world are taking notice
almost every school that gets an IM team
starts creating a program Berkeley and
MIT were some of the first now there's
schools all over the world that are
building undergraduate programs and
considering departments this is really
interesting biology is very slow to move
but some of these investments have been
fantastic this fellow J kiessling
received about 42 million dollars in the
Bill and Melinda Gates Foundation for
doing some work on an anti-malarial drug
Berkeley also got a large biofuels
project recently as people move beyond
essentially the IgM program in these
parts they can graduate into something
called open wetware which is model along
the lines of MIT opencourseware looking
forward well according to economists
this technology is just going to
continue to outpace and replace existing
by recombinant DNA techniques and really
accelerate the process I don't know if
we're ready for it there's a lot of
sorry yes the recombinant DNA process is
you're actually working with physical
DNA doing the editing it's like that
it's like the ransom note model with
with synthetic DNA processes you're
basically just typing a sequence of an
electronic text file and sending it
directly for synthesis much faster so
there's a lot of forces driving us along
low-cost DNA synthesis and and global
dissemination of these technologies is
amongst them but there's a lot of risks
as well we don't know how the the
ability to synthesize DNA and really
opens up the ability to create things
like advanced bio weapons be able to
generate viruses that would normally be
under strict government control because
most of the genome sequence of things
like Ebola smallpox anthrax etc are
already in the public domain so this is
there are some there are some concerns
moving forward with this rightly so how
the government responds to it and how
people respond to these technologies
over the next few years will really will
really change the way it's their adopted
if they're if if they become if they
move into a military domain and and
people are highly regulated it could go
underground it could be that this we
just never solve some of the problems
and it the rate of development just ends
up moving very slowly this has been the
case in some of in some of the
engineering that we that has gone on in
the past or it could just open up wide
we could have modular life garage people
working in their garages designing cures
for cancer on using software I'm hoping
that this is the way we move because
this is the way computers have moved
over the last 25 years and I think that
the the changes we've seen in the
electronic world are fantastic but we do
know there's a risk this fellow a fellow
by the name of Wimmer synthesized the
first virus back in 2002 and he did it
was a polio virus and that concerned a
lot of people today we're moving towards
creating a minimal bacterium this is
getting to your question no one has yet
booted up a synthetics fully independent
cell but we expected very soon we have
we're starting to get to the point of
synthesizing genomes large enough to
support a minimal bacterium that's
probably a Nobel Prize you know that's
closing a cycle of evolution that's
billions of years old from primordial
oceans to you know really are being able
to build a bacterium this could very
well lead to a next-generation biotech
industry where that is much more
responsive because they can take in for
example cancer is always a failure of
your genome at some point in in some
cells to respond properly to to and so
the shortest route to correcting it is
to work at the level of the genome of
DNA this type of technology would allow
us to take diagnostic data put it
through a therapeutic engine and
potentially craft a customized solution
just for you using it using us using
algorithms we're already seeing
next-generation biotech companies
appearing focusing on things like
engineering organisms for biofuels
and and therapeutic molecules there's a
there's probably a dozen of these
companies globally now and I see more
and more business plans force for
companies like it because you don't need
the overhead you don't need the labs
drew endy has been going around and
talking with people about buildings
what's what he calls bio fabs
essentially high quality parts along the
same lines as the IgM program but but
but very well designed well documented
that could be used for for much more
serious research or possibly
therapeutics and there's a number of
different groups around the world that
are looking at putting together bio fabs
sharing information sharing parts
there'll be proprietary ones as well
we're also seeing very new ways of going
out and raising support for this type of
work which is considered very
leading-edge this fellow Aubrey de Grey
created his own foundation has raised
millions of dollars towards towards
doing some cellular repair type
engineering and the very interesting man
but you know no one would give him
grants so he just built essentially a
granting agency this is a simulation of
a very simple virus t7 bacteriophage
it's a virus that only infects bacteria
drew and he gave me the simulation that
was done by a couple of people in his
labs sarika suri and jason kelly it's
really fast it's dynamic it's basically
showing the amount of em messenger RNAs
which correlates to proteins in this
virus when it infects a bacterial cell
but there's a problem it really doesn't
correlate to the real world data this is
one of the simplest organisms that we
can work with and we still can't really
tie the models into real-world data yet
we're still learning how to do things
this is a graphic of just a cellular
membrane it's an agent-based model done
by this group in Alberta and they keep
the keep adding to the membrane and it
starts doing these interesting very
getting all these features on the
surface but they don't know why we don't
know this is a very simple model of the
cell we still don't understand the
processes that are going on eventually
this it the model tears into the cell
and takes a look at various channels
that have been made internally but we
this is about as simple as it gets we
still don't understand it quite possibly
it'll take thousands hundreds of
thousands of people contributing their
expertise into some sort of global model
for us to really start getting our heads
wrapped around this I love the idea of
Google Earth and people being able to do
the do sketch up and build you know
their favorite buildings etc maybe one
day we need something like that with
with various molecules operating in
biochemical systems at the level of the
bacteria it probably wouldn't be too
burdensome overall the area that I look
for and what keeps me very interested in
this this is this is from 25 years ago
today may third 19 you know the 1982
it's the kids that are going to really
take this technology and keep pushing it
forward they're attracted to genomic
programming as they were to programming
pcs 25 years ago and I just really hope
that we can provide this technology for
them as early as possible in the
educational system so that they can
become acquainted with it and play with
it 25 years ago you could get these
comics from radioshack that basically
put in an easy-to-read form you know
some of the basics of computing the
group at MIT has done the same thing
this comic describes a lot of the
processes they're doing it's been
translated into a dozen different
languages now will they're working on
issue 2 but this is what really i think
exemplifies where we need to be going
this is a kid i bought at a science
center last year it's a very powerful
DNA extraction kit it comes with all
these little exercises freeze-dried
bacterial plasmid DNA for making what's
called green fluorescent protein and
all the instructions for getting that
DNA into the cell and even comes with a
little LED keychain so you can check to
see if you're so if your cells have
taken up the DNA and are glowing under
UV light wonderful little kit 2495 ages
eight and out I think that's where we
need to be taking these technologies you
know and moving them forward for the
next 25 years and that's basically it
I'd like to thank drew endy for a lot of
the slides and graphics and for for his
work with the I gym program the I gym
program and all the people that I work
with bio air for some of the economic
data that I used in this cyber cell for
some of the models yonas for the
invitation thank you and Alberta
ingenuity for supporting my going around
and talking to people about this work
thank you oh thank you much and andrew
and i just want to mention that audrey
is going to be visiting here in the end
of the month kudos to you who managed to
connect us so he will be here giving
another interesting talk and I leave now
the floor open to questions and
discussions certainly
you spent a lot of time in you're
talking about being a liar or create the
self it seems to me that there's a step
there that you missed from the digital
part which is in your analogy the
program in the cell which is the
computer operates on the program
generator itself that can't do this
completely digitally because you're
going to end up with just a program
executed I don't see your some closures
intransitive right so the question is
basically how do we go from from a
digital program to a fully operating
cellular system that's an open question
at this point there's a groups are
taking a variety of approaches one is we
can take bacterial cells and essentially
destroy the endogenous DNA in that cell
creating kind of an egg so it's a bag of
chemicals that is non metabolizing just
that if we add a genome to it can start
to run that program so that's one route
other groups are taking the approach
well that's that's good but not good
enough and they are trying to create
what are called protocells essentially
fully defined artificial cells that they
could put define synthetic genome in and
have it operate or not not always a
genome sometimes just enough genetic
material to run a metabolic reaction
over a period of time so almost like a
disposable cell so those two approaches
are underway by a number of different
groups it's no one has has successfully
made a a protocell at this point or
fully or booted up the bacterium but we
expected very soon
research party by ocean so basically
right now you want is like something to
be done like breakthrough in biology
right so how computer industry or like
how computer can help you in that phase
so like what's required by one of the
one of the things that we're missing
today I showed you that first catalog of
parts that that is a database that is
not easily isn't very open it is it's
not easy to share the it's hard to put
data in and extract it out we don't have
a simple editor yet where we can just
take apart that someone's defined and
drop it into and and literally do drag
and drop editing of different modules to
create a synthetic constructor circuit I
don't think that's a major programming
effort but but trying to for the a lot
of biology works on in very small groups
sometimes just a few people and trying
to coordinate trying to get someone
tasked full time to even a small project
can be very difficult for a lot of labs
most research labs operated on a few
hundred thousand dollars per year total
i would i would really like if anyone is
interested in doing some sort of
software development for synthetic
biology not going out and trying to
aggregate and pull together and make
sense out of the masses of biological
data in the world the systems biology
effort which again I think in some ways
gets bogged down after a while but if
someone wants to start doing some sort
of programming to facilitate the design
of genomes and and the collection of
parts I'd certainly like to put them
together with some folks at MIT that
have a lot of experience with this I had
another question for you now we've been
successful using recombinant DNA
technology for kind of cloning different
forms of it could be
lamb sheep and so on so what is so
complicated about the bacterium that we
are not able to generate or evolve a
complete bacterium right from scratch
what constitutes the the level of
difficulty well the cloning experiments
are actually a little different reigning
cloning really doesn't do any
re-engineering of the genome it's it's
just it's just taking DNA from one cell
and getting it to work in an egg to
regrow the whole organism so you're not
actually doing genetic engineering on
the genomic material itself okay but for
example some some companies manufacture
therapeutic proteins by taking the gene
for for that protein and putting it into
a different cell so that they can grow
it in a bioreactor right that type of
work is more traditional genetic
engineering it's extremely slow and
difficult because again they're working
with the physical DNA molecule and and
having to use these molecular scissors
to cut and paste it together and it's a
very slow process today we can actually
do that just by using a computer program
output the DNA and do this far faster
and test much more diverse a diverse
space in in the code to make this work
and different groups are doing this now
it is actually cheaper to synthesize
most genes than it is to clone it now so
using the synthesis of genes as against
cloning have you been able to maybe
generate something similar as in let's
say in the past we've seen that
recombinant DNA even though it's a very
slow technology has been able to
generate results so what form of results
have you been able to generate using V
the newer form as in do you have
libraries of different forms of
organisms yes well even just in the
genetic parts that I mentioned we have
over a thousand genetic parts in the
collection now defined so things like
promoters for turning genes on various
coding regions for various proteins
terminators to turn genes off very
switches and control device
we're actually building up a very large
library of parts that are very modular
and can interchange all of the work
that's done with recombinant DNA
technology we can replicate very quickly
but more than that if we know that for
example that we have this
three-dimensional structure of an enzyme
we know where the active site of that
enzyme is where the actual chemical
reactions are occurring we can program
the synthesizers to change the the
nucleic acids in certain regions
ultimately leading to diversity and the
amino acids in those regions and then we
can take all of those variants and put
them through high-throughput screens and
ultimately select for four variants that
have different activities so we're
getting very good at single protein
engineering using synthetic technologies
now it's can we do something more with
that build more complex systems with
fail-safes checksums balances etc and
ultimately synthetic organisms by the
way viruses are a little bit different
than the synthetic bacterial because you
don't actually need the egg so to speak
you just literally take the DNA or the
nucleic material and infect a cell with
it and it will produce virus particles
all you have to do is lysis L and filter
them out okay so in in computer language
we have a control statements which here
we'll just do dictates what a control
flow will go to this function unit or
that function you need so you wonder in
in this synthetic world you there any
such kind of counterparts and how so
look to me that you cannot just assemble
all the functional you need together and
expect the cell to do exactly the same
thing because you need to also have some
logic and direct how the various
function are going on forward and how
they are coordinate with each others so
what's those kind of logic in the
synthetic world we're going to have to
build it this is this is one of the
challenges now we're built the circuits
that we're building
if you're doing it with electronics
would be their toys and yet this is
leading edge genetic engineering trying
to put together complex circuits and
measure how all of those circuits are
working is is one of the challenges that
we all face we don't know how to do it
so we're just starting at the ground up
building the simplest circuits and
trying to to run them and measure them
and test them and have them work
reliably and hopefully we can build
complexity step by step over time and
keep control yes
so when you say synthesis of DNA so if
you if you are given up like a coal for
DNA so how accurate that synthesis can
be made so if there is even a single
point one percent of error or legs let's
say some modification then in terms of
like real human you are like living
being it can be some sort of disasters
also like the the gene synthesis
companies that work today actually make
many times more make multiple versions
of the DNA and they end up having a
sequence to verify that the assemblies
have have worked properly so it's not as
accurate as as a compiler the all the
biological processes have have our fuzzy
there's been some techniques that were
developed over the last few years where
they actually use the the DNA repair
systems that are present even in very
simple cells that look for mismatches of
DNA to to remove some of the to remove
some of the inappropriately assembled
constructs so so we're getting better at
assembly direct from a text file to the
final output but today the gold standard
is we still have to sequence the result
to make sure that it's accurate to the
base pair of course once you actually
load that into the cell and have it
operating there's no there's no
guarantee that it's going to stay stable
because the code can still mutate and
change so these are these are problems
but I guess for the most interesting the
programming language analogy it's
actually quite interesting I think
because I think a little bit more of
data flow and it comes to programming
and more functional we have filter
functions and the flow of data where all
the processes can have continuously
exists and this is more close to how
electrical engineering is design
circuits and so on right that there's no
circuits which is an if it's more over
something being on and off and I think
that's kind of is an interesting analogy
which might work very well with
my humble opinion is there any more
questions here right so the the actual
process is still you make a bunch of
material put it together a bunch of
stuff and then mmm then you look for a
percentage or you somehow we can cause
it to reproduce or something like that
to look at the results it's not like you
make one of something inject into one of
something else it would be great to have
that type of control we don't yet when
when we do the chemical synthesis of DNA
we're working in very small volumes
today Pico moles and pico leaders but
but we're still making thousands of
strands of DNA at once so ultimately
ultimately if this process is done right
only one only one genome goes into the
cell at a time or one circuit into the
cell at a time and then we use that and
then we screen the cells that have taken
up the DNA to make sure that it has the
function that we expect so we use a
functional selection at the end of it
and DNA does tend to be fairly stable
for a number of generations once it's in
the cell so as long as the as long as
the cell isn't being too taxed by
whatever circuit is put in there it
should stay stable for a number for some
time but how do we prevent it from
evolving after that difficult because
DNA seems to be seems to want to change
under under pressure it's very adaptable
material
okay this concludes our session if
there's any more questions please feel
free to meet up afterwards and thank you
very much thank you</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>