<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Brian Naughton, Expert: 23andMe Sample Processing - Tales from the Genome | Coder Coacher - Coaching Coders</title><meta content="Brian Naughton, Expert: 23andMe Sample Processing - Tales from the Genome - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Udacity/">Udacity</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Brian Naughton, Expert: 23andMe Sample Processing - Tales from the Genome</b></h2><h5 class="post__date">2015-02-23</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/jg7V8mlQX2M" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">hi i'm brian acton I'm founding
scientists at 23andme I'm from Dublin
Ireland so after dropping the DNA sample
in the mail we don't see anything until
we get to our results what happens to
that sample once we drop it in the mail
where does it go where does it end up so
your sample actually gets shipped to our
lab in Los Angeles where it gets
processed there are four major
components to that it starts with DNA
extraction then amplification of the DNA
then hybridization and finally analysis
of the data to make sure it's good okay
great about how long does that process
take for to do all those steps so if you
take all of the steps and you did them
end to end it would probably take three
days or so a lot of the process is just
waiting for the equipment to become
available because of volume I say what
of all those sort of four processes
which of those is the most
time-intensive of all of them the most
time-intensive is the hybridization step
which happens overnight amplification
also happens overnight those so there
are two major parts where we have to
leave the DNA for a while and come back
to it so it has time to about what a
what I mean I'm curious what's the most
expensive part of the of the process so
that the chip and the reagents are ready
the expensive part of the process when
you buy the chip you also get the
reagents those are like the chemicals
that you need to run the chip the other
expensive components are probably labor
obviously because we actually need to
hire what's called high complexity staff
to run our process CLS stuff and then
the rest is actually fairly inexpensive
what are the are there any international
restrictions like obviously I'm guessing
if the lab time only takes you like you
said three days but you no longer if
you're processing multiple samples um
you know is it take a lot longer for any
international people who want to have
the service done so the lab process
doesn't take longer obviously the
shipping is an issue and shipping can be
expensive if you're coming from Asia to
California that's part of the reason why
we want to add another lab actually is
to help with that there are also issues
with exporting DNA from certain
countries so China and India
specifically have laws around what DNA
you are allowed export out of that
country so that the that plus other
regulations regarding biological
specimens mean that it's actually quite
difficult to add another country to our
list of countries tell us a bit more
about the DNA extraction step what
actually happened you get the the DNA or
the saliva sample in the lab what do you
do to get the DNA how so DNA extraction
is a very you know normal thing that
happens in labs all over the world
there's nothing special about who we
have we do our DNA extraction but the
process is really trying to separate the
DNA in the nucleus of the cell from the
proteins and other components of the
cell that otherwise would be in the way
I see in I mean how much DNA do you get
out of the saliva sample that someone
sends in it's actually a very variable
so depending on the person we can get
anywhere from 10 micrograms to over a
hundred micrograms DNA a lot and as most
of that variation happened just based on
the number of cells that were in that
spit sample or saliva sample we think so
but we actually don't have very good
information on that we know that some
people saliva is more watery than other
people and we know that for example
infants do not often give us good data
and we think that's partially because of
the water enos of the that's the like
that well for those people who might be
wondering you know what kinds of cells
are you actually getting in my spit what
what are the actual cells that you're
extracting DNA from when you're looking
at my saliva so in a sense it doesn't
matter in that almost all of your cells
have your genome in there and it's
always the same genome so we don't look
at that too closely we know that the
major components there are epithelial
cells so the inside of your cheek the
skin cells there and the and white blood
cells are the other major component now
I i do floss and brush my teeth but I
know that I'm
sure I have some kind of bacteria or
germs in there even still um is that a
concern at all worrying about sort of
having other non me DNA in my spit
sample again it's not something we worry
about too much we know that people have
a lot of bacteria in their mouths and we
know that from person to person there's
huge variability in the composition of
those bacteria but for amplification we
are happy to amplify up all of your
human and bacterial DNA and then let the
sequencing or genotyping process sort
out the human from the bacterium IC so
as long as it's human it's good but does
that mean like I you know I shouldn't go
kiss anybody right before I give my DNA
sample for the 23andme test we actually
collect two milliliters of saliva from
you which doesn't sound like a lot but
it can take up to five to ten minutes to
fill that it's a lot of salida you would
have to be kissing a lot to make a
difference there so when you finish the
first step the the DNA extraction you
essentially have taken our sample and
like you said sort of purified the DNA
from it gotten rid of all the other
stuff two proteins the other molecules
that might be part of cells and other
kinds of contaminants and you pretty
much just have a purified DNA product
exactly now is that enough already to do
what you want to do with it or what do
you have to do next not really it's
common in many situations like this to
want to amplify the DNA so you always
want to give yourself the best
opportunity to sequence what's in there
and the easiest way to do that is to
take the DNA you got and amplify it or
copy it as many times as you can so that
you just have enough signal that you
make it easy on yourself I see so when
you say you're amplifying or you're
making copies I mean are you controlling
kind of how many copies you make you
just you know you're making like 10 more
20 more you know 100 more how does that
work it's not tightly controlled but in
essence at the end of the process you
end up with a thousandfold enrichment of
the DNA so you really do end up with a
lot more DNA so do you end up with DNA
left over you know after you're done
with the procedure you actually have
leftover DNA for four people sometimes
we do but usually we store your D
it's in the saliva form so that we're
going back to the original copy one
issue with amplification is that every
time you copy just like in DNA
replication humans there's a possibility
for error so those errors could compound
if you keep copying the copy just like
xeroxing a Xerox right now we have our
extracted DNA sample that's been
amplified so we have enough of our
sample to do something with are we ready
to go sequence it now is that what we're
going to do so the process that your DNA
undergoes at 23 me is a little different
to sequencing it's called chip
hybridization or genotyping and so how
does this differ from sequencing what's
the difference between sequencing and
sort of which you mentioned is sort of
chip hybridization or for for genotyping
so you can think of sequencing as just
reading the genome from beginning to end
almost you want to know every single
letter and what it is with genotyping we
actually decide beforehand which parts
of the genome which letters are
interesting to us and then we
specifically look at those letters with
the knowledge that these are the letters
that maybe cause diseases or are
associated with some trait that you have
I see so so you actually have to put a
lot of thought into it beforehand to
sort of know exactly which positions
which bases which letters you're
interested in how how many letters do
you actually um or how many bases do you
look at an individual sample it's a good
a good about a million on the current
23andme chip that means that when you
get your results back from us you can
download a million different calls or
base pairs from us sequencing sounds
like it gives you a lot more information
what are the prohibitive aspects of
sequencing why doesn't 23andme do
sequencing yeah sequencing is definitely
the future of this field and we were
definitely interested in sequencing when
it gets cheap enough but the main
advantage of genotyping is cost so we
know that to get the same results that
we get with the 23andme chip we would
have to charge at least 10 times more
with the sequencing product
Wow probably a lot more genotyping has
other advantages as well we because you
genotype using the process of
hybridization you actually get a lot of
different shots on goal if you like you
get to hybridize different molecules
many times and you get the the sum of
all of those or the average of all those
reeds so you get a lot of information
compiled into one data point with
sequencing you might actually miss the
part of the genome you were interested
in and that could be a problem i see so
in a way you kind of have higher
confidence even though you have maybe
more limited information in the end you
have a higher confidence about the
actual you know letter or base result
for that position exactly and some
people describe genotyping is almost
like single base sequencing where we're
seeing we're sequencing hundreds of
thousands of molecules on each part of
the chip so there's an advantage there
that's hard to replicate the sequencing
tell us a bit more about how this sort
of chip hybridization works for people
who are unfamiliar on I here Chip and I
well sometimes I think potato chip but I
know in this case we're probably talking
about something more like maybe a
computer chip how does this work exactly
yeah so the chip is really a glass slide
that contains an array of DNA and those
and the DNA is bound to glass beads that
are scattered on a regular pattern on
the on the chip so then that DNA is
designed by us to bind to the parts of
the genome we're interested in that's
the process I talked about a minute ago
and it comes together through the
process of base pairing just like double
helix base pairing and then either you
have let's say an a it position the
position we're interested in orgy and
depending on which of those two you have
will either light up green or light up
red on the chip at a specific position
on the chip I see so you essentially
have sort of a discrete or quantum unit
for every little DNA molecule that
you're testing you get a little signal
that says oh you have an a or
you have a G here exactly and because
humans are diploid you'll also get
situations where you have a ng one from
other one from father so then what we
see at that position on the chip is red
and green or kind of a yellow signal you
can think of it oh that's really clever
so that it allows you to easily
distinguish any given position that
you're looking at yes so when we get the
data at 23andme we actually can plot it
on a simple XY plot with red on the
y-axis and green and me sorry and the
x-axis and then we'll see three clusters
of people who are a a ag and GG because
those are the three possible situations
for this position in your genome then we
can say well if you're in this cluster
just because the process is called
clustering and your AG and if you're in
this cluster than your AAA so there's a
kind of a simple analysis problem at the
end of the at the end of the output
there I seen it sounds like you're
actually using sort of the information
from a pooled set of individuals you
know you're sort of looking and so even
though an individual call may be
difficult to know where on that graph
your place if you use this clustering
like you said you can more readily
identify oh you're part of this group so
you're a a not AJ kind of thing that's
exactly right and they're also
situations you can imagine where we get
even more information so you could
imagine a situation where you have a
deletion of the part of the genome we're
interested in then what we'll see is a
new cluster another cluster perfectly
three with lower intensity because
there's less signal there because there
are a few or DNA molecules so then a new
cluster could emerge and it could be as
infrequent as one in ten thousand people
or 100,000 people but we'll see a little
nucleus of samples there now in lesson
two of our course we actually talked for
the first time there about the idea of
base pairing and we learn about the base
pairing rules and and how things fit
together what I find really interesting
is the idea that in a double-stranded
DNA you know you have all these letters
matching with each other is is what
you're telling is the idea that a single
nucleotide variant is just one letter
change here that if one letter doesn't
matter
that essentially what's happening is the
the strand can't actually hybridize
completely it can't base pair completely
and that's why you you don't see the
signal for one but you do for another is
that essentially what's going on it's
something very close to that so what we
expect is that all of the letters
leading up to the letter we're
interested in will bind and like you say
if there is a an error there one
difference then it won't find it will
bind weekly so the signal will be weak
but then the specific letter we're
interested in which will kind of come
out one more base has there's a pool of
ASG season T's floating around that can
bind there or not they'll only bind if
it's the correct base pair and it's
unoccupied and it's unoccupied then
we'll see the extension happened the
reaction that leads to the the color you
know the color that we can see in our
imaging system that's great and so
essentially you have this you said the
chip is a slide and you're looking at a
million bases in this one little chip
this one little space yeah so they're
even more than that there's more than a
million beats per chip because it's
better to read the same variant many
many times just to get a good average
signal the reason for that is that if
you're in the corner or maybe you'll get
slightly different optics because the
cameras looking at the center of the
chip or so or something like that so you
just want some redundancy I see it's a
more than 10 million leads per chip and
that process has been improving over the
years the density event of the chips has
been increasing a lot how many chips so
I mean do you have you always use the
same chip or how do you make changes you
know if there's a new I guess I don't
know a new trade or a new new snv that
you become interested in are you able to
change your chip so you can look at that
yes so we update the chip approximately
every 18 months we're on the third
iteration right now and we're constantly
looking for new variants to add to the
chip to improve the performance of it so
you can imagine that if
there's a paper published that says this
mutation in this gene causes a disease
and it's you know it's very interesting
to people to find that out we can design
what we call a probe for that just the
base the sequence of DNA is attached to
the glass bead and make sure it's on the
next iteration of the chip you mentioned
that the hybridization process is sort
of the the longest part he said it
happens over overnight um yeah what I
mean for those of us and we've learned
conceptually about base pairing but any
technical insight you might have I mean
what's why is this an overnight process
essentially it just takes time for the
DNA in your saliva and the extracted
saliva sample to find the right beat so
you want to give it time to just search
around in the solution find the right
bead and bind there and that that takes
time I say it takes time just based on
how quickly the DNA moves around and how
ty Kieth minds and what do you use to
measure the results or you I mean you
just like snapping a picture of the chip
um what's actually happening yes it's
very close to that in essence you get
what's called a fluorophore attached to
the the bead because of the presence of
the a or the G at that position that's
the red or green signal you shoot a
laser at it to excite the floor for and
then you take a photograph of it digital
camera so a lot of it is very much you
know digital camera based so it's that
interesting that that technology allowed
our technology yeah it also sounds like
I mean I know how big my digital camera
picture files are that sounds like it's
a lot of data a lot of information that
you end up with ya so all of that data
from this giant image file gets
processed by software that's actually
run at our lab in Los Angeles and
translate it into the XY plot that I
mentioned before because for each bead
on the chip really all we're interested
in is how much red is there in there and
how much green the photograph itself is
not terribly useful after we've
extracted that data
well it sounds still like there's a lot
of there's there's a lot of stuff that
goes on from the time that you guys get
that raw data and you sort of do your
quality control your QC check before you
know I see it on my 23andme profile page
I mean is is all of that just sort of
computational biology at that point
who's looking at that data who's
organizing it what's sort of going on
there yeah it really becomes a
computational biology problem once we
get the data from the hub and that's
really where 23andme concentrates most
of our staff here are computational
biologists or software engineers that's
really where we're best so the next
major component there is called calling
that's turning the data that we get from
the lab into the AA's ag's or gee-gees
that you can download in your raw data
or produce the reports you see on our
website and that's kind of an involved
process but it in essence relies on
looking at these XY plots and making
sure that everything is where we expect
it to be and so the final results that
we get back seem to look so so curated
and so so official I mean how much of
the computational sort of side of this
is sort of deriving you know what does
this letter mean I mean how do you I
mean are you guys doing all the
literature and and selecting things and
how are those decisions made in general
spiritually have a pretty big content
team at 23andme they're the ones who
scour the literature look at what's
going on in the current science of
genetics and find out this mutation or
this variant causes this disease and
it's a new discovery then they would
write up what they think about that try
to put it in language that makes sense
to our users and if it meets our
scientific criteria it goes out to the
website it is automated and it can take
time just to process the data through
through to the website for the actual
process of
your data calling all of the variants
and all of that is quite quick you can
think of the website is being in two
parts there's the part where we just
look at the calls that you have for
disease and the report just links that
call to the results so for example we
could say you are a carriers of cystic
fibrosis because you have this variant
so that's it that's a scientific
challenge to figure out what's going on
in the literature but it's not an
algorithmic challenge the other part of
the website includes things like our
ancestry composition future and our
relative Finder feature those are
difficult algorithmic problems you want
to take the whole you know genotyped
sample Oh million variants try and
figure out based on that what is this
person's ancestral composition so you
can imagine that that's actually a more
complex task and requires a lot of code
taking it from you know ABC through to
the end</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>