<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Chip Hybridization - Tales from the Genome | Coder Coacher - Coaching Coders</title><meta content="Chip Hybridization - Tales from the Genome - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Udacity/">Udacity</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Chip Hybridization - Tales from the Genome</b></h2><h5 class="post__date">2015-02-23</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/-ni5KxN1C88" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">tell us a bit more about how this sort
of chip hybridization works for people
who are unfamiliar on I here Chip and I
well sometimes I think potato chip but I
know in this case we're probably talking
about something more like maybe a
computer chip how does this work exactly
yeah so the chip is really a glass slide
that contains an array of DNA and those
and the DNA is bound to glass beads that
are scattered on a regular pattern on
the on the chip so then that DNA is
designed by us to bind to the parts of
the genome we're interested in that's
the process I talked about a minute ago
and it comes together through the
process of base pairing just like double
helix base pairing and then either you
have let's say an a it position the
position we're interested in orgy and
depending on which of those two you have
will either light up green or light up
red on the chip at a specific position
on the chip I see so you essentially
have sort of a discrete or quantum unit
for every little DNA molecule that
you're testing you get a little signal
that says oh you have an A or you have a
G here exactly and because humans are
deployed you'll also get situations
where you have a ng one from other one
from father so then what we see at that
position on the chip is red and green or
kind of a yellow signal you can think of
it oh that's really clever so that it
allows you to easily distinguish any
given position that you're looking at
yeah so when we get the data at 23andme
we actually can plot it on a simple XY
plot with red on the y-axis and green
and me sorry on the x-axis and then
we'll see three clusters of people who
are a a ag and GG because those are the
three possible situations for this
position in your genome then we can say
well if you're in this cluster because
the process is called clustering and
your AG and if you're in this cluster
than your AAA so there's a kind of a
simple analysis problem at the end of
the at the end of the output
there I seen it sounds like you're
actually using sort of the information
from a pooled set of individuals you
know you're sort of looking and so even
though an individual call may be
difficult to know where on that graph
you're placed if you use this clustering
like you said you can more readily
identify oh you're part of this group so
you're a a not AJ kind of thing that's
exactly right and there are also
situations you can imagine where we get
even more information so you could
imagine a situation where you have a
deletion of the part of the genome we're
interested in then what we'll see is a
new cluster another cluster perfectly
three with lower intensity because
there's less signal there because there
are fewer DNA molecules so then a new
cluster could emerge and it could be as
infrequent as one in 10,000 people or
100,000 people but we'll see a little
nucleus of samples there now in lesson 2
of our course we actually talked for the
first time there about the idea of base
pairing and we learn about the base
pairing rules and and how things fit
together what I find really interesting
is the idea that in a double-stranded
DNA you know you have all these letters
matching with each other is what you're
telling is the idea that a single
nucleotide variant is just one letter
change here that if one letter doesn't
match that essentially what's happening
is the the strand can't actually
hybridize completely it can't base pair
completely and that's why you you don't
see the signal for one but you do for
another is that essentially what's going
on it's something very close to that so
what we expect is that all of the
letters leading up to the letter we're
interested in will bind and like you say
if there is a an error there one
difference then it won't find a real
bind weekly so the signal will be weak
but then the specific letter we're
interested in which will kind of come
out one more base has there's a pool of
a's geez season T's floating around that
can bind there or not they'll only bind
if it's the correct base pair and it's
unoccupied and it's unoccupied then
we'll see the extension happened the
reaction that leads to the the color
you know the the color that we can see
in our imaging system that's great and
so essentially you have this you said
the chip is a slide and you're looking
at a million bases in this one little
chip this one little space yeah so
they're even more than that there's more
than a million beats per chip because
it's better to read the same variant
many many times just to get a good
average signal the reason for that is
that if you're in the corner or maybe
you'll get slightly different optics
because the camera is looking at the
center of the chip or so or something
like that so you just want some
redundancy I see so it's a more than 10
million leads per tube and that process
has been improving over the years the
density event of the chips has been
increasing a lot how many chips so I
mean do you have you always used the
same chip or how do you make changes you
know if there's a new I guess I don't
know a new trade or a new new snv that
you become interested in are you able to
change your chip so you can look at that
yes so we update the chip approximately
every 18 months we're on the third
iteration right now and we're constantly
looking for new variants to add to the
chip to improve the performance of it so
you can imagine that if there's a paper
published that says this mutation and
this gene causes a disease and it's you
know it's very interesting to people to
find that out we can design what we call
it a probe for that just the base the
sequence of DNA is attached to the glass
bead and make sure it's on the next
iteration of the chip you mentioned that
the hybridization process is sort of the
the longest part he said it happens over
overnight yeah what I mean for those of
us and we've learned conceptually about
base pairing but any technical insight
you might have I mean what's why is this
an overnight process essentially it just
takes time for the DNA in your saliva
and the extracted saliva sample to find
the right beat so you want to give it
time to just search around in the
solution find the right bead
bind their and that that takes time I
say it takes time just based on how
quickly the DNA moves around and how ty
Kieth minds and what do you use to
measure the results or you I mean you
just like snapping a picture of the chip
what's actually happening yes it's very
close to that in essence you get what's
called a fluorophore attached to the the
bead because of the presence of the a or
the G at that position that's the red or
green signal you shoot a laser at it to
excite the floor for and then you take a
photograph of it the digital camera so a
lot of it is very much you know digital
camera based is that interesting that
that technology allowed our technology
yeah it also sounds like I mean I know
how big my digital camera picture files
are that sounds like it's a lot of data
a lot of information that you end up
with yeah so all of our data from this
giant image file gets processed by
software that's actually run at our lab
in Los Angeles and translate it into the
XY plot that I mentioned before because
for each bead on the chip really all
we're interested in is how much red is
there in there and how much green the
photograph itself is not terribly useful
after we've extracted that data</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>